// @slicemachine/core/build/src/models/Library.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LibraryMeta = exports.ComponentInfo = void 0;
const t = __importStar(require("io-ts"));
const Either_1 = require("fp-ts/lib/Either");
const path_1 = __importDefault(require("path"));
const files_1 = __importDefault(require("../utils/files"));
exports.ComponentInfo = {
    hasPreviewsMissing(info) {
        const { screenshotPaths, model } = info;
        if (!screenshotPaths)
            return true;
        return model.variations
            .map((v) => v.id)
            .some((variationId) => !screenshotPaths[variationId]);
    },
};
exports.LibraryMeta = {
    reader: t.exact(t.partial({
        name: t.string,
        version: t.string,
    })),
    build(libPath) {
        const meta = files_1.default.safeReadEntity(path_1.default.join(libPath, "meta.json"), (payload) => {
            return (0, Either_1.getOrElseW)(() => null)(exports.LibraryMeta.reader.decode(payload));
        });
        if (!meta)
            return;
        return {
            name: meta.name,
            version: meta.version,
        };
    },
};
