// @slicemachine/core/build/src/models/Manifest.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManifestHelper = exports.Manifest = void 0;
const t = __importStar(require("io-ts"));
const Framework_1 = require("./Framework");
class ApiEndPointType extends t.Type {
    constructor() {
        super("apiEndpoint", (input) => typeof input === "string", (input, context) => {
            if (typeof input !== "string")
                return t.failure(input, context);
            const regx = new RegExp("^https?://[a-z0-9][a-z0-9-]{2,}[a-z0-9](.cdn)?.(prismic.io|wroom.io|wroom.test)/api/v2/?$", "gi");
            const result = regx.test(input);
            return result
                ? t.success(input)
                : t.failure(input, context, `apiEndpoint should match ${regx.source}`);
        }, t.identity);
    }
}
const apiEndpoint = new ApiEndPointType();
exports.Manifest = t.intersection([
    t.type({
        apiEndpoint,
    }),
    t.partial({
        storybook: t.string,
        localSliceSimulatorURL: t.string,
        libraries: t.array(t.string),
        framework: Framework_1.FrameworksC,
        chromaticAppId: t.string,
        _latest: t.string,
        tracking: t.boolean,
    }),
]);
exports.ManifestHelper = {
    localLibraries(manifest) {
        return (manifest.libraries || [])
            .filter((l) => l.startsWith("~/") || l.startsWith("@/"))
            .map((lib) => {
            const prefix = lib.indexOf("~") != -1 ? `~` : `@`;
            return {
                prefix,
                path: lib.replace(`${prefix}/`, ""),
                fullPath: lib,
            };
        });
    },
};
