// @slicemachine/core/build/src/models/Slice.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SliceMock = exports.SliceAsObject = exports.SliceAsArray = exports.Slice = void 0;
const t = __importStar(require("io-ts"));
const Variation_1 = require("./Variation");
const Variation_2 = require("./Variation");
const Slice = (formatReader) => {
    return t.intersection([
        t.type({
            id: t.string,
            type: t.literal("SharedSlice"),
            name: t.string,
            variations: t.array((0, Variation_2.Variation)(formatReader)),
        }),
        t.partial({
            description: t.string,
        }),
    ]);
};
exports.Slice = Slice;
exports.SliceAsArray = (0, exports.Slice)(Variation_2.AsArray);
exports.SliceAsObject = (0, exports.Slice)(Variation_2.AsObject);
exports.SliceMock = t.array(Variation_1.VariationMock);
