// @slicemachine/core/build/models/CustomType/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomTypes = exports.CustomTypeSM = void 0;
const validators_1 = require("@prismicio/types-internal/lib/validators");
const io_ts_types_1 = require("io-ts-types");
const t = __importStar(require("io-ts"));
const Tab_1 = require("./Tab");
const customtypes_1 = require("@prismicio/types-internal/lib/customtypes");
const Either_1 = require("fp-ts/lib/Either");
exports.CustomTypeSM = t.exact(t.intersection([
    t.type({
        id: t.string,
        label: validators_1.StringOrNull,
        repeatable: (0, io_ts_types_1.withFallback)(t.boolean, true),
        tabs: t.array(Tab_1.TabSM),
        status: (0, io_ts_types_1.withFallback)(t.boolean, true),
    }),
    t.partial({
        hash: t.string,
    }),
]));
exports.CustomTypes = {
    toSM(ct) {
        const tabs = Object.entries(ct.json).map(([tabKey, tabValue]) => Tab_1.Tabs.toSM(tabKey, tabValue));
        return (0, Either_1.getOrElseW)(() => {
            throw new Error("Error while parsing a prismic custom type.");
        })(exports.CustomTypeSM.decode({
            ...ct,
            tabs,
        }));
    },
    fromSM(ct) {
        return (0, Either_1.getOrElseW)(() => {
            throw new Error("Error while parsing an SM custom type.");
        })(customtypes_1.CustomType.decode({
            ...ct,
            json: ct.tabs.reduce((acc, tab) => {
                return {
                    ...acc,
                    [tab.key]: Tab_1.Tabs.fromSM(tab),
                };
            }, {}),
        }));
    },
};
__exportStar(require("./Tab"), exports);
