// @slicemachine/core/build/models/CustomType/Tab.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tabs = exports.TabSM = exports.TabFields = void 0;
const t = __importStar(require("io-ts"));
const nestable_1 = require("@prismicio/types-internal/lib/customtypes/widgets/nestable");
const widgets_1 = require("@prismicio/types-internal/lib/customtypes/widgets");
const Group_1 = require("../../models/Group");
const Slices_1 = require("../Slices");
exports.TabFields = t.array(t.type({
    key: t.string,
    value: t.union([nestable_1.NestableWidget, widgets_1.UID, Group_1.GroupSM]),
}));
exports.TabSM = t.intersection([
    t.type({
        key: t.string,
        value: exports.TabFields,
    }),
    t.partial({
        sliceZone: Slices_1.SlicesSM,
    }),
]);
exports.Tabs = {
    toSM(key, tab) {
        const maybeSz = Object.entries(tab).find(([, value]) => value.type === widgets_1.WidgetTypes.Slices);
        const sliceZone = maybeSz && Slices_1.SliceZone.toSM(maybeSz[0], maybeSz[1]);
        return {
            key,
            value: Object.entries(tab).reduce((acc, [fieldId, value]) => {
                switch (value.type) {
                    case widgets_1.WidgetTypes.LegacySlices:
                    case widgets_1.WidgetTypes.Slices:
                        return acc;
                    case widgets_1.WidgetTypes.Group:
                        return [
                            ...acc,
                            {
                                key: fieldId,
                                value: Group_1.Groups.toSM(value),
                            },
                        ];
                    default:
                        return [...acc, { key: fieldId, value }];
                }
            }, []),
            ...(sliceZone ? { sliceZone } : {}),
        };
    },
    fromSM(tab) {
        const fields = tab.value
            .reduce((acc, { key, value }) => {
            switch (value.type) {
                case widgets_1.WidgetTypes.Group:
                    return [...acc, [key, Group_1.Groups.fromSM(value)]];
                default:
                    return [...acc, [key, value]];
            }
        }, [])
            .concat(!tab.sliceZone
            ? []
            : [[tab.sliceZone.key, Slices_1.SliceZone.fromSM(tab.sliceZone)]])
            .reduce((acc, [k, v]) => ({ ...acc, [k]: v }), {});
        return fields;
    },
};
