// slice-machine-ui/build/lib/mock/Slice.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const Widgets = __importStar(require("./misc/widgets"));
const str_1 = require("../utils/str");
const handlers_1 = require("./misc/handlers");
const createEmptyMock = (sliceName, variation) => ({
    variation: variation.id,
    name: variation.name,
    slice_type: str_1.snakelize(sliceName),
    items: [],
    primary: {},
});
function MockSlice(sliceName, model, mockConfig) {
    const variations = model.variations.map((variation) => {
        var _a;
        const mock = createEmptyMock(sliceName, variation);
        const handler = handlers_1.handleFields(Widgets);
        const maybeMockConfig = ((_a = mockConfig === null || mockConfig === void 0 ? void 0 : mockConfig[variation.id]) === null || _a === void 0 ? void 0 : _a.primary) || {};
        mock.primary = handler(variation.primary, maybeMockConfig);
        const items = [];
        const repeat = variation.items;
        if (!repeat || (repeat && repeat.length === 0)) {
            return {
                ...mock,
                items,
            };
        }
        for (let i = 0; i < Math.floor(Math.random() * 6) + 2; i++) {
            items.push(handler(repeat, mockConfig ? mockConfig.items || {} : {}));
        }
        mock.items = items;
        return mock;
    });
    return variations;
}
exports.default = MockSlice;
