// slice-machine-ui/build/lib/models/common/widgets/Link/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkWidget = exports.linkConfigSchema = void 0;
const yup = __importStar(require("yup"));
const Form_1 = __importStar(require("./Form"));
const fi_1 = require("react-icons/fi");
const Mock_1 = require("./Mock");
const Form_2 = require("./Mock/Form");
const fields_1 = require("../../CustomType/fields");
const type_1 = require("./type");
const Meta = {
    icon: fi_1.FiLink2,
    title: "Link",
    description: "A link to web, media or Prismic document",
};
exports.linkConfigSchema = yup
    .object()
    .shape({
    label: yup.string().optional(),
    useAsTitle: yup.boolean().optional(),
    placeholder: yup.string().optional(),
    select: yup
        .string()
        .optional()
        .oneOf(["media", "document", "web"])
        .nullable(true),
    customtypes: yup.array(yup.string()).strict().optional(),
    masks: yup.array(yup.string()).optional(),
    tags: yup.array(yup.string()).optional(),
    allowTargetBlank: yup.boolean().strict().optional(),
})
    .required()
    .default(undefined)
    .noUnknown(true);
const schema = yup.object().shape({
    type: yup
        .string()
        .matches(/^Link$/, { excludeEmptyString: true })
        .required(),
    config: exports.linkConfigSchema.optional(),
});
exports.LinkWidget = {
    handleMockConfig: Mock_1.handleMockConfig,
    handleMockContent: Mock_1.handleMockContent,
    MockConfigForm: Form_2.MockConfigForm,
    Meta,
    FormFields: Form_1.FormFields,
    schema,
    Form: Form_1.default,
    create: (label) => new type_1.LinkField({ label }),
    TYPE_NAME: fields_1.FieldType.Link,
};
