// slice-machine-ui/build/lib/models/common/widgets/Embed/Mock/Form.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MockConfigForm = void 0;
const react_oembed_container_1 = __importDefault(require("react-oembed-container"));
const theme_ui_1 = require("theme-ui");
const formik_1 = require("formik");
const _1 = require(".");
const consts_1 = require("@lib/consts");
const react_1 = require("react");
const InputDeleteIcon_1 = __importDefault(require("components/InputDeleteIcon"));
const Form = () => {
    var _a;
    const { values, setFieldValue } = formik_1.useFormikContext();
    const contentValue = (_a = values[consts_1.MockConfigKey]) === null || _a === void 0 ? void 0 : _a.content;
    const [state, setState] = react_1.useState({
        ...contentValue,
        err: null,
        loading: false,
    });
    const onSelect = (url, oembed) => {
        setFieldValue(consts_1.MockConfigKey, {
            content: {
                url,
                oembed,
            },
        });
    };
    const reset = () => {
        setState({
            oembed: null,
            loading: false,
            err: null,
            url: "",
        });
        setFieldValue(consts_1.MockConfigKey, {});
    };
    const getOembed = async (url) => {
        setState({ ...state, url });
        if (!url || !url.length) {
            return reset();
        }
        fetch("/api/parse-oembed", {
            headers: {
                Accept: "application/json",
                "Content-Type": "application/json",
            },
            method: "POST",
            body: JSON.stringify({ url }),
        })
            .then(async (res) => {
            const json = await res.json();
            setState({ url, ...json, loading: false });
            if (json.oembed) {
                onSelect(url, json.oembed);
            }
        })
            .catch((e) => {
            console.error(e);
            setState({
                url,
                oembed: null,
                err: "Unable to fetch oembed",
                loading: false,
            });
        });
    };
    return (<theme_ui_1.Box>
      <theme_ui_1.Label variant="label.primary" sx={{ display: "block", maxWidth: "400px" }}>
        <theme_ui_1.Text as="span">Media url</theme_ui_1.Text>
        <theme_ui_1.Input value={state.url || ""} placeholder="https://www.youtube.com/watch?v=fiOwHYFkUz0" onFocus={(e) => e.target.select()} onChange={(e) => getOembed(e.target.value.trim())}/>
        <InputDeleteIcon_1.default onClick={reset}/>
        <theme_ui_1.Box mt={2}>
          {state.err ? <theme_ui_1.Text sx={{ color: "error" }}>{state.err}</theme_ui_1.Text> : null}
          {state.oembed ? (<theme_ui_1.Box>
                <react_oembed_container_1.default markup={state.oembed.html}>
                  <div dangerouslySetInnerHTML={{ __html: state.oembed.html }}/>
                </react_oembed_container_1.default>
                <theme_ui_1.Text mt={1} as="p">
                  Rendered with{" "}
                  <a target="_blank" href="https://www.npmjs.com/package/react-oembed-container">
                    react-oembed-container
                  </a>
                </theme_ui_1.Text>
              </theme_ui_1.Box>) : null}
        </theme_ui_1.Box>
      </theme_ui_1.Label>
    </theme_ui_1.Box>);
};
Form.initialValues = _1.initialValues;
exports.MockConfigForm = Form;
