// slice-machine-ui/build/lib/builders/common/SelectFieldTypeModal/FieldTypeCard.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const theme_ui_1 = require("theme-ui");
const FieldTypeCard = ({ title, description, icon: WidgetIcon, onSelect }) => {
    const { theme } = theme_ui_1.useThemeUI();
    return (<theme_ui_1.Flex sx={{
            p: 3,
            my: 2,
            alignItems: "center",
            cursor: "pointer",
            borderRadius: "3px",
            border: (t) => { var _a; return `1px solid ${(_a = t.colors) === null || _a === void 0 ? void 0 : _a.borders}`; },
            "&:hover": {
                border: ({ colors }) => `1px solid ${colors.primary}`,
                boxShadow: "0 0 0 3px rgba(81, 99, 186, 0.2)",
            },
        }} onClick={onSelect}>
      <WidgetIcon label={`Select field of type "${title}"`} style={{
            color: theme.colors.primary,
            marginRight: "8px",
            borderRadius: "4px",
            padding: "5px",
            border: "2px solid",
            borderColor: theme.colors.primary,
        }} size={32}/>
      <theme_ui_1.Box ml={1}>
        <theme_ui_1.Heading as="h4" sx={{ fontSize: 1 }}>
          <b>{title}</b>
        </theme_ui_1.Heading>
        <theme_ui_1.Text as="p" variant="xs">
          {description}
        </theme_ui_1.Text>
      </theme_ui_1.Box>
    </theme_ui_1.Flex>);
};
exports.default = FieldTypeCard;
