// slice-machine-ui/build/lib/builders/common/SelectFieldTypeModal/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_modal_1 = __importDefault(require("react-modal"));
const SliceMachineModal_1 = __importDefault(require("@components/SliceMachineModal"));
const theme_ui_1 = require("theme-ui");
const Card_1 = __importDefault(require("components/Card"));
const Flex_1 = require("components/Flex");
const FieldTypeCard_1 = __importDefault(require("./FieldTypeCard"));
if (process.env.NODE_ENV !== "test") {
    react_modal_1.default.setAppElement("#__next");
}
const SelectFieldTypeModal = ({ data, close, onSelect, widgetsArray }) => {
    if (!data.isOpen) {
        return null;
    }
    return (<SliceMachineModal_1.default isOpen shouldCloseOnOverlayClick onRequestClose={close} contentLabel="Widget Form Modal">
      <Card_1.default borderFooter footerSx={{ p: 3 }} bodySx={{ pt: 2, pb: 4, px: 4 }} sx={{ border: "none" }} Header={({ radius }) => (<theme_ui_1.Flex sx={{
                p: 3,
                pl: 4,
                bg: "headSection",
                alignItems: "center",
                justifyContent: "space-between",
                borderTopLeftRadius: radius,
                borderTopRightRadius: radius,
                borderBottom: (t) => { var _a; return `1px solid ${(_a = t.colors) === null || _a === void 0 ? void 0 : _a.borders}`; },
            }}>
            <theme_ui_1.Heading>Add a new field</theme_ui_1.Heading>

            <theme_ui_1.Close onClick={close}/>
          </theme_ui_1.Flex>)}>
        <Flex_1.Flex>
          {widgetsArray
            .filter((e) => e)
            .map((widget) => {
            const { Meta, TYPE_NAME, CUSTOM_NAME } = widget;
            return (<Flex_1.Col key={CUSTOM_NAME || TYPE_NAME}>
                    <FieldTypeCard_1.default {...Meta} onSelect={() => onSelect(CUSTOM_NAME || TYPE_NAME)}/>
                  </Flex_1.Col>);
        })}
        </Flex_1.Flex>
      </Card_1.default>
    </SliceMachineModal_1.default>);
};
exports.default = SelectFieldTypeModal;
