// slice-machine-ui/build/lib/env/manifest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.extractRepo = exports.ManifestState = void 0;
const parse_domain_1 = require("parse-domain");
const node_utils_1 = require("@slicemachine/core/build/node-utils");
const io_ts_reporters_1 = require("io-ts-reporters");
var ManifestState;
(function (ManifestState) {
    ManifestState["Valid"] = "Valid";
    ManifestState["NotFound"] = "NotFound";
    ManifestState["InvalidJson"] = "InvalidJson";
})(ManifestState = exports.ManifestState || (exports.ManifestState = {}));
const Messages = {
    [ManifestState.Valid]: "Manifest is correctly setup.",
    [ManifestState.NotFound]: "Could not find manifest file (./sm.json).",
    [ManifestState.InvalidJson]: "Could not parse manifest (./sm.json).",
};
function extractRepo(parsedRepo) {
    switch (parsedRepo.type) {
        case parse_domain_1.ParseResultType.Listed:
            if (parsedRepo.labels.length) {
                return parsedRepo.labels[0];
            }
            if (parsedRepo.subDomains.length) {
                return parsedRepo.subDomains[0];
            }
        default:
            return;
    }
}
exports.extractRepo = extractRepo;
function handleManifest(cwd, validate = false) {
    try {
        const maybeManifest = node_utils_1.retrieveManifest(cwd, validate);
        if (maybeManifest.exists === false) {
            return {
                state: ManifestState.NotFound,
                message: Messages[ManifestState.NotFound],
                content: null,
            };
        }
        if (maybeManifest.errors) {
            const messages = io_ts_reporters_1.formatValidationErrors(maybeManifest.errors, {});
            const message = messages.map((error) => "[sm.json] " + error).join("\n");
            return {
                state: ManifestState.InvalidJson,
                message,
                content: null,
            };
        }
        if (maybeManifest.content === null) {
            return {
                state: ManifestState.InvalidJson,
                message: Messages[ManifestState.InvalidJson],
                content: null,
            };
        }
        const endpoint = parse_domain_1.fromUrl(maybeManifest.content.apiEndpoint);
        const parsedRepo = parse_domain_1.parseDomain(endpoint);
        const repo = extractRepo(parsedRepo);
        return {
            state: ManifestState.Valid,
            message: Messages[ManifestState.Valid],
            content: maybeManifest.content,
            repo,
        };
    }
    catch (e) {
        return {
            state: ManifestState.InvalidJson,
            message: Messages[ManifestState.InvalidJson],
            content: null,
        };
    }
}
exports.default = handleManifest;
