// slice-machine-ui/build/lib/env/npmApi.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.findPackageVersions = exports.isUpdateAvailable = exports.findLatestNonBreaking = exports.findHighestUpdateByKind = void 0;
const npm_registry_fetch_1 = __importDefault(require("npm-registry-fetch"));
const semver_1 = __importDefault(require("semver"));
const axios_1 = __importDefault(require("axios"));
async function fetchVersionsFromNpm(packageName) {
    return npm_registry_fetch_1.default
        .json(packageName)
        .then((json) => {
        const versions = (json.versions || {});
        return Object.keys(versions);
    })
        .catch(() => []);
}
const findHighestUpdateByKind = (kind, current, versions) => {
    const minorVersion = current.replace(/^(\d+\.\d+).*/, "$1");
    const majorVersion = current.replace(/^(\d+).*/, "$1");
    const result = versions.reduce((acc, { version }) => {
        if (/^\d+\.\d+\.\d+$/.test(version) === false)
            return acc;
        if (semver_1.default.gt(acc, version))
            return acc;
        if (kind === "patch" && version.startsWith(minorVersion))
            return version;
        if (kind === "minor" &&
            version.startsWith(majorVersion) &&
            !version.startsWith(minorVersion)) {
            return version;
        }
        if (kind === "major" && !version.startsWith(majorVersion)) {
            return version;
        }
        return acc;
    }, current);
    return result === current ? null : result;
};
exports.findHighestUpdateByKind = findHighestUpdateByKind;
function findLatestNonBreaking(current, versions) {
    const minor = exports.findHighestUpdateByKind("minor", current, versions);
    if (minor)
        return minor;
    const patch = exports.findHighestUpdateByKind("patch", current, versions);
    if (patch)
        return patch;
    return null;
}
exports.findLatestNonBreaking = findLatestNonBreaking;
;
function isUpdateAvailable(current, versions) {
    return Boolean(exports.findHighestUpdateByKind("patch", current, versions) ||
        exports.findHighestUpdateByKind("minor", current, versions) ||
        exports.findHighestUpdateByKind("major", current, versions));
}
exports.isUpdateAvailable = isUpdateAvailable;
async function findPackageVersions(packageName) {
    const versions = await fetchVersionsFromNpm(packageName);
    const stableVersions = versions.filter((version) => /^\d+\.\d+\.\d+$/.test(version) && semver_1.default.lte("0.1.0", version));
    const stableVersionsOrdered = stableVersions.sort().reverse();
    const releaseNotesMap = await axios_1.default
        .get("https://api.github.com/repos/prismicio/slice-machine/releases")
        .then((response) => {
        const releaseNotes = response.data;
        return releaseNotes.reduce((map, releaseNote) => {
            map[releaseNote.name] = releaseNote;
            return map;
        }, {});
    })
        .catch(() => {
        console.log("Couldn't retrieve Github release notes");
        return {};
    });
    return stableVersionsOrdered.map((stableVersion) => {
        return {
            version: stableVersion,
            releaseNote: releaseNotesMap[stableVersion]
                ? releaseNotesMap[stableVersion].body
                : null,
        };
    });
}
exports.findPackageVersions = findPackageVersions;
