// slice-machine-ui/build/lib/env/semver.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createComparator = void 0;
const path_1 = __importDefault(require("path"));
const files_1 = __importDefault(require("../utils/files"));
const paths_1 = require("../models/paths");
const versions_1 = require("./versions");
const versions_2 = require("../../server/src/api/services/versions");
const DefaultComparison = {
    currentVersion: "",
    onlinePackage: null,
    updateAvailable: false,
    updateCommand: "",
    packageManager: "npm",
    availableVersions: [],
};
const createComparator = (pathToPkg) => async ({ cwd }) => {
    try {
        if (!files_1.default.exists(pathToPkg)) {
            return DefaultComparison;
        }
        const versions = await versions_2.findPackageVersions(manifest.name);
        const updateAvailable = versions_1.isUpdateAvailable(manifest.version, versions);
        const isYarnPackageManager = files_1.default.exists(paths_1.YarnLock(cwd));
        const updateCommand = "toto";
        return {
            currentVersion: manifest.version,
            onlinePackage,
            updateAvailable,
            updateCommand,
            packageManager: isYarnPackageManager ? "yarn" : "npm",
            availableVersions: versions,
        };
    }
    catch (e) {
        return DefaultComparison;
    }
};
exports.createComparator = createComparator;
async function comparator({ cwd, }) {
    const pathToPkg = path_1.default.join(cwd, "package.json");
    const compareNpmVersions = exports.createComparator(pathToPkg);
    return await compareNpmVersions({ cwd });
}
exports.default = comparator;
