// slice-machine-ui/build/lib/env/versions.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.findPackageVersions = exports.isUpdateAvailable = exports.findLatestNonBreakingUpdate = exports.getPackageChangelog = void 0;
const npm_registry_fetch_1 = __importDefault(require("npm-registry-fetch"));
const semver_1 = __importDefault(require("semver"));
const axios_1 = __importDefault(require("axios"));
const versions_1 = require("@lib/models/common/versions");
const NodeUtils = __importStar(require("@slicemachine/core/build/node-utils"));
async function getPackageChangelog(dependencyCwd) {
    var _a;
    const pkg = NodeUtils.retrieveJsonPackage(dependencyCwd);
    if (!pkg.exists || !((_a = pkg.content) === null || _a === void 0 ? void 0 : _a.name))
        return {
            currentVersion: "",
            updateAvailable: false,
            latestNonBreakingVersion: null,
            versions: [],
        };
    const currentVersion = pkg.content.version;
    const versions = await findPackageVersions(pkg.content.name);
    const updateAvailable = isUpdateAvailable(currentVersion, versions);
    const latestNonBreakingVersion = findLatestNonBreakingUpdate(currentVersion, versions);
    return {
        currentVersion,
        updateAvailable,
        latestNonBreakingVersion,
        versions,
    };
}
exports.getPackageChangelog = getPackageChangelog;
function findLatestNonBreakingUpdate(current, versions) {
    const minorUpdate = findHighestUpdateByKind("minor", current, versions);
    if (minorUpdate)
        return minorUpdate;
    const patchUpdate = findHighestUpdateByKind("patch", current, versions);
    if (patchUpdate)
        return patchUpdate;
    return null;
}
exports.findLatestNonBreakingUpdate = findLatestNonBreakingUpdate;
function isUpdateAvailable(current, versions) {
    return Boolean(findHighestUpdateByKind("patch", current, versions) ||
        findHighestUpdateByKind("minor", current, versions) ||
        findHighestUpdateByKind("major", current, versions));
}
exports.isUpdateAvailable = isUpdateAvailable;
async function findPackageVersions(packageName) {
    const versions = await fetchVersionsFromNpm(packageName);
    const stableVersions = versions.filter((version) => /^\d+\.\d+\.\d+$/.test(version) && semver_1.default.lte("0.1.0", version));
    const stableVersionsOrdered = stableVersions.sort().reverse();
    const releaseNotesMap = await axios_1.default
        .get("https://api.github.com/repos/prismicio/slice-machine/releases")
        .then((response) => {
        const releaseNotes = response.data;
        return releaseNotes.reduce((map, releaseNote) => {
            map[releaseNote.name] = releaseNote;
            return map;
        }, {});
    })
        .catch(() => {
        return {};
    });
    return stableVersionsOrdered.map((stableVersion, index, versions) => {
        const kind = index !== versions.length - 1
            ? findVersionKind(stableVersion, versions[index + 1])
            : null;
        return {
            versionNumber: stableVersion,
            releaseNote: releaseNotesMap[stableVersion]
                ? releaseNotesMap[stableVersion].body
                : null,
            kind,
        };
    });
}
exports.findPackageVersions = findPackageVersions;
async function fetchVersionsFromNpm(packageName) {
    return npm_registry_fetch_1.default
        .json(packageName)
        .then((json) => {
        const versions = (json.versions || {});
        return Object.keys(versions);
    })
        .catch(() => []);
}
const findHighestUpdateByKind = (kind, current, versions) => {
    const minorVersion = extractMinorVersionFromVersion(current);
    const majorVersion = extractMajorVersionFromVersion(current);
    const result = versions.reduce((acc, { versionNumber }) => {
        if (!/^\d+\.\d+\.\d+$/.test(versionNumber))
            return acc;
        if (semver_1.default.gt(acc, versionNumber))
            return acc;
        if (kind === "patch" && versionNumber.startsWith(minorVersion))
            return versionNumber;
        if (kind === "minor" &&
            versionNumber.startsWith(majorVersion) &&
            !versionNumber.startsWith(minorVersion)) {
            return versionNumber;
        }
        if (kind === "major" && !versionNumber.startsWith(majorVersion)) {
            return versionNumber;
        }
        return acc;
    }, current);
    return result === current ? null : result;
};
const extractMinorVersionFromVersion = (version) => version.replace(/^(\d+\.\d+).*/, "$1");
const extractMajorVersionFromVersion = (version) => version.replace(/^(\d+).*/, "$1");
const findVersionKind = (targetVersion, previousVersion) => {
    if (semver_1.default.lte(targetVersion, previousVersion))
        return null;
    if (isAPatchVersion(targetVersion, previousVersion)) {
        return versions_1.VersionKind.PATCH;
    }
    if (isAMinorVersion(targetVersion, previousVersion)) {
        return versions_1.VersionKind.MINOR;
    }
    if (isAMajorVersion(targetVersion, previousVersion)) {
        return versions_1.VersionKind.MAJOR;
    }
    return null;
};
const isAPatchVersion = (targetVersion, previousVersion) => {
    if (semver_1.default.lte(targetVersion, previousVersion))
        return null;
    const minorVersion = extractMinorVersionFromVersion(previousVersion);
    return targetVersion.startsWith(minorVersion);
};
const isAMinorVersion = (targetVersion, previousVersion) => {
    if (semver_1.default.lte(targetVersion, previousVersion))
        return null;
    const majorVersion = extractMajorVersionFromVersion(previousVersion);
    return (targetVersion.startsWith(majorVersion) &&
        !isAPatchVersion(targetVersion, previousVersion));
};
const isAMajorVersion = (targetVersion, previousVersion) => {
    if (semver_1.default.lte(targetVersion, previousVersion))
        return null;
    const majorVersion = extractMajorVersionFromVersion(previousVersion);
    return !targetVersion.startsWith(majorVersion);
};
