// slice-machine-ui/build/lib/utils/files.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const fs_1 = __importDefault(require("fs"));
const path_1 = __importDefault(require("path"));
const ERROR_CODES = {
    ENOENT: "ENOENT",
};
const Files = {
    _format: "utf8",
    write(pathToFile, value, options = { recursive: true }) {
        if (options.recursive) {
            const directoryPath = path_1.default.dirname(pathToFile);
            Files.mkdir(directoryPath, { recursive: true });
        }
        if (typeof value === "string")
            fs_1.default.writeFileSync(pathToFile, value, Files._format);
        else
            fs_1.default.writeFileSync(pathToFile, JSON.stringify(value, null, 2), Files._format);
    },
    readString(pathToFile) {
        return fs_1.default.readFileSync(pathToFile, { encoding: Files._format });
    },
    readEntity(pathToFile, validate) {
        return validate(JSON.parse(this.readString(pathToFile)));
    },
    safeReadEntity(pathToFile, validate) {
        try {
            const result = this.readEntity(pathToFile, validate);
            if (result instanceof Error)
                return null;
            return result;
        }
        catch (e) {
            return null;
        }
    },
    readJson(pathToFile) {
        return JSON.parse(this.readString(pathToFile));
    },
    safeReadJson(pathToFile) {
        try {
            return JSON.parse(this.readString(pathToFile));
        }
        catch (e) {
            return null;
        }
    },
    readFirstOf(filePaths) {
        return (converter) => {
            return filePaths.reduce((acc, filePath) => {
                if (acc)
                    return acc;
                else {
                    const pathWithOpts = typeof filePath === "string" ? { path: filePath } : filePath;
                    if (this.exists(pathWithOpts.path)) {
                        const optsOrDefault = pathWithOpts.options || {};
                        const test = {
                            path: pathWithOpts.path,
                            ...optsOrDefault,
                            value: converter(this.readString(pathWithOpts.path)),
                        };
                        return test;
                    }
                    else
                        return acc;
                }
            }, undefined);
        };
    },
    isDirectory: (source) => fs_1.default.lstatSync(source).isDirectory(),
    isFile: (source) => fs_1.default.lstatSync(source).isFile(),
    readDirectory: (source) => fs_1.default.readdirSync(source, { encoding: Files._format }),
    mkdir: (target, options) => fs_1.default.mkdirSync(target, options),
    exists(pathToFile) {
        try {
            return Boolean(fs_1.default.lstatSync(pathToFile));
        }
        catch (e) {
            if (e.code === ERROR_CODES.ENOENT)
                return false;
            throw e;
        }
    },
    append(filePath, data) {
        fs_1.default.appendFileSync(filePath, data, { encoding: Files._format });
    },
    copy(src, dest, options = { recursive: false }) {
        if (options.recursive) {
            const directoryPath = path_1.default.dirname(dest);
            Files.mkdir(directoryPath, { recursive: true });
        }
        fs_1.default.copyFileSync(src, dest);
    },
    remove(src) {
        fs_1.default.unlinkSync(src);
    },
    removeAll(srcs) {
        srcs.forEach((src) => Files.remove(src));
    },
    flushDirectories(directory, recursive = true) {
        try {
            const paths = fs_1.default.readdirSync(directory);
            paths.forEach((p) => {
                const maybedir = path_1.default.join(directory, p);
                if (this.isDirectory(maybedir)) {
                    fs_1.default.rmdirSync(maybedir, { recursive });
                }
            });
        }
        catch (e) { }
    },
};
exports.default = Files;
