// slice-machine-ui/build/lib/utils/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.simulatorIsSupported = exports.sanitizeSbId = exports.createDefaultHandleMockContentFunction = exports.createDefaultWidgetValues = exports.compareVariations = exports.ensureWidgetTypeExistence = exports.ensureDnDDestination = exports.removeProp = void 0;
const yup = __importStar(require("yup"));
const fast_deep_equal_1 = __importDefault(require("fast-deep-equal"));
const Framework_1 = require("@slicemachine/core/build/models/Framework");
const defaults_1 = require("../forms/defaults");
const forms_1 = require("../forms");
const removeProp = (obj, prop) => {
    const { [prop]: __removed, ...rest } = obj;
    return rest;
};
exports.removeProp = removeProp;
const ensureDnDDestination = (result) => {
    if (!result.destination || result.source.index === result.destination.index) {
        return true;
    }
    if (result.source.droppableId !== result.destination.droppableId) {
        return true;
    }
    return false;
};
exports.ensureDnDDestination = ensureDnDDestination;
const ensureWidgetTypeExistence = (Widgets, type) => {
    const widget = Widgets[type];
    if (!widget) {
        console.log(`Could not find widget with type name "${type}".`);
        return true;
    }
    return false;
};
exports.ensureWidgetTypeExistence = ensureWidgetTypeExistence;
const compareVariations = (lhs, rhs) => {
    return fast_deep_equal_1.default(lhs.map((e) => ({ ...e, imageUrl: undefined })), rhs.map((e) => ({ ...e, imageUrl: undefined })));
};
exports.compareVariations = compareVariations;
const createDefaultWidgetValues = (TYPE_NAME) => ({
    TYPE_NAME,
    FormFields: defaults_1.DefaultFields,
    schema: yup.object().shape({
        type: yup.string().test({
            name: "type",
            test: function (value) {
                return value === TYPE_NAME;
            },
        }),
        config: forms_1.createValidationSchema(exports.removeProp(defaults_1.DefaultFields, "id")),
    }),
    create: () => ({
        type: TYPE_NAME,
        config: forms_1.createInitialValues(exports.removeProp(defaults_1.DefaultFields, "id")),
    }),
});
exports.createDefaultWidgetValues = createDefaultWidgetValues;
const createDefaultHandleMockContentFunction = (widget, TYPE_NAME, checkFn) => {
    return function handleMockContent(mockContent, config) {
        if (!checkFn(mockContent, config)) {
            console.error(`Type check for type "${TYPE_NAME}" failed. Using default mock configuration`);
            if (widget.handleMockConfig) {
                return widget.handleMockConfig(null, config);
            }
        }
        return mockContent;
    };
};
exports.createDefaultHandleMockContentFunction = createDefaultHandleMockContentFunction;
const sanitizeSbId = (str) => {
    return str
        .toLowerCase()
        .replace(/[ ’–—―′¿'`~!@#$%^&*()_|+\-=?;:'",.<>\{\}\[\]\\\/]/gi, "-")
        .replace(/-+/g, "-")
        .replace(/^-+/, "")
        .replace(/-+$/, "");
};
exports.sanitizeSbId = sanitizeSbId;
const simulatorIsSupported = (framework) => {
    return [
        Framework_1.Frameworks.next,
        Framework_1.Frameworks.nuxt,
        Framework_1.Frameworks.previousNext,
        Framework_1.Frameworks.previousNuxt,
    ].includes(framework);
};
exports.simulatorIsSupported = simulatorIsSupported;
