// slice-machine-ui/build/lib/utils/logger.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.instance = exports.logger = exports.colors = void 0;
const npmlog_1 = __importDefault(require("npmlog"));
exports.instance = npmlog_1.default;
const chalk_1 = __importDefault(require("chalk"));
exports.colors = {
    pink: chalk_1.default.hex("F1618C"),
    purple: chalk_1.default.hex("B57EE5"),
    orange: chalk_1.default.hex("F3AD38"),
    green: chalk_1.default.hex("A2E05E"),
    blue: chalk_1.default.hex("6DABF5"),
    red: chalk_1.default.hex("F16161"),
    gray: chalk_1.default.gray,
};
exports.logger = {
    info: (message) => npmlog_1.default.info("", message),
    plain: (message) => console.log(message),
    line: (count = 1) => console.log(`${Array(count - 1).fill("\n")}`),
    warn: (message) => npmlog_1.default.warn("", message),
    error: (message) => npmlog_1.default.error("", message),
    setLevel: (level = "info") => {
        npmlog_1.default.level = level;
    },
};
