// slice-machine-ui/build/lib/models/common/widgets/StructuredText/Form.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormFields = void 0;
const react_1 = require("react");
const yup = __importStar(require("yup"));
const forms_1 = require("@lib/forms");
const fields_1 = require("@lib/forms/fields");
const defaults_1 = require("@lib/forms/defaults");
const options_1 = __importStar(require("./options"));
const Field_1 = __importDefault(require("@lib/builders/common/EditModal/Field"));
const theme_ui_1 = require("theme-ui");
const Flex_1 = require("@components/Flex");
const SliceMachineIconButton_1 = __importDefault(require("@components/SliceMachineIconButton"));
const isAllSet = (curr) => !options_1.optionValues.find((e) => !curr.includes(e));
const _createInitialOptions = (str) => {
    const arr = str.split(",");
    return options_1.default.reduce((acc, { value }) => {
        if (arr.includes(value)) {
            return [...acc, value];
        }
        return acc;
    }, []);
};
const FormFields = {
    ...defaults_1.DefaultFields,
    allowTargetBlank: fields_1.CheckBox("Allow target blank for links"),
};
exports.FormFields = FormFields;
const accessors = ["config.single", "config.multi"];
const WidgetForm = (props) => {
    const { initialValues, values: formValues, errors, fields, setFieldValue, } = props;
    const { config: { single, multi }, } = formValues;
    const initialOptions = single
        ? _createInitialOptions(single)
        : (multi && _createInitialOptions(multi)) || options_1.optionValues;
    const [isMulti, setIsMulti] = react_1.useState(single ? false : true);
    const [acceptOptions, setAcceptOptions] = react_1.useState(initialOptions);
    react_1.useEffect(() => {
        const fieldNameIndex = isMulti ? 1 : 0;
        setFieldValue(accessors[fieldNameIndex], acceptOptions.join(","));
        setTimeout(() => {
            setFieldValue(accessors[1 - fieldNameIndex], undefined);
        }, 100);
    }, [isMulti, acceptOptions]);
    return (<Flex_1.Flex>
      {Object.entries(FormFields).map(([key, field]) => (<Flex_1.Col key={key}>
            <Field_1.default fieldName={forms_1.createFieldNameFromKey(key)} formField={field} fields={fields} initialValues={initialValues}/>
          </Flex_1.Col>))}
      <Flex_1.Col>
        <theme_ui_1.Label htmlFor="accept" variant="label.primary" sx={{
            mt: 1,
            display: "flex",
            alignItems: "center",
        }}>
          Accept*
          <theme_ui_1.Button type="button" variant="buttons.textButton" onClick={() => {
            if (isAllSet(acceptOptions)) {
                return setAcceptOptions([]);
            }
            setAcceptOptions(options_1.optionValues);
        }}>
            {isAllSet(acceptOptions) ? "Unselect All" : "Select all"}
          </theme_ui_1.Button>
        </theme_ui_1.Label>
        {options_1.default.map((opt) => (<SliceMachineIconButton_1.default fitButton useActive key={opt.value} label={opt.label} size={14} Icon={opt.icon} sx={{
                p: "16px",
                mb: 2,
                mr: 2,
                variant: "buttons.selectIcon",
            }} active={acceptOptions.find((e) => e === opt.value)} onClick={() => {
                if (acceptOptions.find((e) => e === opt.value)) {
                    return setAcceptOptions(acceptOptions.filter((e) => e !== opt.value));
                }
                setAcceptOptions([...acceptOptions, opt.value]);
            }}/>))}
        {errors[accessors[isMulti ? 1 : 0]] ? (<theme_ui_1.Box sx={{ position: "absolute" }}>
              <theme_ui_1.Text as="span" variant="text.labelError" pl={0}>
                {errors[accessors[isMulti ? 1 : 0]]}
              </theme_ui_1.Text>
            </theme_ui_1.Box>) : null}
      </Flex_1.Col>
      <Flex_1.Col>
        <theme_ui_1.Flex sx={{
            mt: 2,
            alignItems: "center",
            height: "100%",
        }}>
          <theme_ui_1.Label variant="label.border">
            <theme_ui_1.Checkbox value={isMulti} defaultChecked={isMulti} onChange={() => setIsMulti(!isMulti)}/>
            Allow multiple paragraphs
          </theme_ui_1.Label>
        </theme_ui_1.Flex>
      </Flex_1.Col>
    </Flex_1.Flex>);
};
const keys = ["single", "multi"];
keys.forEach((key, index) => {
    FormFields[key] = {
        validate: () => yup.string().test({
            name: key,
            message: `Options cannot be empty`,
            test: function () {
                if (this.parent[keys[1 - index]] &&
                    typeof this.parent[keys[1 - index]] === "string") {
                    return true;
                }
                if (typeof this.parent[key] !== "string") {
                    return false;
                }
                if ((!this.parent[key] || !this.parent[key].length) &&
                    (!this.parent[keys[1 - index]] ||
                        !this.parent[keys[1 - index]].length)) {
                    return false;
                }
                const arr = this.parent[key].split(",");
                if (arr.find((e) => !options_1.optionValues.includes(e))) {
                    return false;
                }
                return true;
            },
        }),
    };
});
exports.default = WidgetForm;
