// slice-machine-ui/build/lib/models/common/widgets/StructuredText/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StructuredTextWidget = void 0;
const yup = __importStar(require("yup"));
const Form_1 = __importStar(require("./Form"));
const md_1 = require("react-icons/md");
const Mock_1 = require("./Mock");
const Form_2 = require("./Mock/Form");
const forms_1 = require("../../../../forms");
const utils_1 = require("../../../../utils");
const type_1 = require("./type");
const fields_1 = require("../../CustomType/fields");
const Meta = {
    icon: md_1.MdTextFields,
    title: "Rich Text",
    description: "A rich text field with formatting options",
};
const schema = yup.object().shape({
    type: yup
        .string()
        .matches(/^StructuredText$/, { excludeEmptyString: true })
        .required(),
    config: forms_1.createValidationSchema(utils_1.removeProp(Form_1.FormFields, "id")),
});
exports.StructuredTextWidget = {
    create: (label) => new type_1.StructuredTextField({ label }),
    handleMockConfig: Mock_1.handleMockConfig,
    handleMockContent: Mock_1.handleMockContent,
    FormFields: Form_1.FormFields,
    Meta,
    schema,
    TYPE_NAME: fields_1.FieldType.StructuredText,
    Form: Form_1.default,
    MockConfigForm: Form_2.MockConfigForm,
};
