// slice-machine-ui/build/lib/models/common/widgets/Color/Mock/Form.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MockConfigForm = exports.SelectionCard = void 0;
const react_1 = require("react");
const theme_ui_1 = require("theme-ui");
const formik_1 = require("formik");
const react_color_1 = require("react-color");
const _1 = require(".");
const consts_1 = require("@lib/consts");
const RAND = "random";
const SelectionCard = ({ name, checked, onSelect, children }) => {
    return (<theme_ui_1.Label sx={{
            border: "1px solid",
            borderRadius: "3px",
            borderColor: "borders",
            display: "block",
            position: "relative",
            bg: "headSection",
            mb: 2,
        }}>
      <theme_ui_1.Flex sx={{ p: 2 }}>
        <theme_ui_1.Radio name={name} onChange={onSelect} checked={checked}/>
        {children}
      </theme_ui_1.Flex>
    </theme_ui_1.Label>);
};
exports.SelectionCard = SelectionCard;
const Form = () => {
    var _a;
    const { values, setFieldValue } = formik_1.useFormikContext();
    const contentValue = ((_a = values[consts_1.MockConfigKey]) === null || _a === void 0 ? void 0 : _a.content) || null;
    const [transitionPreserved, preserveTransition] = react_1.useState(false);
    const [lastValue, setLastValue] = react_1.useState(contentValue);
    const onChange = (color) => {
        setLastValue(color.hex);
        setFieldValue(consts_1.MockConfigKey, {
            content: color.hex,
        });
    };
    const reset = () => {
        setFieldValue(consts_1.MockConfigKey, {});
    };
    const startPreserveTransition = () => {
        preserveTransition(true);
        setTimeout(() => preserveTransition(false), 1000);
    };
    return (<theme_ui_1.Box>
      <theme_ui_1.Label variant="label.primary" sx={{ display: "block" }}>
        <theme_ui_1.Text as="span" mb={1} sx={{ display: "inline-block" }}>
          Color value
        </theme_ui_1.Text>
        <exports.SelectionCard value={contentValue} name={RAND} onSelect={reset} checked={contentValue === null}>
          Random Color
        </exports.SelectionCard>
        <exports.SelectionCard name="custom" value={contentValue} onSelect={() => onChange({ hex: lastValue || "#111" })} checked={contentValue !== null}>
          Custom Color
        </exports.SelectionCard>
        {contentValue || transitionPreserved ? (<react_color_1.BlockPicker color={contentValue || lastValue} onChangeComplete={onChange} onChange={startPreserveTransition}/>) : null}
      </theme_ui_1.Label>
    </theme_ui_1.Box>);
};
Form.initialValues = _1.initialValues;
exports.MockConfigForm = Form;
