// slice-machine-ui/build/lib/models/ui/CustomTypeState.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomTypeState = exports.CustomTypeStatus = void 0;
var CustomTypeStatus;
(function (CustomTypeStatus) {
    CustomTypeStatus["New"] = "NEW_CT";
    CustomTypeStatus["Modified"] = "MODIFIED";
    CustomTypeStatus["Synced"] = "SYNCED";
})(CustomTypeStatus = exports.CustomTypeStatus || (exports.CustomTypeStatus = {}));
exports.CustomTypeState = {
    tab(state, tabId) {
        if (state.current.tabs.length) {
            if (tabId)
                return state.current.tabs.find((v) => v.key === tabId);
            return state.current.tabs[0];
        }
    },
    updateTab(state, tabId) {
        return (mutate) => {
            const tabs = state.current.tabs.map((v) => {
                if (v.key === tabId)
                    return mutate(v);
                else
                    return v;
            });
            return {
                ...state,
                current: {
                    ...state.current,
                    tabs,
                },
            };
        };
    },
    deleteTab(state, tabId) {
        const tabs = state.current.tabs.filter((v) => v.key !== tabId);
        return {
            ...state,
            current: {
                ...state.current,
                tabs,
            },
        };
    },
    getPool(tabs) {
        return tabs.reduce((acc, curr) => {
            return [...acc, ...curr.value];
        }, []);
    },
};
