// slice-machine-ui/build/lib/models/common/widgets/ContentRelationship/Form.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormFields = void 0;
const yup = __importStar(require("yup"));
const react_select_1 = __importDefault(require("react-select"));
const theme_ui_1 = require("theme-ui");
const defaults_1 = require("@lib/forms/defaults");
const Field_1 = __importDefault(require("@lib/builders/common/EditModal/Field"));
const Flex_1 = require("components/Flex");
const forms_1 = require("@lib/forms");
const react_redux_1 = require("react-redux");
const customTypes_1 = require("@src/modules/customTypes");
const FormFields = {
    label: defaults_1.DefaultFields.label,
    id: defaults_1.DefaultFields.id,
    customtypes: {
        validate: () => yup.array().of(yup.string()),
    },
};
exports.FormFields = FormFields;
const WidgetForm = ({ initialValues, values: formValues, fields, setFieldValue, }) => {
    const customTypes = react_redux_1.useSelector(customTypes_1.selectLocalCustomTypes);
    const options = customTypes.map((ct) => ({
        value: ct === null || ct === void 0 ? void 0 : ct.id,
        label: ct === null || ct === void 0 ? void 0 : ct.label,
    }));
    const selectValues = formValues.config.customtypes.map((id) => {
        const ct = customTypes.find((e) => e && e.id === id);
        return { value: ct === null || ct === void 0 ? void 0 : ct.id, label: ct === null || ct === void 0 ? void 0 : ct.label };
    });
    return (<Flex_1.Flex>
      {Object.entries(FormFields)
            .filter((e) => e[0] !== "customtypes")
            .map(([key, field]) => (<Flex_1.Col key={key}>
            <Field_1.default fieldName={forms_1.createFieldNameFromKey(key)} formField={field} fields={fields} initialValues={initialValues}/>
          </Flex_1.Col>))}
      <Flex_1.Col>
        <theme_ui_1.Box sx={{
            mt: 2,
            alignItems: "center",
        }}>
          <theme_ui_1.Label htmlFor="origin" mb="1">
            Custom Types
          </theme_ui_1.Label>
          <react_select_1.default isMulti name="origin" options={options} onChange={(v) => {
            if (v) {
                setFieldValue("config.customtypes", v.map(({ value }) => value));
            }
        }} value={selectValues} theme={(theme) => {
            return {
                ...theme,
                colors: {
                    ...theme.colors,
                    text: "text",
                    primary: "background",
                },
            };
        }}/>
        </theme_ui_1.Box>
      </Flex_1.Col>
    </Flex_1.Flex>);
};
exports.default = WidgetForm;
