// slice-machine-ui/build/lib/models/common/widgets/ContentRelationship/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentRelationshipWidget = void 0;
const yup = __importStar(require("yup"));
const Form_1 = __importStar(require("./Form"));
const md_1 = require("react-icons/md");
const fields_1 = require("../../CustomType/fields");
const type_1 = require("./type");
const Link_1 = require("@models/common/widgets/Link");
const Meta = {
    icon: md_1.MdSettingsEthernet,
    title: "Content Relationship",
    description: "Define content relations & internal links",
};
const contentRelationShipConfigSchema = Link_1.linkConfigSchema.shape({
    label: yup.string().max(35, "String is too long. Max: 35"),
    select: yup
        .string()
        .required()
        .matches(/^document$/, { excludeEmptyString: true }),
    customtypes: yup.array(yup.string()).strict().optional(),
});
const schema = yup.object().shape({
    type: yup
        .string()
        .matches(/^Link$/, { excludeEmptyString: true })
        .required(),
    config: contentRelationShipConfigSchema,
});
exports.ContentRelationshipWidget = {
    create: (label) => new type_1.ContentRelationshipField({ label }),
    Meta,
    schema,
    TYPE_NAME: fields_1.FieldType.ContentRelationship,
    FormFields: Form_1.FormFields,
    CUSTOM_NAME: "ContentRelationship",
    Form: Form_1.default,
};
