// @slicemachine/core/build/src/models/CustomType/fields.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Field = exports.SimpleField = exports.FieldTypeCodec = exports.FieldType = void 0;
const t = __importStar(require("io-ts"));
const sliceZone_1 = require("./sliceZone");
var FieldType;
(function (FieldType) {
    FieldType["Boolean"] = "Boolean";
    FieldType["GeoPoint"] = "GeoPoint";
    FieldType["Select"] = "Select";
    FieldType["Color"] = "Color";
    FieldType["Group"] = "Group";
    FieldType["StructuredText"] = "StructuredText";
    FieldType["Image"] = "Image";
    FieldType["Text"] = "Text";
    FieldType["Date"] = "Date";
    FieldType["Link"] = "Link";
    FieldType["LinkToMedia"] = "Link";
    FieldType["ContentRelationship"] = "Link";
    FieldType["Timestamp"] = "Timestamp";
    FieldType["Embed"] = "Embed";
    FieldType["Number"] = "Number";
    FieldType["UID"] = "UID";
    FieldType["IntegrationFields"] = "IntegrationFields";
})(FieldType = exports.FieldType || (exports.FieldType = {}));
exports.FieldTypeCodec = t.keyof({
    Boolean: null,
    GeoPoint: null,
    Select: null,
    Color: null,
    Group: null,
    StructuredText: null,
    Image: null,
    Text: null,
    Date: null,
    Link: null,
    LinkToMedia: null,
    ContentRelationship: null,
    Timestamp: null,
    Embed: null,
    Number: null,
    UID: null,
    IntegrationFields: null,
});
exports.SimpleField = {
    default: { label: "", placeholder: "" },
};
exports.Field = t.intersection([
    t.type({
        type: t.union([t.keyof({ [sliceZone_1.sliceZoneType]: null }), exports.FieldTypeCodec]),
        config: t.unknown,
    }),
    t.partial({
        fieldset: t.string,
    }),
]);
