// slice-machine-ui/build/lib/migrate.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const str_1 = require("./utils/str");
const uniqid_1 = __importDefault(require("uniqid"));
const files_1 = __importDefault(require("./utils/files"));
const paths_1 = require("./models/paths");
const migrate = (model, info, env, write = true) => {
    const { type, fieldset, "non-repeat": nonRepeat = {}, repeat = {} } = model;
    if (type !== "Slice") {
        return { model, migrated: false };
    }
    const newModel = {
        id: str_1.snakelize(info.sliceName),
        type: "SharedSlice",
        name: info.sliceName,
        description: fieldset,
        variations: [
            {
                id: "default-slice",
                name: "Default slice",
                docURL: "...",
                version: uniqid_1.default(),
                description: fieldset,
                primary: nonRepeat,
                items: repeat,
            },
        ],
    };
    if (write) {
        const modelPath = paths_1.CustomPaths(env.cwd)
            .library(info.from)
            .slice(info.sliceName)
            .model();
        files_1.default.write(modelPath, newModel);
    }
    return { model: newModel, migrated: true };
};
exports.default = migrate;
