// slice-machine-ui/build/lib/schemas.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.libraries = exports.slice = void 0;
const yup = __importStar(require("yup"));
exports.slice = yup
    .object()
    .shape({
    extension: yup.string().required(),
    fileName: yup.string().required(),
    from: yup.string().required(),
    hasMock: yup.boolean().required(),
    hasModel: yup.boolean().required(),
    isDirectory: yup.boolean().required(),
    migrated: yup.boolean().required(),
    mock: yup.object().shape({}),
    model: yup
        .object()
        .shape({
        variations: yup.array().required(),
    })
        .required(),
    pathToSlice: yup.string().required(),
    hasPreview: yup.boolean().required(),
    previewUrl: yup.string(),
    sliceName: yup.string().required(),
})
    .required();
exports.libraries = yup
    .array()
    .min(1)
    .of(yup.array().test({
    name: "example text",
    message: "${path} is not right",
    test: function (value) {
        return true;
    },
}));
