// slice-machine-ui/build/lib/mock/misc/widgets.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UID = exports.Timestamp = exports.Text = exports.StructuredText = exports.Select = exports.Number = exports.Link = exports.Image = exports.GeoPoint = exports.Embed = exports.Date = exports.Color = exports.Boolean = void 0;
const Boolean = __importStar(require("../../models/common/widgets/Boolean/Mock"));
exports.Boolean = Boolean;
const Color = __importStar(require("../../models/common/widgets/Color/Mock"));
exports.Color = Color;
const Date = __importStar(require("../../models/common/widgets/Date/Mock"));
exports.Date = Date;
const Embed = __importStar(require("../../models/common/widgets/Embed/Mock"));
exports.Embed = Embed;
const GeoPoint = __importStar(require("../../models/common/widgets/GeoPoint/Mock"));
exports.GeoPoint = GeoPoint;
const Image = __importStar(require("../../models/common/widgets/Image/Mock"));
exports.Image = Image;
const Link = __importStar(require("../../models/common/widgets/Link/Mock"));
exports.Link = Link;
const Number = __importStar(require("../../models/common/widgets/Number/Mock"));
exports.Number = Number;
const Select = __importStar(require("../../models/common/widgets/Select/Mock"));
exports.Select = Select;
const StructuredText = __importStar(require("../../models/common/widgets/StructuredText/Mock"));
exports.StructuredText = StructuredText;
const Text = __importStar(require("../../models/common/widgets/Text/Mock"));
exports.Text = Text;
const Timestamp = __importStar(require("../../models/common/widgets/Timestamp/Mock"));
exports.Timestamp = Timestamp;
const UID = __importStar(require("../../models/common/widgets/UID/Mock"));
exports.UID = UID;
