// slice-machine-ui/build/lib/models/common/widgets/Select/FormFields.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const yup = __importStar(require("yup"));
const FormFields_1 = require("components/FormFields");
const types_1 = require("@lib/forms/types");
const defaults_1 = require("@lib/forms/defaults");
const FormFields_2 = require("components/FormFields");
const FormFields = {
    ...defaults_1.DefaultFields,
    default_value: {
        type: types_1.FormTypes.CHECKBOX,
        yupType: "string",
        defaultValue: null,
        validate: function () {
            return yup.string().test({
                name: "default_value",
                message: 'Default value is not part of field "options" in Select',
                test: function (value) {
                    return (value === undefined ||
                        (this.parent &&
                            this.parent.options &&
                            this.parent.options.find((e) => e === value)));
                },
            });
        },
        component: (props) => (<FormFields_1.FormFieldCheckboxControl {...props} defaultValue={props.field.value} getFieldControl={(formValues) => { var _a; return (_a = formValues.config) === null || _a === void 0 ? void 0 : _a.options; }} setControlFromField={(options, isChecked) => isChecked ? options.length && options[0] : undefined} label={(options) => `use first value as default ${options.length ? `("${options[0]}")` : ""}`}/>),
    },
    options: {
        yupType: "array",
        defaultValue: ["1", "2"],
        validate: {
            required: ["Select requires a minimum of 2 options"],
            min: [2, "Choose at least 2 options"],
            test: {
                name: "non-empty values",
                message: "Values cannot be empty",
                test(value) {
                    return !value.some((e) => !e || !e.length);
                },
            },
        },
        component: (props) => {
            return (<FormFields_2.FormFieldArray label="Options" inputPlaceholder="Select option (eg. 'image_left')" buttonLabel="Add option" {...props}/>);
        },
    },
};
exports.default = FormFields;
