// slice-machine-ui/build/lib/models/common/widgets/Select/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SelectWidget = void 0;
const yup = __importStar(require("yup"));
const md_1 = require("react-icons/md");
const forms_1 = require("../../../../forms");
const utils_1 = require("../../../../utils");
const FormFields_1 = __importDefault(require("./FormFields"));
const Mock_1 = require("./Mock");
const Form_1 = require("./Mock/Form");
const type_1 = require("./type");
const fields_1 = require("../../CustomType/fields");
const Meta = {
    icon: md_1.MdDns,
    title: "Select",
    description: "A rich text field with formatting options",
};
const schema = yup.object().shape({
    type: yup
        .string()
        .matches(/^Select$/, { excludeEmptyString: true })
        .required(),
    config: forms_1.createValidationSchema(utils_1.removeProp(FormFields_1.default, "id")),
});
exports.SelectWidget = {
    FormFields: FormFields_1.default,
    handleMockConfig: Mock_1.handleMockConfig,
    handleMockContent: Mock_1.handleMockContent,
    MockConfigForm: Form_1.MockConfigForm,
    create: (label) => new type_1.SelectField({ label }),
    schema,
    Meta,
    TYPE_NAME: fields_1.FieldType.Select,
};
