// slice-machine-ui/build/server/src/api/screenshots/generate.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.removeCustomScreenshot = exports.generateScreenshot = exports.generateScreenshotAndRemoveCustom = void 0;
const files_1 = __importDefault(require("@lib/utils/files"));
const NodeUtils = __importStar(require("@slicemachine/core/build/node-utils"));
const puppeteer_1 = __importDefault(require("./puppeteer"));
const screenshot_1 = require("@slicemachine/core/build/libraries/screenshot");
const ComponentUI_1 = require("@lib/models/common/ComponentUI");
const IO = __importStar(require("../io"));
async function generateScreenshotAndRemoveCustom(env, libraryName, sliceName) {
    const { screenshots, failure } = await generateScreenshot(env, libraryName, sliceName);
    Object.keys(screenshots).forEach((variationId) => removeCustomScreenshot(env, libraryName, sliceName, variationId));
    return {
        screenshots: screenshots,
        failure: failure,
    };
}
exports.generateScreenshotAndRemoveCustom = generateScreenshotAndRemoveCustom;
async function generateScreenshot(env, libraryName, sliceName) {
    const slice = IO.Slice.readSlice(NodeUtils.CustomPaths(env.cwd).library(libraryName).slice(sliceName).model());
    const variationIds = slice.variations.map((v) => v.id);
    const promises = variationIds.map((id) => generateForVariation(env, libraryName, slice, id));
    const results = await Promise.allSettled(promises);
    return results.reduce((acc, result, index) => {
        const key = variationIds[index];
        const screenshots = result.status === "fulfilled"
            ? {
                ...acc.screenshots,
                [key]: result.value,
            }
            : acc.screenshots;
        const failure = result.status === "rejected"
            ? [
                ...acc.failure,
                { error: result.reason, variationId: key },
            ]
            : acc.failure;
        return { screenshots, failure };
    }, { screenshots: {}, failure: [] });
}
exports.generateScreenshot = generateScreenshot;
async function generateForVariation(env, libraryName, slice, variationId) {
    const screenshotUrl = `${env.manifest.localSliceSimulatorURL}?lid=${encodeURIComponent(libraryName)}&sid=${encodeURIComponent(slice.id)}&vid=${encodeURIComponent(variationId)}`;
    const pathToFile = NodeUtils.GeneratedPaths(env.cwd)
        .library(libraryName)
        .slice(slice.name)
        .variation(variationId)
        .preview();
    await puppeteer_1.default.handleScreenshot({
        screenshotUrl,
        pathToFile,
    });
    return ComponentUI_1.createScreenshotUI(env.baseUrl, pathToFile);
}
function removeCustomScreenshot(env, libraryName, sliceName, variationId) {
    const maybeCustomScreenshot = screenshot_1.resolvePathsToScreenshot({
        paths: [env.cwd],
        from: libraryName,
        sliceName,
        variationId,
    });
    if (maybeCustomScreenshot)
        files_1.default.remove(maybeCustomScreenshot.path);
}
exports.removeCustomScreenshot = removeCustomScreenshot;
