// slice-machine-ui/build/server/src/api/screenshots/screenshots.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateEnv = void 0;
const getEnv_1 = __importDefault(require("../services/getEnv"));
const generate_1 = require("./generate");
const utils_1 = require("@lib/utils");
function validateEnv(framework, simulatorUrl) {
    if (!utils_1.simulatorIsSupported(framework)) {
        const reason = "Could not generate preview: framework is not supported";
        return {
            err: new Error(reason),
            reason,
            screenshots: {},
        };
    }
    if (!simulatorUrl) {
        const reason = "Could not generate screenshots: localSlicePreviewURL undefined in sm.json file";
        return {
            err: new Error(reason),
            reason,
            screenshots: {},
        };
    }
}
exports.validateEnv = validateEnv;
async function handler({ libraryName, sliceName, }) {
    const { env } = await getEnv_1.default();
    const maybeErr = validateEnv(env.framework, env.manifest.localSliceSimulatorURL);
    if (maybeErr) {
        return maybeErr;
    }
    try {
        const { screenshots, failure } = await generate_1.generateScreenshotAndRemoveCustom(env, libraryName, sliceName);
        if (failure.length > 0) {
            const message = `Could not generate screenshots for variations: ${failure
                .map((f) => f.variationId)
                .join(" | ")}`;
            const isError = Object.keys(screenshots).length === 0;
            return {
                err: isError ? new Error(message) : null,
                reason: isError ? message : null,
                warning: isError ? null : message,
                screenshots,
            };
        }
        return {
            err: null,
            reason: null,
            screenshots,
        };
    }
    catch (e) {
        const crashMessage = "Could not generate screenshots for this slice, upload manually a screenshot";
        return {
            err: new Error(crashMessage),
            reason: crashMessage,
            warning: null,
            screenshots: {},
        };
    }
}
exports.default = handler;
