// slice-machine-ui/build/server/src/api/custom-types/push.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const state_1 = require("../state");
const push_1 = require("../slices/push");
const error_1 = require("../common/error");
const paths_1 = require("@lib/models/paths");
const tab_1 = require("@lib/models/common/CustomType/tab");
const index_1 = require("@slicemachine/core/build/models/CustomType/index");
const IO = __importStar(require("../io"));
const createOrUpdate = (client, smModel, remoteCustomType) => {
    const model = index_1.CustomTypes.fromSM(smModel);
    if (remoteCustomType) {
        return client.updateCustomType(model);
    }
    return client.insertCustomType(model);
};
async function handler(req) {
    const { id } = req.query;
    const state = await state_1.getBackendState(req.errors, req.env);
    if (!state.libraries) {
        const code = 400;
        const message = `Error ${code}: Slice libraries needs to be define in your sm.json file.`;
        return {
            err: new Error(message),
            reason: message,
            status: code,
        };
    }
    if (state.clientError || !state.env.isUserLoggedIn) {
        const isAnAuthenticationError = !state.env.isUserLoggedIn ||
            (state.clientError && state.clientError.status === 403);
        const errorExplanation = isAnAuthenticationError
            ? "Please log in to Prismic!"
            :
                `You don\'t have access to the repo \"${state.env.repo}\"`;
        const errorCode = !state.env.isUserLoggedIn
            ? 403
            : state.clientError
                ? state.clientError.status
                : 403;
        const message = `Error ${errorCode}: Could not fetch remote custom types. ${errorExplanation}`;
        return {
            err: new Error(message),
            reason: message,
            status: errorCode,
        };
    }
    const modelPath = paths_1.CustomTypesPaths(state.env.cwd)
        .customType(id)
        .model();
    let model;
    try {
        model = IO.CustomType.readCustomType(modelPath);
    }
    catch (e) {
        const msg = `[custom-types/push] Model ${id} is invalid.`;
        console.error(msg);
        return error_1.onError(null, msg);
    }
    const remoteCustomType = state.remoteCustomTypes.find((e) => e.id === id);
    if (remoteCustomType && remoteCustomType.repeatable !== model.repeatable) {
        const msg = `[custom-types/push] Model not pushed: property "repeatable" in local Model differs from remote source`;
        console.error(msg);
        return error_1.onError(null, msg);
    }
    const sliceKeysToPush = [];
    for (const [, tab] of Object.entries(model.tabs)) {
        const { sliceZone } = tab_1.Tab.organiseFields(tab);
        if (sliceZone === null || sliceZone === void 0 ? void 0 : sliceZone.value) {
            sliceKeysToPush.push(...new Set(sliceZone.value.map((e) => e.key)));
        }
    }
    const localSlices = state.libraries
        .filter((e) => e.isLocal)
        .reduceRight((acc, curr) => {
        return {
            ...acc,
            ...curr.components.reduce((acc, curr) => ({
                ...acc,
                [curr.model.id]: curr,
            }), {}),
        };
    }, {});
    for await (const sliceKey of sliceKeysToPush) {
        const slice = localSlices[sliceKey];
        if (slice) {
            try {
                console.log("[custom-types/push] Pushing slice", sliceKey);
                await push_1.pushSlice(state.env, state.remoteSlices, {
                    sliceName: slice.model.name,
                    from: slice.from,
                });
            }
            catch (e) {
                console.error(`[custom-types/push] Full error: ${e}`);
            }
        }
    }
    console.log("[custom-types/push] Pushing Custom Type...");
    const res = await createOrUpdate(state.env.client, model, remoteCustomType);
    if (res.status > 209) {
        const message = res.text ? await res.text() : res.status.toString();
        const msg = `[custom-types/push] Unexpected error returned. Server message: ${message}`;
        console.error(msg);
        return error_1.onError(null, msg);
    }
    console.log(`[custom-types/push] Custom Type ${id} was pushed!`);
    return {};
}
exports.default = handler;
