// slice-machine-ui/build/server/src/api/state.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getBackendState = void 0;
const libraries_1 = __importDefault(require("./libraries"));
const index_1 = __importDefault(require("./custom-types/index"));
const generate_1 = require("./common/generate");
const DefaultClient_1 = __importDefault(require("@lib/models/common/http/DefaultClient"));
const getAndSetUserProfile_1 = require("./services/getAndSetUserProfile");
const preferWroomBase_1 = __importDefault(require("../../../lib/utils/preferWroomBase"));
const prismic_1 = require("@slicemachine/core/build/prismic");
const node_utils_1 = require("@slicemachine/core/build/node-utils");
const getBackendState = async (configErrors, env) => {
    const { libraries, remoteSlices, clientError } = await libraries_1.default(env);
    const { customTypes, remoteCustomTypes } = await index_1.default(env);
    const base = preferWroomBase_1.default(env.manifest.apiEndpoint);
    if (env.isUserLoggedIn && env.prismicData.auth) {
        try {
            const newTokenResponse = await DefaultClient_1.default.refreshToken(base, env.prismicData.auth);
            if (newTokenResponse.status &&
                Math.floor(newTokenResponse.status / 100) === 2) {
                const newToken = await newTokenResponse.text();
                prismic_1.PrismicSharedConfigManager.setAuthCookie(newToken);
                if (!env.prismicData.shortId || !env.prismicData.intercomHash)
                    await getAndSetUserProfile_1.getAndSetUserProfile(env, newToken);
            }
        }
        catch (e) {
            console.error("[Refresh token]: Internal error : ", e);
        }
    }
    generate_1.generate(env, libraries);
    return {
        libraries,
        customTypes,
        remoteCustomTypes,
        remoteSlices,
        clientError,
        configErrors,
        env,
    };
};
exports.getBackendState = getBackendState;
async function handler(req) {
    const { errors: configErrors, env } = req;
    const serverState = await exports.getBackendState(configErrors, env);
    const { client, cwd, prismicData, baseUrl, ...frontEnv } = serverState.env;
    const frontEndEnv = {
        ...frontEnv,
        sliceMachineAPIUrl: baseUrl,
        packageManager: node_utils_1.Files.exists(node_utils_1.YarnLockPath(cwd)) ? "yarn" : "npm",
        shortId: prismicData.shortId,
        intercomHash: prismicData.intercomHash,
    };
    return {
        customTypes: serverState.customTypes,
        remoteCustomTypes: serverState.remoteCustomTypes,
        libraries: serverState.libraries,
        remoteSlices: serverState.remoteSlices,
        env: frontEndEnv,
    };
}
exports.default = handler;
