// slice-machine-ui/build/server/src/api/storybook.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const path_1 = __importDefault(require("path"));
const ejs_1 = __importDefault(require("ejs"));
const files_1 = __importDefault(require("@lib/utils/files"));
const paths_1 = require("@lib/models/paths");
const core_1 = require("@slicemachine/core");
const str_1 = require("@lib/utils/str");
const Paths = {
    nuxtTemplate: (appRoot) => path_1.default.join(appRoot, "templates/storybook/nuxt.template.ejs"),
    nextTemplate: (appRoot) => path_1.default.join(appRoot, "templates/storybook/next.template.ejs"),
    svelteTemplate: (appRoot) => path_1.default.join(appRoot, "templates/storybook/svelte.template.ejs"),
    getTemplate(appRoot, framework) {
        switch (framework) {
            case core_1.Models.Frameworks.nuxt:
            case core_1.Models.Frameworks.previousNuxt:
                return Paths.nuxtTemplate(appRoot);
            case core_1.Models.Frameworks.vue:
                return Paths.nuxtTemplate(appRoot);
            case core_1.Models.Frameworks.next:
            case core_1.Models.Frameworks.previousNext:
                return Paths.nextTemplate(appRoot);
            case core_1.Models.Frameworks.react:
                return Paths.nextTemplate(appRoot);
            case core_1.Models.Frameworks.svelte:
                return Paths.svelteTemplate(appRoot);
            case core_1.Models.Frameworks.vanillajs:
                return Paths.nextTemplate(appRoot);
            default:
                return null;
        }
    },
};
exports.default = {
    generateStories(appRoot, framework, cwd, libraryName, sliceName) {
        if (files_1.default.exists(paths_1.CustomPaths(cwd).library(libraryName).slice(sliceName).stories()))
            return;
        const generatedMocksPath = paths_1.GeneratedPaths(cwd)
            .library(libraryName)
            .slice(sliceName)
            .mocks();
        const customMocksPath = paths_1.CustomPaths(cwd)
            .library(libraryName)
            .slice(sliceName)
            .mocks();
        const mocks = files_1.default.readFirstOf([
            customMocksPath,
            generatedMocksPath,
        ])((value) => JSON.parse(value));
        if (!mocks) {
            console.error(`No mocks available, cannot generate stories`);
            return;
        }
        const templatePath = Paths.getTemplate(appRoot, framework);
        if (!templatePath) {
            console.error(`We don't support storybook generated stories for ${framework} yet`);
            return;
        }
        const template = files_1.default.readString(templatePath);
        const withPascalizedIds = (mocks.value || []).map((m) => {
            const id = str_1.createStorybookId(m.variation || m.name);
            return {
                ...m,
                id,
            };
        });
        const componentPath = path_1.default
            .join("..", path_1.default.relative(paths_1.GeneratedPaths(cwd).library(libraryName).value(), paths_1.CustomPaths(cwd).library(libraryName).value()), sliceName)
            .split(path_1.default.sep)
            .join(path_1.default.posix.sep);
        const componentTitle = `${libraryName}/${sliceName}`;
        const stories = ejs_1.default.render(template, {
            mocks: withPascalizedIds,
            componentPath,
            componentTitle,
        });
        files_1.default.write(paths_1.GeneratedPaths(cwd)
            .library(libraryName)
            .slice(sliceName)
            .stories(framework === core_1.Models.Frameworks.svelte
            ? "index.stories.svelte"
            : undefined), stories);
    },
};
