// slice-machine-ui/build/lib/models/common/ComponentUI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComponentUI = exports.LibStatus = exports.buildScreenshotUrls = exports.createScreenshotUI = exports.createScreenshotUrl = void 0;
const utils_1 = require("../../utils");
const createScreenshotUrl = (baseUrl, pathToScreenshot) => `${baseUrl}/api/__preview?q=${encodeURIComponent(pathToScreenshot)}&uniq=${Math.random()}`;
exports.createScreenshotUrl = createScreenshotUrl;
const createScreenshotUI = (baseUrl, pathToScreenshot) => ({
    path: pathToScreenshot,
    url: exports.createScreenshotUrl(baseUrl, pathToScreenshot),
});
exports.createScreenshotUI = createScreenshotUI;
const buildScreenshotUrls = (screenshotPaths, baseUrl) => {
    if (!screenshotPaths) {
        return {};
    }
    return Object.entries(screenshotPaths).reduce((acc, [variationId, screenshot]) => {
        return screenshot.path
            ? {
                ...acc,
                [variationId]: exports.createScreenshotUI(baseUrl, screenshot.path),
            }
            : acc;
    }, {});
};
exports.buildScreenshotUrls = buildScreenshotUrls;
var LibStatus;
(function (LibStatus) {
    LibStatus["Modified"] = "MODIFIED";
    LibStatus["Synced"] = "SYNCED";
    LibStatus["Invalid"] = "INVALID";
    LibStatus["NewSlice"] = "NEW_SLICE";
})(LibStatus = exports.LibStatus || (exports.LibStatus = {}));
exports.ComponentUI = {
    build(component, remoteSlices, env) {
        return {
            ...component,
            screenshotUrls: exports.buildScreenshotUrls(component.screenshotPaths, env.baseUrl),
            __status: computeStatus(component, remoteSlices),
        };
    },
};
function computeStatus(component, remoteSlices) {
    const slice = remoteSlices.find((s) => component.model.id === s.id);
    if (!slice)
        return LibStatus.NewSlice;
    const sameVersion = utils_1.compareVariations(component.model.variations, slice.variations);
    if (sameVersion)
        return LibStatus.Synced;
    else
        return LibStatus.Modified;
}
