// slice-machine-ui/build/lib/models/common/MockConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomTypeMockConfig = exports.SliceMockConfig = void 0;
exports.SliceMockConfig = {
    getSliceMockConfig(globalMockConfig, libName, sliceName) {
        var _a;
        return ((_a = globalMockConfig === null || globalMockConfig === void 0 ? void 0 : globalMockConfig[libName]) === null || _a === void 0 ? void 0 : _a[sliceName]) || {};
    },
    getVariationMockConfig(sliceMockConfig, variationId) {
        return sliceMockConfig[variationId] || {};
    },
    getFieldMockConfig(sliceMockConfig, variationId, widgetArea, fieldId) {
        var _a;
        return (((_a = this.getVariationMockConfig(sliceMockConfig, variationId)[widgetArea]) === null || _a === void 0 ? void 0 : _a[fieldId]) || {});
    },
    deleteFieldMockConfig(sliceMockConfig, variationId, widgetArea, fieldId) {
        return {
            ...sliceMockConfig,
            [variationId]: {
                ...this.getVariationMockConfig(sliceMockConfig, variationId),
                [widgetArea]: {
                    ...this.getVariationMockConfig(sliceMockConfig, variationId)[widgetArea],
                    [fieldId]: undefined,
                },
            },
        };
    },
    updateFieldMockConfig(sliceMockConfig, variationId, widgetArea, previousKey, fieldId, value) {
        return {
            ...sliceMockConfig,
            [variationId]: {
                ...this.getVariationMockConfig(sliceMockConfig, variationId),
                [widgetArea]: {
                    ...this.getVariationMockConfig(sliceMockConfig, variationId)[widgetArea],
                    ...(fieldId !== previousKey
                        ? {
                            [previousKey]: undefined,
                            [fieldId]: value,
                        }
                        : {
                            [fieldId]: value,
                        }),
                },
            },
        };
    },
};
exports.CustomTypeMockConfig = {
    getCustomTypeMockConfig(globalMockConfig, ctId) {
        var _a;
        return ((_a = globalMockConfig === null || globalMockConfig === void 0 ? void 0 : globalMockConfig._cts) === null || _a === void 0 ? void 0 : _a[ctId]) || {};
    },
    getFieldMockConfig(ctMockConfig, fieldId) {
        return ctMockConfig[fieldId] || {};
    },
    deleteFieldMockConfig(ctMockConfig, fieldId) {
        return {
            ...ctMockConfig,
            [fieldId]: undefined,
        };
    },
    updateFieldMockConfig(ctMockConfig, previousFieldId, fieldId, value) {
        return {
            ...ctMockConfig,
            ...(previousFieldId !== fieldId
                ? {
                    [previousFieldId]: undefined,
                    [fieldId]: value,
                }
                : {
                    [fieldId]: value,
                }),
        };
    },
    updateGroupFieldMockConfig(ctMockConfig, groupId, previousFieldId, fieldId, value) {
        return {
            ...ctMockConfig,
            [groupId]: {
                ...this.getFieldMockConfig(ctMockConfig, groupId),
                ...(previousFieldId !== fieldId
                    ? {
                        [fieldId]: value,
                        [previousFieldId]: undefined,
                    }
                    : {
                        [fieldId]: value,
                    }),
            },
        };
    },
    deleteGroupFieldMockConfig(ctMockConfig, groupId, fieldId) {
        return {
            ...ctMockConfig,
            [groupId]: {
                ...this.getFieldMockConfig(ctMockConfig, groupId),
                [fieldId]: undefined,
            },
        };
    },
};
