// slice-machine-ui/build/lib/models/common/Variation.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Variation = void 0;
("./CustomType/fields");
const camelCase_1 = __importDefault(require("lodash/camelCase"));
exports.Variation = {
    generateId(str) {
        return camelCase_1.default(str);
    },
    toObject(variation) {
        var _a, _b;
        return {
            ...variation,
            primary: (_a = variation.primary) === null || _a === void 0 ? void 0 : _a.reduce((acc, { key, value }) => ({ ...acc, [key]: value }), {}),
            items: (_b = variation.items) === null || _b === void 0 ? void 0 : _b.reduce((acc, { key, value }) => ({ ...acc, [key]: value }), {}),
        };
    },
    toArray(variation) {
        return {
            ...variation,
            primary: Object.entries(variation.primary || {}).map(([key, value]) => ({
                key,
                value,
            })),
            items: Object.entries(variation.items || {}).map(([key, value]) => ({
                key,
                value,
            })),
        };
    },
    reorderWidget(variation, widgetsArea, start, end) {
        const widgets = variation[widgetsArea] || [];
        const reorderedWidget = widgets && widgets[start];
        if (!reorderedWidget)
            throw new Error(`Unable to reorder the widget at index ${start}. the list of widgets contains only ${widgets.length} elements.`);
        const reorderedArea = widgets.reduce((acc, widget, index) => {
            const elems = [widget, reorderedWidget];
            switch (index) {
                case start:
                    return acc;
                case end:
                    return [...acc, ...(end > start ? elems : elems.reverse())];
                default:
                    return [...acc, widget];
            }
        }, []);
        return {
            ...variation,
            [widgetsArea]: reorderedArea,
        };
    },
    replaceWidget(variation, widgetsArea, previousKey, newKey, newValue) {
        const widgets = variation[widgetsArea] || [];
        return {
            ...variation,
            [widgetsArea]: widgets.reduce((acc, { key, value }) => {
                if (key === previousKey) {
                    return acc.concat([{ key: newKey, value: newValue }]);
                }
                else {
                    return acc.concat([{ key, value }]);
                }
            }, []),
        };
    },
    addWidget(variation, widgetsArea, key, value) {
        var _a;
        return {
            ...variation,
            [widgetsArea]: (_a = variation[widgetsArea]) === null || _a === void 0 ? void 0 : _a.concat([{ key, value }]),
        };
    },
    deleteWidget(variation, widgetsArea, widgetKey) {
        var _a;
        return {
            ...variation,
            [widgetsArea]: (_a = variation[widgetsArea]) === null || _a === void 0 ? void 0 : _a.filter(({ key }) => widgetKey !== key),
        };
    },
    copyValue(variation, key, name) {
        return {
            ...variation,
            id: key,
            name,
        };
    },
};
