// slice-machine-ui/build/lib/models/common/CustomType/group.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Group = void 0;
exports.Group = {
    addWidget(group, newField) {
        return {
            ...group,
            config: {
                ...group.config,
                fields: [...group.config.fields, newField],
            },
        };
    },
    deleteWidget(group, wkey) {
        return {
            ...group,
            config: {
                ...group.config,
                fields: group.config.fields.filter(({ key }) => key !== wkey),
            },
        };
    },
    reorderWidget(group, start, end) {
        const reorderedWidget = group.config.fields[start];
        if (!reorderedWidget)
            throw new Error(`Unable to reorder the widget at index ${start}.`);
        const reorderedArea = group.config.fields.reduce((acc, field, index) => {
            const elems = [field, reorderedWidget];
            switch (index) {
                case start:
                    return acc;
                case end:
                    return [...acc, ...(end > start ? elems : elems.reverse())];
                default:
                    return [...acc, field];
            }
        }, []);
        return {
            ...group,
            config: {
                ...group.config,
                fields: reorderedArea,
            },
        };
    },
    replaceWidget(group, previousKey, newKey, value) {
        return {
            ...group,
            config: {
                ...group.config,
                fields: group.config.fields.map((t) => {
                    if (t.key === previousKey) {
                        return {
                            key: newKey,
                            value,
                        };
                    }
                    return t;
                }),
            },
        };
    },
    toArray(group) {
        return {
            ...group,
            config: {
                ...group.config,
                fields: Object.entries(group.config.fields).map(([key, value]) => ({
                    key,
                    value,
                })),
            },
        };
    },
    toObject(group) {
        return {
            ...group,
            config: {
                ...group.config,
                fields: group.config.fields.reduce((acc, { key, value }) => ({ ...acc, [key]: value }), {}),
            },
        };
    },
};
