// slice-machine-ui/build/lib/models/common/CustomType/sliceZone.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SliceZone = void 0;
const slices_1 = require("@prismicio/types-internal/lib/customtypes/widgets/slices");
exports.SliceZone = {
    addSharedSlice(sz, key) {
        const value = sz.value.concat([
            {
                key,
                value: {
                    type: slices_1.SlicesTypes.SharedSlice,
                },
            },
        ]);
        return {
            ...sz,
            value,
        };
    },
    replaceSharedSlice(sz, keys, preserve = []) {
        const value = sz.value
            .filter(({ key }) => preserve.includes(key))
            .concat(keys.map((key) => ({
            key,
            value: { type: slices_1.SlicesTypes.SharedSlice },
        })));
        return {
            ...sz,
            value,
        };
    },
    removeSharedSlice(sz, key) {
        const value = sz.value.filter(({ key: k }) => k === key);
        return {
            ...sz,
            value,
        };
    },
    createEmpty(key) {
        return { key, value: [] };
    },
};
