// slice-machine-ui/build/lib/models/common/CustomType/tab.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tab = void 0;
const widgets_1 = require("@prismicio/types-internal/lib/customtypes/widgets");
const sliceZone_1 = require("@lib/models/common/CustomType/sliceZone");
exports.Tab = {
    init(id) {
        return { key: id, value: [] };
    },
    updateSliceZone(tab) {
        return (mutate) => {
            return {
                ...tab,
                sliceZone: tab.sliceZone && mutate(tab.sliceZone),
            };
        };
    },
    updateGroup(tab, groupId) {
        return (mutate) => {
            return {
                ...tab,
                value: tab.value.map((field) => {
                    if (field.key === groupId && field.value.type === widgets_1.WidgetTypes.Group) {
                        return {
                            key: groupId,
                            value: mutate(field.value),
                        };
                    }
                    return field;
                }),
            };
        };
    },
    addWidget(tab, id, widget) {
        const elem = { key: id, value: widget };
        return {
            ...tab,
            value: [...tab.value, elem],
        };
    },
    replaceWidget(tab, previousKey, newKey, value) {
        return {
            ...tab,
            value: tab.value.map((t) => {
                if (t.key === previousKey) {
                    return {
                        key: newKey,
                        value,
                    };
                }
                return t;
            }),
        };
    },
    reorderWidget(tab, start, end) {
        const reorderedWidget = tab.value[start];
        if (!reorderedWidget)
            throw new Error(`Unable to reorder the widget at index ${start}.`);
        const reorderedArea = tab.value.reduce((acc, widget, index) => {
            const elems = [widget, reorderedWidget];
            switch (index) {
                case start:
                    return acc;
                case end:
                    return [...acc, ...(end > start ? elems : elems.reverse())];
                default:
                    return [...acc, widget];
            }
        }, []);
        return {
            ...tab,
            value: reorderedArea,
        };
    },
    removeWidget(tab, id) {
        const newTab = {
            ...tab,
            value: tab.value.filter((e) => e.key !== id),
        };
        return newTab;
    },
    createSliceZone(tab, key) {
        return {
            ...tab,
            sliceZone: sliceZone_1.SliceZone.createEmpty(key),
        };
    },
    organiseFields(tabSM) {
        const { fields, groups } = tabSM.value.reduce((acc, current) => {
            switch (current.value.type) {
                case widgets_1.WidgetTypes.UID:
                    return acc;
                case widgets_1.WidgetTypes.Group:
                    return {
                        ...acc,
                        groups: [
                            ...acc.groups,
                            { key: current.key, value: current.value },
                        ],
                    };
                    return acc;
                default:
                    return {
                        ...acc,
                        fields: [
                            ...acc.fields,
                            current,
                        ],
                    };
            }
        }, { fields: [], groups: [] });
        return {
            fields,
            groups,
            sliceZone: tabSM.sliceZone,
        };
    },
};
