// slice-machine-ui/build/server/src/api/common/LibrariesState.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.formatComponent = exports.formatLibrary = exports.formatLibraries = exports.generateState = void 0;
const probe_image_size_1 = __importDefault(require("probe-image-size"));
const Slice_1 = require("@slicemachine/core/build/models/Slice");
const libraries_1 = require("@slicemachine/core/build/libraries");
const node_utils_1 = require("@slicemachine/core/build/node-utils");
const DEFAULT_IMAGE_DIMENSIONS = {
    width: undefined,
    height: undefined,
};
function generateState(env) {
    const libraries = (env.manifest.libraries || [])
        .map((lib) => libraries_1.handleLibraryPath(env.cwd, lib))
        .filter(Boolean);
    const state = formatLibraries(libraries);
    node_utils_1.Files.write(node_utils_1.LibrariesStatePath(env.cwd), state);
}
exports.generateState = generateState;
function formatLibraries(libraries) {
    return libraries.reduce((acc, library) => {
        return { ...acc, [library.name]: formatLibrary(library) };
    }, {});
}
exports.formatLibraries = formatLibraries;
function formatLibrary(library) {
    var _a, _b;
    const components = library.components.reduce((acc, component) => ({
        ...acc,
        [component.model.id]: formatComponent(component),
    }), {});
    return {
        name: (_a = library.meta) === null || _a === void 0 ? void 0 : _a.name,
        version: (_b = library.meta) === null || _b === void 0 ? void 0 : _b.version,
        components,
    };
}
exports.formatLibrary = formatLibrary;
function getImageDimensions(imagePath) {
    if (!imagePath || !node_utils_1.Files.exists(imagePath))
        return DEFAULT_IMAGE_DIMENSIONS;
    const imageBuffer = node_utils_1.Files.readBuffer(imagePath);
    const result = probe_image_size_1.default.sync(imageBuffer);
    if (!result)
        return DEFAULT_IMAGE_DIMENSIONS;
    return { width: result.width, height: result.height };
}
function formatComponent(slice) {
    return {
        library: slice.from,
        id: slice.model.id,
        name: slice.model.name,
        description: slice.model.description,
        model: Slice_1.Slices.fromSM(slice.model),
        mocks: (slice.mock || []).reduce((acc, variationMock) => ({
            ...acc,
            [variationMock.variation]: variationMock,
        }), {}),
        meta: {
            fileName: slice.fileName,
            extension: slice.extension,
        },
        screenshotPaths: !slice.screenshotPaths
            ? {}
            : Object.entries(slice.screenshotPaths).reduce((acc, [variationId, screenshot]) => {
                return {
                    ...acc,
                    [variationId]: {
                        path: screenshot.path,
                        ...getImageDimensions(screenshot.path),
                    },
                };
            }, {}),
    };
}
exports.formatComponent = formatComponent;
