// slice-machine-ui/build/scripts/migration/versions/0.0.41/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const fs_1 = __importDefault(require("fs"));
const path_1 = __importDefault(require("path"));
const glob_1 = __importDefault(require("glob"));
const slash_1 = __importDefault(require("slash"));
const NodeUtils = __importStar(require("@slicemachine/core/build/node-utils"));
const Libraries = __importStar(require("@slicemachine/core/build/libraries"));
const prompts_1 = __importDefault(require("prompts"));
const migration = {
    version: "0.0.41",
    main: async function main({ cwd, ignorePromptForTest }) {
        console.info("\nSliceMachine now supports both default and custom previews (screenshots)!");
        console.info("Default screenshots are now stored in a special .slicemachine folder.");
        if (!ignorePromptForTest) {
            const doTheMigration = await shouldIRun("Would you like me to move current previews to .slicemachine folder?");
            if (!doTheMigration.yes)
                return;
        }
        const manifest = NodeUtils.retrieveManifest(cwd);
        if (!manifest.exists || !manifest.content)
            return;
        try {
            fs_1.default.mkdirSync(path_1.default.join(cwd, ".slicemachine", "assets"), {
                recursive: true,
            });
            const { libraries } = manifest.content;
            if (!libraries)
                return;
            libraries.forEach((lib) => {
                const { isLocal, pathExists, pathToSlices } = Libraries.getInfoFromPath(cwd, lib);
                if (!(isLocal && pathExists))
                    return;
                const matches = glob_1.default.sync(`${slash_1.default(pathToSlices)}/**/preview.png`);
                matches.forEach((match) => {
                    const split = match.split(path_1.default.posix.sep);
                    const sliceName = split[split.length - 2];
                    if (sliceName) {
                        const pathToNewFile = path_1.default.join(cwd, ".slicemachine/assets", split[split.length - 3], sliceName, "preview.png");
                        fs_1.default.mkdirSync(path_1.default.dirname(pathToNewFile), {
                            recursive: true,
                        });
                        fs_1.default.renameSync(match, pathToNewFile);
                    }
                });
            });
        }
        catch (e) {
            console.log(e);
        }
    },
};
function shouldIRun(message) {
    return prompts_1.default({
        type: "select",
        name: "yes",
        message,
        choices: [
            { title: "Yes", value: true },
            { title: "No (skip)", value: false },
        ],
        initial: 0,
    });
}
exports.default = migration;
