// slice-machine-ui/build/lib/models/common/widgets/UID/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UIDWidget = void 0;
const yup = __importStar(require("yup"));
const md_1 = require("react-icons/md");
const utils_1 = require("../../../../utils");
const defaults_1 = require("../../../../forms/defaults");
const forms_1 = require("../../../../forms");
const type_1 = require("./type");
const fields_1 = require("../../CustomType/fields");
const FormFields = {
    label: defaults_1.DefaultFields.label,
    id: {
        ...defaults_1.DefaultFields.id,
        validate: {
            required: "This field is required",
            matches: [/^uid$/, "Api ID must be 'uid' for this field."],
        },
    },
    placeholder: defaults_1.DefaultFields.placeholder,
};
const schema = yup.object().shape({
    type: yup.string().matches(/^UID$/, { excludeEmptyString: true }).required(),
    config: forms_1.createValidationSchema(utils_1.removeProp(FormFields, "id")),
});
const Meta = {
    icon: md_1.MdVpnKey,
    title: "UID",
    description: "Unique Identifier",
};
exports.UIDWidget = {
    create: (label) => new type_1.UIDField({ label }),
    Meta,
    schema,
    TYPE_NAME: fields_1.FieldType.UID,
    FormFields,
};
