// slice-machine-ui/build/lib/models/paths.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MocksConfig = exports.YarnLock = exports.Pkg = exports.SliceTemplateConfig = exports.SMConfig = exports.PrismicConfig = exports.LibrariesStatePath = exports.PackagePaths = exports.CustomPaths = exports.CustomTypesPaths = exports.GeneratedCustomTypesPaths = exports.GeneratedPaths = exports.paths = void 0;
const path_1 = __importDefault(require("path"));
const paths = (cwd, prefix) => ({
    value: () => path_1.default.join(cwd, prefix),
    customType: (id) => ({
        value: () => path_1.default.join(exports.paths(cwd, prefix).value()),
        model: () => path_1.default.join(exports.paths(cwd, prefix).value(), id, "index.json"),
        mock: () => path_1.default.join(exports.paths(cwd, prefix).value(), id, "mocks.json"),
    }),
    library: (libraryName) => ({
        value: () => path_1.default.join(exports.paths(cwd, prefix).value(), libraryName),
        slice: (sliceName) => ({
            value: () => path_1.default.join(exports.paths(cwd, prefix).library(libraryName).value(), sliceName),
            preview: (filename = "preview.png") => path_1.default.join(exports.paths(cwd, prefix).library(libraryName).slice(sliceName).value(), filename),
            stories: (filename = "index.stories.js") => path_1.default.join(exports.paths(cwd, prefix).library(libraryName).slice(sliceName).value(), filename),
            mocks: () => path_1.default.join(exports.paths(cwd, prefix).library(libraryName).slice(sliceName).value(), "mocks.json"),
            model: () => path_1.default.join(exports.paths(cwd, prefix).library(libraryName).slice(sliceName).value(), "model.json"),
            variation: (variationId) => ({
                value: () => path_1.default.join(exports.paths(cwd, prefix).library(libraryName).slice(sliceName).value(), variationId),
                preview: (filename = "preview.png") => path_1.default.join(exports.paths(cwd, prefix)
                    .library(libraryName)
                    .slice(sliceName)
                    .variation(variationId)
                    .value(), filename),
            }),
        }),
    }),
});
exports.paths = paths;
const GeneratedPaths = (cwd) => exports.paths(cwd, path_1.default.join(".slicemachine", "assets"));
exports.GeneratedPaths = GeneratedPaths;
const GeneratedCustomTypesPaths = (cwd) => exports.paths(cwd, path_1.default.join(".slicemachine", "assets", "customtypes"));
exports.GeneratedCustomTypesPaths = GeneratedCustomTypesPaths;
const CustomTypesPaths = (cwd) => exports.paths(cwd, "customtypes");
exports.CustomTypesPaths = CustomTypesPaths;
const CustomPaths = (cwd) => exports.paths(cwd, "");
exports.CustomPaths = CustomPaths;
const PackagePaths = (cwd) => exports.paths(cwd, "node_modules");
exports.PackagePaths = PackagePaths;
const LibrariesStatePath = (cwd) => path_1.default.join(cwd, ".slicemachine", "libraries-state.json");
exports.LibrariesStatePath = LibrariesStatePath;
const PrismicConfig = (cwd) => path_1.default.join(cwd, ".prismic");
exports.PrismicConfig = PrismicConfig;
const SMConfig = (cwd) => path_1.default.join(cwd, "sm.json");
exports.SMConfig = SMConfig;
const SliceTemplateConfig = (cwd, customPathToTemplate) => customPathToTemplate
    ? path_1.default.join(cwd, customPathToTemplate)
    : path_1.default.join(exports.PrismicConfig(cwd), "slice-template");
exports.SliceTemplateConfig = SliceTemplateConfig;
const Pkg = (cwd) => path_1.default.join(cwd, "package.json");
exports.Pkg = Pkg;
const YarnLock = (cwd) => path_1.default.join(cwd, "yarn.lock");
exports.YarnLock = YarnLock;
const MocksConfig = (cwd) => path_1.default.join(cwd, ".slicemachine", "mock-config.json");
exports.MocksConfig = MocksConfig;
