// slice-machine-ui/build/lib/builders/common/Zone/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const prop_types_1 = require("prop-types");
const Card_1 = __importDefault(require("./Card"));
const theme_ui_1 = require("theme-ui");
const fa_1 = require("react-icons/fa");
const SelectFieldTypeModal_1 = __importDefault(require("../SelectFieldTypeModal"));
const NewField_1 = __importDefault(require("./Card/components/NewField"));
const ZoneHeader_1 = __importDefault(require("./components/ZoneHeader"));
const EmptyState_1 = __importDefault(require("./components/EmptyState"));
const Zone = ({ Model, store, tabId, title, fields, poolOfFieldsToCheck, widgetsArray, isRepeatable, onDeleteItem, onSaveNewField, onDragEnd, EditModal, onSave, dataTip, getFieldMockConfig, renderHintBase, renderFieldAccessor, }) => {
    const widgetsArrayWithCondUid = (() => {
        const hasUid = !!Object.entries(poolOfFieldsToCheck).find(([, { value }]) => value.type === "UID");
        return hasUid
            ?
                widgetsArray.filter(({ TYPE_NAME }) => TYPE_NAME !== "UID")
            : widgetsArray;
    })();
    const [showHints, setShowHints] = react_1.useState(false);
    const [editModalData, setEditModalData] = react_1.useState({ isOpen: false });
    const [selectModalData, setSelectModalData] = react_1.useState({ isOpen: false });
    const [newFieldData, setNewFieldData] = react_1.useState(null);
    const enterEditMode = (field) => {
        setEditModalData({ isOpen: true, field });
    };
    const enterSelectMode = () => {
        setSelectModalData({ isOpen: true });
    };
    const closeEditModal = () => {
        setEditModalData({ isOpen: false });
    };
    const closeSelectModal = () => setSelectModalData({ isOpen: false });
    const onSelectFieldType = (widgetTypeName) => {
        setNewFieldData({ widgetTypeName, fields });
        setSelectModalData({ isOpen: false });
    };
    const onCancelNewField = () => setNewFieldData(null);
    return (<react_1.Fragment>
      <ZoneHeader_1.default Heading={<theme_ui_1.Heading as="h6">{title}</theme_ui_1.Heading>} Actions={fields.length ? (<react_1.Fragment>
              <theme_ui_1.Button variant="buttons.lightSmall" onClick={() => setShowHints(!showHints)}>
                <fa_1.FaCode style={{
                marginRight: "8px",
                position: "relative",
                top: "2px",
            }}/>{" "}
                {showHints ? "Hide" : "Show"} code snippets
              </theme_ui_1.Button>
              <theme_ui_1.Button ml={2} variant="buttons.darkSmall" onClick={() => enterSelectMode()}>
                <fa_1.FaPlus style={{
                marginRight: "8px",
                position: "relative",
                top: "2px",
            }}/>
                Add a new Field
              </theme_ui_1.Button>
            </react_1.Fragment>) : null}/>
      {!fields.length && !newFieldData && (<EmptyState_1.default onEnterSelectMode={() => enterSelectMode()} zoneName={isRepeatable ? "repeatable" : "static"}/>)}
      <Card_1.default tabId={tabId} isRepeatable={isRepeatable} fields={fields} showHints={showHints} store={store} Model={Model} dataTip={dataTip} title={title} renderFieldAccessor={renderFieldAccessor} renderHintBase={renderHintBase} enterEditMode={enterEditMode} enterSelectMode={enterSelectMode} onDragEnd={onDragEnd} onDeleteItem={onDeleteItem} NewFieldC={() => newFieldData && (<NewField_1.default {...newFieldData} fields={poolOfFieldsToCheck || fields} onSave={(...args) => {
                onSaveNewField(...args);
                setNewFieldData(null);
            }} onCancelNewField={onCancelNewField}/>)}/>
      <EditModal Model={Model} data={editModalData} close={closeEditModal} onSave={onSave} fields={poolOfFieldsToCheck} getFieldMockConfig={getFieldMockConfig}/>
      <SelectFieldTypeModal_1.default data={selectModalData} close={closeSelectModal} onSelect={onSelectFieldType} widgetsArray={widgetsArrayWithCondUid}/>
    </react_1.Fragment>);
};
Zone.propTypes = {
    isRepeatable: prop_types_1.bool,
    Model: prop_types_1.object.isRequired,
    title: prop_types_1.string.isRequired,
    dataTip: prop_types_1.string.isRequired,
    onSave: prop_types_1.func.isRequired,
    onSaveNewField: prop_types_1.func.isRequired,
    onDragEnd: prop_types_1.func.isRequired,
    onDeleteItem: prop_types_1.func.isRequired,
    poolOfFieldsToCheck: prop_types_1.arrayOf(prop_types_1.shape({ key: prop_types_1.string, value: prop_types_1.object })),
    getFieldMockConfig: prop_types_1.func.isRequired,
    renderHintBase: prop_types_1.func.isRequired,
    renderFieldAccessor: prop_types_1.func.isRequired,
    widgetsArray: prop_types_1.array.isRequired,
    fields: prop_types_1.arrayOf(prop_types_1.shape({
        key: prop_types_1.string.isRequired,
        value: prop_types_1.shape({
            config: prop_types_1.object,
            fields: prop_types_1.array,
            type: prop_types_1.string.isRequired,
        }),
    })),
};
exports.default = Zone;
