// @slicemachine/core/build/src/filesystem/authConfig.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateAuthCookie = exports.setAuthConfig = exports.removeAuthConfig = exports.getOrCreateAuthConfig = exports.createDefaultAuthConfig = exports.DEFAULT_CONFIG = exports.AuthConfig = void 0;
const Either_1 = require("fp-ts/lib/Either");
const t = __importStar(require("io-ts"));
const utils_1 = require("../utils");
const paths_1 = require("./paths");
exports.AuthConfig = t.intersection([
    t.type({
        base: t.string,
        cookies: t.string,
    }),
    t.partial({
        oauthAccessToken: t.string,
        authUrl: t.string,
    }),
]);
exports.DEFAULT_CONFIG = {
    base: "https://prismic.io",
    cookies: "",
};
function createDefaultAuthConfig(directory) {
    const configPath = (0, paths_1.PrismicConfigPath)(directory);
    utils_1.Files.write(configPath, JSON.stringify(exports.DEFAULT_CONFIG, null, "\t"), {
        recursive: false,
    });
    return exports.DEFAULT_CONFIG;
}
exports.createDefaultAuthConfig = createDefaultAuthConfig;
function getOrCreateAuthConfig(directory) {
    const configPath = (0, paths_1.PrismicConfigPath)(directory);
    if (!utils_1.Files.exists(configPath))
        return createDefaultAuthConfig(directory);
    const conf = utils_1.Files.safeReadEntity(configPath, (payload) => {
        return (0, Either_1.getOrElseW)(() => null)(exports.AuthConfig.decode(payload));
    });
    return { ...exports.DEFAULT_CONFIG, ...conf };
}
exports.getOrCreateAuthConfig = getOrCreateAuthConfig;
function removeAuthConfig(directory) {
    const configPath = (0, paths_1.PrismicConfigPath)(directory);
    return utils_1.Files.remove(configPath);
}
exports.removeAuthConfig = removeAuthConfig;
function setAuthConfig(cookies = [], base, directory) {
    const currentConfig = getOrCreateAuthConfig(directory);
    const formattedCookies = utils_1.Cookie.serializeCookies(cookies);
    const configPath = (0, paths_1.PrismicConfigPath)(directory);
    const newConfig = {
        cookies: formattedCookies,
        base: base || currentConfig.base,
    };
    return utils_1.Files.write(configPath, { ...currentConfig, ...newConfig }, { recursive: false });
}
exports.setAuthConfig = setAuthConfig;
function updateAuthCookie(authCookie, directory) {
    const currentConfig = getOrCreateAuthConfig(directory);
    const currentCookieIndexByCookieName = utils_1.Cookie.parse(currentConfig.cookies);
    const newCookies = {
        ...currentCookieIndexByCookieName,
        [utils_1.Cookie.AUTH_KEY]: authCookie,
    };
    const cookiesSerialized = Object.entries(newCookies).map(([cookieName, cookieValue]) => {
        return utils_1.Cookie.serializeCookie(cookieName, cookieValue);
    });
    return setAuthConfig(cookiesSerialized);
}
exports.updateAuthCookie = updateAuthCookie;
