// @slicemachine/core/build/src/filesystem/manifest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateManifestSMVersion = exports.patchManifest = exports.maybeRepoNameFromSMFile = exports.retrieveManifest = exports.createManifest = void 0;
const utils_1 = require("../utils");
const paths_1 = require("./paths");
function createManifest(cwd, manifest) {
    const manifestPath = (0, paths_1.SMConfigPath)(cwd);
    utils_1.Files.write(manifestPath, JSON.stringify(manifest, null, "\t"), {
        recursive: false,
    });
}
exports.createManifest = createManifest;
function retrieveManifest(cwd) {
    const manifestPath = (0, paths_1.SMConfigPath)(cwd);
    if (!utils_1.Files.exists(manifestPath)) {
        return {
            exists: false,
            content: null,
        };
    }
    const content = utils_1.Files.safeReadJson(manifestPath);
    return {
        exists: true,
        content,
    };
}
exports.retrieveManifest = retrieveManifest;
function maybeRepoNameFromSMFile(cwd, base) {
    var _a;
    try {
        const baseUrl = new URL(base);
        const maybeSMFile = retrieveManifest(cwd);
        if (maybeSMFile.exists === false)
            return null;
        if (!((_a = maybeSMFile.content) === null || _a === void 0 ? void 0 : _a.apiEndpoint))
            return null;
        const repoUrl = new URL(maybeSMFile.content.apiEndpoint);
        const correctBase = repoUrl.hostname.includes(baseUrl.hostname);
        if (correctBase === false)
            return null;
        return repoUrl.hostname.split(".")[0];
    }
    catch {
        return null;
    }
}
exports.maybeRepoNameFromSMFile = maybeRepoNameFromSMFile;
function patchManifest(cwd, data) {
    const manifest = retrieveManifest(cwd);
    if (!manifest.exists || !manifest.content)
        return false;
    const updatedManifest = {
        ...manifest.content,
        ...data,
    };
    utils_1.Files.write((0, paths_1.SMConfigPath)(cwd), updatedManifest);
    return true;
}
exports.patchManifest = patchManifest;
function updateManifestSMVersion(cwd, version) {
    var _a;
    const manifest = retrieveManifest(cwd);
    if ((_a = manifest.content) === null || _a === void 0 ? void 0 : _a._latest)
        return false;
    return patchManifest(cwd, { _latest: version });
}
exports.updateManifestSMVersion = updateManifestSMVersion;
