// @slicemachine/core/build/src/filesystem/paths.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MocksConfigPath = exports.YarnLockPath = exports.JsonPackagePath = exports.SliceTemplateConfigPath = exports.PrismicConfigPath = exports.LibrariesStatePath = exports.SMConfigPath = exports.PackagePaths = exports.CustomPaths = exports.CustomTypesPaths = exports.GeneratedCustomTypesPaths = exports.GeneratedPaths = void 0;
const path_1 = __importDefault(require("path"));
const os = __importStar(require("os"));
const Paths = (cwd, prefix) => ({
    value: () => path_1.default.join(cwd, prefix),
    customType: (id) => ({
        value: () => path_1.default.join(Paths(cwd, prefix).value()),
        model: () => path_1.default.join(Paths(cwd, prefix).value(), id, "index.json"),
        mock: () => path_1.default.join(Paths(cwd, prefix).value(), id, "mocks.json"),
    }),
    library: (libraryName) => ({
        value: () => path_1.default.join(Paths(cwd, prefix).value(), libraryName),
        slice: (sliceName) => ({
            value: () => path_1.default.join(Paths(cwd, prefix).library(libraryName).value(), sliceName),
            preview: (filename = "preview.png") => path_1.default.join(Paths(cwd, prefix).library(libraryName).slice(sliceName).value(), filename),
            stories: (filename = "index.stories.js") => path_1.default.join(Paths(cwd, prefix).library(libraryName).slice(sliceName).value(), filename),
            mocks: () => path_1.default.join(Paths(cwd, prefix).library(libraryName).slice(sliceName).value(), "mocks.json"),
            model: () => path_1.default.join(Paths(cwd, prefix).library(libraryName).slice(sliceName).value(), "model.json"),
            variation: (variationId) => ({
                value: () => path_1.default.join(Paths(cwd, prefix).library(libraryName).slice(sliceName).value(), variationId),
                preview: (filename = "preview.png") => path_1.default.join(Paths(cwd, prefix)
                    .library(libraryName)
                    .slice(sliceName)
                    .variation(variationId)
                    .value(), filename),
            }),
        }),
    }),
});
const GeneratedPaths = (cwd) => Paths(cwd, path_1.default.join(".slicemachine", "assets"));
exports.GeneratedPaths = GeneratedPaths;
const GeneratedCustomTypesPaths = (cwd) => Paths(cwd, path_1.default.join(".slicemachine", "assets", "customtypes"));
exports.GeneratedCustomTypesPaths = GeneratedCustomTypesPaths;
const CustomTypesPaths = (cwd) => Paths(cwd, "customtypes");
exports.CustomTypesPaths = CustomTypesPaths;
const CustomPaths = (cwd) => Paths(cwd, "");
exports.CustomPaths = CustomPaths;
const PackagePaths = (cwd) => Paths(cwd, "node_modules");
exports.PackagePaths = PackagePaths;
const SMConfigPath = (cwd) => path_1.default.join(cwd, "sm.json");
exports.SMConfigPath = SMConfigPath;
const LibrariesStatePath = (cwd) => path_1.default.join(cwd, ".slicemachine", "libraries-state.json");
exports.LibrariesStatePath = LibrariesStatePath;
exports.PrismicConfigPath = path_1.default.join(os.homedir(), ".prismic");
const SliceTemplateConfigPath = (cwd, customPathToTemplate) => customPathToTemplate
    ? path_1.default.join(cwd, customPathToTemplate)
    : path_1.default.join(cwd, "slice-template");
exports.SliceTemplateConfigPath = SliceTemplateConfigPath;
const JsonPackagePath = (cwd) => path_1.default.join(cwd, "package.json");
exports.JsonPackagePath = JsonPackagePath;
const YarnLockPath = (cwd) => path_1.default.join(cwd, "yarn.lock");
exports.YarnLockPath = YarnLockPath;
const MocksConfigPath = (cwd) => path_1.default.join(cwd, ".slicemachine", "mock-config.json");
exports.MocksConfigPath = MocksConfigPath;
