// @slicemachine/core/build/src/filesystem/pkg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addJsonPackageSmScript = exports.patchJsonPackage = exports.retrieveJsonPackage = void 0;
const paths_1 = require("./paths");
const utils_1 = require("../utils");
function retrieveJsonPackage(cwd) {
    const pkgPath = (0, paths_1.JsonPackagePath)(cwd);
    if (!utils_1.Files.exists(pkgPath)) {
        return {
            exists: false,
            content: null,
        };
    }
    const content = utils_1.Files.safeReadJson(pkgPath);
    return {
        exists: true,
        content,
    };
}
exports.retrieveJsonPackage = retrieveJsonPackage;
function patchJsonPackage(cwd, data) {
    const pkg = retrieveJsonPackage(cwd);
    if (!pkg.exists || !pkg.content)
        return false;
    const updatedPkg = {
        ...pkg.content,
        ...data,
    };
    utils_1.Files.write((0, paths_1.JsonPackagePath)(cwd), updatedPkg, { recursive: false });
    return true;
}
exports.patchJsonPackage = patchJsonPackage;
function addJsonPackageSmScript(cwd) {
    const pkg = retrieveJsonPackage(cwd);
    if (!pkg.exists || !pkg.content)
        return false;
    const { scripts = {} } = pkg.content;
    if (scripts[utils_1.CONSTS.SCRIPT_NAME])
        return false;
    return patchJsonPackage(cwd, {
        scripts: { ...scripts, [utils_1.CONSTS.SCRIPT_NAME]: utils_1.CONSTS.SCRIPT_VALUE },
    });
}
exports.addJsonPackageSmScript = addJsonPackageSmScript;
