// @slicemachine/core/build/src/filesystem/PrismicSharedConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrismicSharedConfigManager = exports.DEFAULT_CONFIG = void 0;
const Either_1 = require("fp-ts/lib/Either");
const utils_1 = require("../utils");
const paths_1 = require("./paths");
const PrismicSharedConfig_1 = require("../models/PrismicSharedConfig");
exports.DEFAULT_CONFIG = {
    base: "https://prismic.io",
    cookies: "",
};
exports.PrismicSharedConfigManager = {
    default() {
        utils_1.Files.write(paths_1.PrismicConfigPath, JSON.stringify(exports.DEFAULT_CONFIG, null, "\t"), {
            recursive: false,
        });
        return exports.DEFAULT_CONFIG;
    },
    get: () => {
        if (!utils_1.Files.exists(paths_1.PrismicConfigPath))
            return exports.PrismicSharedConfigManager.default();
        const conf = utils_1.Files.safeReadEntity(paths_1.PrismicConfigPath, (payload) => {
            return (0, Either_1.getOrElseW)(() => null)(PrismicSharedConfig_1.PrismicSharedConfig.decode(payload));
        });
        return { ...exports.DEFAULT_CONFIG, ...conf };
    },
    getAuth() {
        const config = exports.PrismicSharedConfigManager.get();
        return utils_1.Cookie.parsePrismicAuthToken(config.cookies);
    },
    set(config) {
        utils_1.Files.write(paths_1.PrismicConfigPath, config, { recursive: false });
    },
    setProperties(props, baseConfig) {
        const config = baseConfig || exports.PrismicSharedConfigManager.get();
        const updated = { ...config, ...props };
        exports.PrismicSharedConfigManager.set(updated);
    },
    setCookie(cookie) {
        const config = exports.PrismicSharedConfigManager.get();
        const cookiesMap = utils_1.Cookie.parse(config.cookies);
        const updatedCookiesMap = { ...cookiesMap, ...cookie };
        const serializedCookies = utils_1.Cookie.serializeCookies(Object.entries(updatedCookiesMap).map(([k, v]) => utils_1.Cookie.serializeCookie(k, v)));
        exports.PrismicSharedConfigManager.setProperties({ cookies: serializedCookies }, config);
    },
    setAuthCookie(authToken) {
        exports.PrismicSharedConfigManager.setCookie({ [utils_1.Cookie.AUTH_KEY]: authToken });
    },
    remove() {
        return utils_1.Files.remove(paths_1.PrismicConfigPath);
    },
    reset() {
        return exports.PrismicSharedConfigManager.set(exports.DEFAULT_CONFIG);
    },
};
