// @slicemachine/core/build/libraries/component.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getComponentInfo = void 0;
const path_1 = __importDefault(require("path"));
const Either_1 = require("fp-ts/lib/Either");
const screenshot_1 = require("./screenshot");
const files_1 = __importDefault(require("../node-utils/files"));
const mocks_1 = require("./mocks");
const Slice_1 = require("../models/Slice");
const errors_1 = __importDefault(require("../utils/errors"));
const slices_1 = require("@prismicio/types-internal/lib/customtypes/widgets/slices");
function getComponentName(slicePath) {
    const split = slicePath.split(path_1.default.sep);
    const pop = split.pop();
    if (!pop)
        return;
    if (pop.indexOf("index.") === 0) {
        return split.pop();
    }
    if (pop.indexOf(split[split.length - 1]) === 0) {
        return slicePath.split(path_1.default.sep).pop();
    }
    return pop.split(".")[0];
}
function findComponentFile(files, componentName) {
    const possiblePaths = ["index", componentName].reduce((acc, f) => [
        ...acc,
        `${f}.vue`,
        `${f}.js`,
        `${f}.jsx`,
        `${f}.ts`,
        `${f}.tsx`,
        `${f}.svelte`,
    ], []);
    return files.find((e) => possiblePaths.indexOf(e) > -1);
}
function matchPossiblePaths(files) {
    const modelFilename = "model.json";
    return files.includes(modelFilename);
}
function splitExtension(str) {
    const fullName = str.split("/").pop();
    if (!fullName) {
        return {
            fileName: null,
            extension: null,
        };
    }
    const [fileName, extension] = fullName.split(".");
    return {
        fileName,
        extension,
    };
}
function fromJsonFile(pathToFile, validate) {
    const fullPath = path_1.default.join(pathToFile);
    const hasFile = files_1.default.exists(fullPath);
    if (hasFile) {
        return files_1.default.readEntity(fullPath, validate);
    }
    return null;
}
function getFileInfoFromPath(slicePath, componentName) {
    const isDirectory = files_1.default.isDirectory(slicePath);
    if (!isDirectory) {
        return { ...splitExtension(slicePath) };
    }
    const files = files_1.default.readDirectory(slicePath);
    const match = matchPossiblePaths(files);
    if (match) {
        const maybeFileComponent = findComponentFile(files, componentName);
        if (maybeFileComponent) {
            return { ...splitExtension(maybeFileComponent) };
        }
        return { fileName: null, extension: null };
    }
    throw new Error(`[slice-machine] Could not find module file for component "${componentName}" at path "${slicePath}"`);
}
function getComponentInfo(slicePath, assetsPaths, from) {
    const sliceName = getComponentName(slicePath);
    if (!sliceName || !sliceName.length) {
        console.error(`[queries/component-info] Could not find slice name at path "${slicePath}". Skipping...`);
        return;
    }
    const fileInfo = (() => {
        try {
            return getFileInfoFromPath(slicePath, sliceName);
        }
        catch (e) {
            return null;
        }
    })();
    if (fileInfo === null) {
        return;
    }
    const { fileName, extension } = fileInfo;
    const model = fromJsonFile(path_1.default.join(slicePath, "model.json"), (payload) => (0, Either_1.getOrElseW)((e) => new Error(errors_1.default.report(e)))(slices_1.SharedSlice.decode(payload)));
    if (!model) {
        return;
    }
    if (model instanceof Error) {
        console.error(`Could not parse model ${path_1.default.basename(slicePath)}\nFull error: ${model.toString()}`);
        return;
    }
    const smModel = Slice_1.Slices.toSM(model);
    const screenshotPaths = (smModel.variations || [])
        .map((v) => {
        const activeScreenshot = (0, screenshot_1.resolvePathsToScreenshot)({
            paths: assetsPaths,
            from,
            sliceName,
            variationId: v.id,
        });
        return activeScreenshot && { [v.id]: activeScreenshot };
    })
        .reduce((acc, variationPreview) => {
        return { ...acc, ...variationPreview };
    }, {});
    const maybeMock = (0, mocks_1.resolvePathsToMock)({
        paths: assetsPaths,
        from,
        sliceName,
    });
    return {
        fileName,
        extension,
        model: smModel,
        mock: maybeMock === null || maybeMock === void 0 ? void 0 : maybeMock.value,
        screenshotPaths,
    };
}
exports.getComponentInfo = getComponentInfo;
