// @slicemachine/core/build/libraries/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LibraryMeta = exports.libraries = exports.handleLibraryPath = void 0;
const fs_1 = __importDefault(require("fs"));
const path_1 = __importDefault(require("path"));
const t = __importStar(require("io-ts"));
const Either_1 = require("fp-ts/lib/Either");
const node_utils_1 = require("../node-utils");
const path_2 = require("./path");
const component_1 = require("./component");
__exportStar(require("./component"), exports);
__exportStar(require("./screenshot"), exports);
__exportStar(require("./path"), exports);
__exportStar(require("./mocks"), exports);
function handleLibraryPath(cwd, libPath) {
    const { from, isLocal, pathExists, pathToSlices, pathToLib } = (0, path_2.getInfoFromPath)(cwd, libPath);
    if (!pathExists) {
        return;
    }
    const pathsToComponents = node_utils_1.Files.readDirectory(pathToSlices.split(path_1.default.sep).join(path_1.default.posix.sep))
        .map((curr) => path_1.default.join(pathToSlices, curr))
        .filter((e) => {
        const f = e.split(path_1.default.sep).pop();
        return fs_1.default.lstatSync(e).isDirectory() && !(f === null || f === void 0 ? void 0 : f.startsWith("."));
    });
    const pathToSlice = `${isLocal ? "./" : ""}${from}${pathToSlices
        .split(from)
        .slice(1)
        .join("")}`;
    const allComponents = pathsToComponents.reduce((acc, curr) => {
        const componentInfo = (0, component_1.getComponentInfo)(curr, [cwd, path_1.default.join(cwd, ".slicemachine/assets")], from);
        if (!componentInfo) {
            return acc;
        }
        const { model } = componentInfo;
        return [
            ...acc,
            {
                from,
                href: from.replace(/\//g, "--"),
                pathToSlice,
                fileName: componentInfo.fileName,
                extension: componentInfo.extension,
                screenshotPaths: componentInfo.screenshotPaths,
                mock: componentInfo.mock,
                model,
            },
        ];
    }, []);
    const meta = exports.LibraryMeta.build(pathToLib);
    return {
        path: pathToLib,
        isLocal,
        name: from,
        components: allComponents,
        meta,
    };
}
exports.handleLibraryPath = handleLibraryPath;
function libraries(cwd, libraries) {
    return (libraries || [])
        .map((lib) => handleLibraryPath(cwd, lib))
        .filter(Boolean);
}
exports.libraries = libraries;
exports.LibraryMeta = {
    reader: t.exact(t.partial({
        name: t.string,
        version: t.string,
    })),
    build(libPath) {
        const meta = node_utils_1.Files.safeReadEntity(path_1.default.join(libPath, "meta.json"), (payload) => {
            return (0, Either_1.getOrElseW)(() => null)(exports.LibraryMeta.reader.decode(payload));
        });
        if (!meta)
            return;
        return {
            name: meta.name,
            version: meta.version,
        };
    },
};
