// @slicemachine/core/build/libraries/path.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getInfoFromPath = exports.getFormattedLibIdentifier = void 0;
const path_1 = __importDefault(require("path"));
const node_utils_1 = require("../node-utils");
var Prefix;
(function (Prefix) {
    Prefix["A"] = "@/";
    Prefix["B"] = "~/";
    Prefix["C"] = "/";
})(Prefix || (Prefix = {}));
const Identifiers = {
    "@/": 2,
    "~/": 2,
    "/": 1,
};
const getFormattedLibIdentifier = (libPath) => {
    const maybeIdentifier = Object.keys(Identifiers).find((e) => libPath.indexOf(e) === 0);
    const isLocal = !!maybeIdentifier;
    return {
        isLocal,
        identifier: maybeIdentifier,
        from: isLocal
            ? libPath.slice(Identifiers[maybeIdentifier])
            : libPath,
    };
};
exports.getFormattedLibIdentifier = getFormattedLibIdentifier;
function getInfoFromPath(startPath, libPath) {
    const { isLocal, from } = (0, exports.getFormattedLibIdentifier)(libPath);
    const pathToLib = path_1.default.join(startPath || process.cwd(), isLocal ? "" : "node_modules", isLocal ? libPath.substring(1, libPath.length) : libPath);
    const pathExists = node_utils_1.Files.exists(pathToLib);
    const pathToSlices = path_1.default.join(pathToLib, isLocal ? "." : "slices");
    return {
        isLocal,
        from,
        pathExists,
        pathToLib,
        pathToSlices,
    };
}
exports.getInfoFromPath = getInfoFromPath;
