// @slicemachine/core/build/src/core/auth.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.startServerAndOpenBrowser = exports.askSingleChar = exports.validatePayload = exports.isHandlerData = void 0;
const hapi = __importStar(require("@hapi/hapi"));
const open_1 = __importDefault(require("open"));
const utils_1 = require("../utils");
const PrismicSharedConfig_1 = require("../filesystem/PrismicSharedConfig");
const utils_2 = require("../utils");
const isHandlerData = (data) => {
    if (typeof data != "object")
        return false;
    const { email, cookies } = data;
    return (Boolean(email && cookies) &&
        Array.isArray(cookies) &&
        !cookies.some((c) => typeof c !== "string"));
};
exports.isHandlerData = isHandlerData;
const Routes = {
    authentication: (server) => (onSuccess, onFail) => ({
        method: "POST",
        path: "/",
        handler: authenticationHandler(server)(onSuccess, onFail),
    }),
    notFound: {
        method: ["GET", "POST"],
        path: "/{any*}",
        handler: (request, h) => {
            return h
                .response(`not found: [${request.method}]: ${request.url.toString()}`)
                .code(404);
        },
    },
};
function validatePayload(payload) {
    if (Buffer.isBuffer(payload))
        return null;
    return (0, exports.isHandlerData)(payload) ? payload : null;
}
exports.validatePayload = validatePayload;
const authenticationHandler = (server) => (onSuccess, onFail) => {
    return (request, h) => {
        try {
            const data = validatePayload(request.payload);
            if (!data) {
                onFail();
                h.response("Error with cookies").code(400);
                throw new Error("It seems the server didn't respond properly, please contact us.");
            }
            onSuccess(data);
            return h.response(data).code(200);
        }
        finally {
            void server.stop({ timeout: 10000 });
        }
    };
};
function buildServer(base, port, host) {
    const server = hapi.server({
        port,
        host,
        routes: {
            cors: {
                origin: [base],
                headers: ["Origin", "X-Requested-With", "Content-Type", "Accept"],
            },
        },
    });
    return server;
}
function askSingleChar(title) {
    return new Promise((resolve) => {
        process.stdout.write(title);
        const rawMode = process.stdin.isRaw;
        if (process.stdin.setRawMode)
            process.stdin.setRawMode(true);
        function handler(key) {
            const response = key.toString("utf-8");
            if (process.stdin.setRawMode)
                process.stdin.setRawMode(Boolean(rawMode));
            process.stdin.removeListener("data", handler);
            resolve(response);
        }
        process.stdin.on("data", handler);
    });
}
exports.askSingleChar = askSingleChar;
async function startServerAndOpenBrowser(url, action, base = utils_1.CONSTS.DEFAULT_BASE, port = utils_1.CONSTS.DEFAULT_SERVER_PORT) {
    const confirmation = await askSingleChar(`>> Press any key to open the browser to ${action} or q to exit:`);
    if (confirmation === "q" || confirmation === "\u0003")
        return process.exit(-1);
    const s = (0, utils_1.spinner)("Waiting for the browser response");
    const onLoginFail = () => {
        s.stop();
        (0, utils_1.writeError)(`We failed to log you into your Prismic account`);
        console.log(`Run ${(0, utils_1.bold)("npx slicemachine init")} again!`);
        process.exit(-1);
    };
    function onSuccess(data) {
        s.succeed(`Logged in as ${(0, utils_1.bold)(data.email)}`).stop();
        PrismicSharedConfig_1.PrismicSharedConfigManager.setProperties({
            cookies: utils_2.Cookie.serializeCookies(data.cookies),
            base,
        });
    }
    function onFail() {
        onLoginFail();
    }
    const server = buildServer(base, port, "localhost");
    server.route([
        Routes.authentication(server)(onSuccess, onFail),
        Routes.notFound,
    ]);
    return server.start().then(() => {
        console.log("\nOpening browser to " + (0, utils_1.underline)(url));
        s.start();
        void (0, open_1.default)(url);
        return {
            onLoginFail,
        };
    });
}
exports.startServerAndOpenBrowser = startServerAndOpenBrowser;
