// @slicemachine/core/build/src/core/communication.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createRepository = exports.validateRepositoryName = exports.listRepositories = exports.validateSession = exports.maybeParseRepoData = exports.refreshSession = void 0;
const axios_1 = __importDefault(require("axios"));
const Either_1 = require("fp-ts/Either");
const models_1 = require("../models");
const utils_1 = require("../utils");
const { DEFAULT_BASE, USER_SERVICE_BASE, USER_SERVICE_STAGING_BASE } = utils_1.CONSTS;
function toAuthUrl(path, token, base = DEFAULT_BASE) {
    const url = new URL(base);
    url.hostname = `auth.${url.hostname}`;
    url.pathname = path;
    url.searchParams.set("token", token);
    return url.toString();
}
async function refreshSession(cookies, base) {
    const token = utils_1.Cookie.parsePrismicAuthToken(cookies);
    const url = toAuthUrl("refreshtoken", token, base);
    return axios_1.default.get(url).then((res) => res.data);
}
exports.refreshSession = refreshSession;
function maybeParseRepoData(repos) {
    if (!repos)
        throw new Error("Did not receive repository data");
    if (typeof repos === "string") {
        return (0, Either_1.fold)(() => {
            throw new Error("Can't parse repo data");
        }, (f) => f)(models_1.RepositoriesRoles.decode(JSON.parse(repos)));
    }
    return repos;
}
exports.maybeParseRepoData = maybeParseRepoData;
async function validateSession(cookies, base) {
    const token = utils_1.Cookie.parsePrismicAuthToken(cookies);
    const url = toAuthUrl("validate", token, base);
    return axios_1.default
        .get(url)
        .then((res) => {
        const repositories = maybeParseRepoData(res.data.repositories);
        return {
            ...res.data,
            repositories,
        };
    });
}
exports.validateSession = validateSession;
async function listRepositories(token, base = DEFAULT_BASE) {
    const userServiceBase = DEFAULT_BASE === base ? USER_SERVICE_BASE : USER_SERVICE_STAGING_BASE;
    return axios_1.default
        .get(`${userServiceBase}/repositories`, {
        headers: { Authorization: `Bearer ${token}` },
    })
        .then((res) => {
        return (0, Either_1.getOrElseW)(() => {
            console.error("Unable to parse user repositories.");
            return [];
        })(models_1.Repositories.decode(res.data));
    });
}
exports.listRepositories = listRepositories;
async function validateRepositoryName(name, base = DEFAULT_BASE, existingRepo = false) {
    if (!name)
        return Promise.reject(new Error("repository name is required"));
    const domain = name.trim();
    const errors = [];
    const startsWithLetter = /^[a-zA-Z]/.test(domain);
    if (!startsWithLetter)
        errors.push("Must start with a letter.");
    const acceptedChars = /^[a-z0-9-]+$/.test(domain);
    if (!acceptedChars)
        errors.push("Must contain only lowercase letters, numbers and hyphens.");
    const fourCharactersOrMore = domain.length >= 4;
    if (!fourCharactersOrMore)
        errors.push("Must have four or more alphanumeric characters and/or hyphens.");
    const endsWithALetterOrNumber = /[a-z0-9]$/.test(domain);
    if (!endsWithALetterOrNumber)
        errors.push("Must end in a letter or a number.");
    const thirtyCharacterOrLess = domain.length <= 30;
    if (!thirtyCharacterOrLess)
        errors.push("Must be 30 characters or less");
    if (errors.length > 0) {
        const errorString = errors.map((d, i) => `(${i + 1}: ${d}`).join(" ");
        const msg = `Validation errors: ${errorString}`;
        return Promise.reject(new Error(msg));
    }
    const addr = new URL(base);
    addr.pathname = `/app/dashboard/repositories/${domain}/exists`;
    const url = addr.toString();
    return axios_1.default
        .get(url)
        .then((res) => res.data)
        .then((res) => {
        if (!res && !existingRepo)
            throw new Error(`${domain} is already in use`);
        if (res && existingRepo)
            throw new Error(`${domain} does not exist`);
        return domain;
    });
}
exports.validateRepositoryName = validateRepositoryName;
async function createRepository(domain, cookies, framework = models_1.Frameworks.vanillajs, base = DEFAULT_BASE) {
    const data = {
        domain,
        framework,
        plan: "personal",
        isAnnual: "false",
        role: "developer",
    };
    const address = new URL(base);
    address.pathname = "/authentication/newrepository";
    address.searchParams.append("app", "slicemachine");
    return axios_1.default
        .post(address.toString(), data, {
        headers: {
            Cookie: cookies,
            "User-Agent": "prismic-cli/sm",
        },
    })
        .catch((error) => {
        if (axios_1.default.isAxiosError(error) && error.response) {
            const message = `[${error.response.status}]: ${error.response.statusText}`;
            throw new Error(message);
        }
        throw error;
    });
}
exports.createRepository = createRepository;
