// @slicemachine/core/build/src/core/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Auth = exports.Communication = void 0;
const auth_1 = require("./auth");
const utils_1 = require("../utils");
const Communication = __importStar(require("./communication"));
const PrismicSharedConfig_1 = require("../filesystem/PrismicSharedConfig");
exports.Communication = __importStar(require("./communication"));
function createCore({ cwd, base, manifest }) {
    return {
        cwd,
        base,
        manifest,
        Repository: {
            list: async (token) => Communication.listRepositories(token),
            validateName: (name, existingRepo = false) => Communication.validateRepositoryName(name, base, existingRepo),
            create: async (domain, token, framework) => Communication.createRepository(domain, token, framework, base),
        },
    };
}
exports.default = createCore;
async function startAuth({ base, url, action, }) {
    const { onLoginFail } = await (0, auth_1.startServerAndOpenBrowser)(url, action, base);
    try {
        await utils_1.Poll.startPolling(() => exports.Auth.validateSession(base), (user) => !!user, 3000, 60);
        return;
    }
    catch (e) {
        onLoginFail();
    }
}
exports.Auth = {
    login: async (base) => {
        const endpoints = utils_1.Endpoints.buildEndpoints(base);
        return startAuth({
            base,
            url: endpoints.Dashboard.cliLogin,
            action: "login",
        });
    },
    signup: async (base) => {
        const endpoints = utils_1.Endpoints.buildEndpoints(base);
        return startAuth({
            base,
            url: endpoints.Dashboard.cliSignup,
            action: "signup",
        });
    },
    logout: () => PrismicSharedConfig_1.PrismicSharedConfigManager.remove(),
    validateSession: async (requiredBase) => {
        const config = PrismicSharedConfig_1.PrismicSharedConfigManager.get();
        if (!config.cookies.length)
            return Promise.resolve(null);
        if (requiredBase != config.base)
            return Promise.resolve(null);
        return Communication.validateSession(config.cookies, requiredBase).catch(() => null);
    },
};
