// @prismicio/types-internal/lib/customtypes/widgets/slices/SharedSlice.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedSlice = exports.Variation = void 0;
const tslib_1 = require("tslib");
const t = (0, tslib_1.__importStar)(require("io-ts"));
const withFallback_1 = require("io-ts-types/lib/withFallback");
const NestableWidget_1 = (0, tslib_1.__importDefault)(require("../nestable/NestableWidget"));
const SlicesTypes_1 = (0, tslib_1.__importDefault)(require("./SlicesTypes"));
const IMAGE_PLACEHOLDER_URL = "https://images.prismic.io/slice-machine/621a5ec4-0387-4bc5-9860-2dd46cbc07cd_default_ss.png?auto=compress,format";
exports.Variation = t.exact(t.intersection([
    t.type({
        id: t.string,
        name: t.string,
        description: t.string,
        imageUrl: (0, withFallback_1.withFallback)(t.string, IMAGE_PLACEHOLDER_URL),
        docURL: t.string,
        version: t.string,
    }),
    t.partial({
        display: t.string,
        primary: t.record(t.string, NestableWidget_1.default),
        items: t.record(t.string, NestableWidget_1.default),
    }),
]));
exports.SharedSlice = t.exact(t.intersection([
    t.type({
        id: t.string,
        type: t.literal(SlicesTypes_1.default.SharedSlice),
        name: t.string,
        variations: t.array(exports.Variation),
    }),
    t.partial({
        description: t.string,
    }),
]));
