// slice-machine-ui/build/lib/models/common/widgets/GeoPoint/Mock/Form.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MockConfigForm = void 0;
const react_1 = require("react");
const fast_deep_equal_1 = __importDefault(require("fast-deep-equal"));
const theme_ui_1 = require("theme-ui");
const formik_1 = require("formik");
const _1 = require(".");
const consts_1 = require("@lib/consts");
const InputDeleteIcon_1 = __importDefault(require("components/InputDeleteIcon"));
const Preview_1 = __importDefault(require("components/Card/Preview"));
const places_1 = __importDefault(require("./places"));
const createMapsUrl = (points) => `https://google.com/maps/@${points.latitude},${points.longitude}`;
const InputSrc = ({ contentValue, value, onUpdate, onDelete }) => {
    return (<theme_ui_1.Label variant="label.primary" sx={{ display: "block" }}>
      <theme_ui_1.Text as="span">Google Maps Url</theme_ui_1.Text>
      <theme_ui_1.Input value={value} placeholder="https://maps.google.com/@..." onChange={(e) => onUpdate(e.target.value)}/>
      <InputDeleteIcon_1.default onClick={onDelete}/>
      {contentValue ? (<theme_ui_1.Box sx={{
                position: "absolute",
                top: "34px",
                right: "38px",
                color: "textClear",
            }}>
          <span>
            lat:{" "}
            {contentValue.latitude}
          </span>
          &nbsp;&nbsp;
          <span>
            long:{" "}
            {contentValue.longitude}
          </span>
        </theme_ui_1.Box>) : null}
    </theme_ui_1.Label>);
};
const Form = () => {
    var _a;
    const { values, setFieldValue } = formik_1.useFormikContext();
    const contentValue = ((_a = values[consts_1.MockConfigKey]) === null || _a === void 0 ? void 0 : _a.content) || null;
    const [url, setUrl] = react_1.useState(contentValue ? createMapsUrl(contentValue) : "");
    const onUpdate = (points) => {
        setUrl(createMapsUrl(points));
        setFieldValue(consts_1.MockConfigKey, {
            content: points,
        });
    };
    const parseMapUrl = (url) => {
        setUrl(url);
        if (!url) {
            return;
        }
        const res = url.match(/@(.*),(.*),/);
        if (res && res.length === 3) {
            onUpdate({
                latitude: parseFloat(res[1]),
                longitude: parseFloat(res[2]),
            });
        }
    };
    const onDelete = () => {
        setUrl("");
        setFieldValue(consts_1.MockConfigKey, {});
    };
    return (<theme_ui_1.Box>
      <InputSrc contentValue={contentValue} value={url} onUpdate={parseMapUrl} onDelete={onDelete}/>
      <theme_ui_1.Label variant="label.primary" mt={3} mb={2} htmlFor="places">
        Presets
      </theme_ui_1.Label>
      <theme_ui_1.Box sx={{
            display: "grid",
            gridColumnGap: "8px",
            gridTemplateColumns: "1fr 1fr",
        }}>
        {places_1.default.map((place) => (<Preview_1.default key={place.name} title={place.name} imageUrl={place.imageUrl} selected={fast_deep_equal_1.default(place.points, contentValue)} titleSx={{
                fontSize: 2,
                maxWidth: "80%",
            }} sx={{
                mb: 3,
            }} onClick={() => onUpdate(place.points)}/>))}
      </theme_ui_1.Box>
    </theme_ui_1.Box>);
};
Form.initialValues = _1.initialValues;
exports.MockConfigForm = Form;
