// slice-machine-ui/build/lib/models/common/widgets/Image/Mock/dataset/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Portrait = exports.Nature = exports.Minimal = exports.Computer = exports.Colorful = exports.Business = exports.Art = exports.Architecture = exports.Animals = void 0;
const Animal_1 = __importDefault(require("./Animal"));
exports.Animals = Animal_1.default;
const Architecture_1 = __importDefault(require("./Architecture"));
exports.Architecture = Architecture_1.default;
const Art_1 = __importDefault(require("./Art"));
exports.Art = Art_1.default;
const Business_1 = __importDefault(require("./Business"));
exports.Business = Business_1.default;
const Colorful_1 = __importDefault(require("./Colorful"));
exports.Colorful = Colorful_1.default;
const Computer_1 = __importDefault(require("./Computer"));
exports.Computer = Computer_1.default;
const Minimal_1 = __importDefault(require("./Minimal"));
exports.Minimal = Minimal_1.default;
const Nature_1 = __importDefault(require("./Nature"));
exports.Nature = Nature_1.default;
const Portrait_1 = __importDefault(require("./Portrait"));
exports.Portrait = Portrait_1.default;
