// slice-machine-ui/build/lib/builders/common/EditModal/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_modal_1 = __importDefault(require("react-modal"));
const SliceMachineModal_1 = __importDefault(require("@components/SliceMachineModal"));
const deepmerge_1 = __importDefault(require("deepmerge"));
const theme_ui_1 = require("theme-ui");
const yup = __importStar(require("yup"));
const Widgets = __importStar(require("@lib/models/common/widgets/withGroup"));
const forms_1 = require("@lib/forms");
const consts_1 = require("@lib/consts");
const WithTabs_1 = __importDefault(require("@components/Card/WithTabs"));
const ItemHeader_1 = __importDefault(require("@components/ItemHeader"));
const Flex_1 = require("@components/Flex");
const Form_1 = __importDefault(require("./Form"));
const Field_1 = __importDefault(require("./Field"));
const utils_1 = require("../../utils");
const utils_2 = require("@lib/utils");
if (process.env.NODE_ENV !== "test") {
    react_modal_1.default.setAppElement("#__next");
}
const FORM_ID = "edit-modal-form";
const EditModal = ({ close, data, fields, onSave, getFieldMockConfig }) => {
    if (!data.isOpen) {
        return null;
    }
    const { theme } = theme_ui_1.useThemeUI();
    const { field: [apiId, initialModelValues], } = data;
    const maybeWidget = utils_1.findWidgetByConfigOrType(Widgets, initialModelValues.config, initialModelValues.type);
    if (!maybeWidget) {
        return <div>{initialModelValues.type} not found</div>;
    }
    const { Meta: { icon: WidgetIcon }, FormFields, MockConfigForm, Form: CustomForm, schema: widgetSchema, } = maybeWidget;
    const initialConfig = {
        ...forms_1.createInitialValues(utils_2.removeProp(FormFields, "id")),
        ...initialModelValues.config,
    };
    const { res: validatedSchema, err } = (() => {
        try {
            return {
                res: widgetSchema.validateSync({
                    type: initialModelValues.type,
                    config: initialConfig,
                }, { stripUnknown: true }),
            };
        }
        catch (e) {
            return { err: e };
        }
    })();
    if (err) {
        console.error(`[EditModal] Failed to validate field of type ${initialModelValues.type}.\n Please update model.json accordingly.`);
        console.error(err);
    }
    const initialValues = {
        id: apiId,
        config: validatedSchema ? validatedSchema.config : initialConfig,
        [consts_1.MockConfigKey]: deepmerge_1.default((MockConfigForm === null || MockConfigForm === void 0 ? void 0 : MockConfigForm.initialValues) || {}, getFieldMockConfig({ apiId }) || {}),
    };
    const [idMatches, idMessage] = FormFields.id.validate.matches;
    const validationSchema = yup.object().shape({
        id: yup.string().matches(idMatches, idMessage).min(3).max(35).required(),
        config: widgetSchema.fields.config,
    });
    const formId = `${FORM_ID}-${Math.random().toString()}`;
    return (<SliceMachineModal_1.default isOpen shouldCloseOnOverlayClick onRequestClose={close} contentLabel="Widget Form Modal">
      <Form_1.default apiId={apiId} formId={formId} initialValues={initialValues} validationSchema={validationSchema} FormFields={FormFields} onSave={({ newKey, value }, mockValue) => {
            const maybeUpdatedMockValue = (MockConfigForm === null || MockConfigForm === void 0 ? void 0 : MockConfigForm.onSave) && mockValue && Object.keys(mockValue).length
                ?
                    MockConfigForm.onSave(mockValue, value)
                : mockValue;
            const definitiveMockValue = (() => {
                if (maybeUpdatedMockValue &&
                    Object.keys(maybeUpdatedMockValue).length &&
                    !!Object.entries(maybeUpdatedMockValue).find(([, v]) => v !== null)) {
                    return maybeUpdatedMockValue;
                }
                return null;
            })();
            const updatedValue = {
                ...initialModelValues,
                ...value,
            };
            onSave({
                apiId,
                newKey,
                value: updatedValue,
                mockValue: definitiveMockValue,
            });
            close();
        }}>
        {(props) => {
            const { values: { id, config: { label }, }, isValid, isSubmitting, initialValues, } = props;
            return (<WithTabs_1.default borderFooter footerSx={{ p: 0, mb: 5 }} tabs={["Field Model", "Mock config"]} Header={({ radius }) => (<theme_ui_1.Flex sx={{
                        p: 3,
                        bg: "headSection",
                        alignItems: "center",
                        justifyContent: "space-between",
                        borderTopLeftRadius: radius,
                        borderTopRightRadius: radius,
                    }}>
                  <ItemHeader_1.default theme={theme} text={label || id} WidgetIcon={WidgetIcon}/>
                  <theme_ui_1.Close onClick={close} type="button"/>
                </theme_ui_1.Flex>)} Footer={<theme_ui_1.Flex sx={{ alignItems: "space-between", bg: "headSection", p: 3 }}>
                  <theme_ui_1.Box sx={{ ml: "auto" }}/>
                  <theme_ui_1.Button mr={2} type="button" onClick={close} variant="secondary">
                    Cancel
                  </theme_ui_1.Button>
                  <theme_ui_1.Button form={formId} type="submit" disabled={!isValid && isSubmitting}>
                    Save
                  </theme_ui_1.Button>
                </theme_ui_1.Flex>}>
              {CustomForm ? (<CustomForm {...props} fields={fields}/>) : (<Flex_1.Flex>
                  {Object.entries(FormFields).map(([key, field]) => (<Flex_1.Col key={key}>
                      <Field_1.default fieldName={forms_1.createFieldNameFromKey(key)} formField={field} fields={fields} initialValues={initialValues}/>
                    </Flex_1.Col>))}
                </Flex_1.Flex>)}

              <theme_ui_1.Box>
                {MockConfigForm ? (<theme_ui_1.Box>
                    <MockConfigForm initialValues={initialValues}/>
                  </theme_ui_1.Box>) : (<p>Mock data for this field is not yet available.</p>)}
              </theme_ui_1.Box>
            </WithTabs_1.default>);
        }}
      </Form_1.default>
    </SliceMachineModal_1.default>);
};
exports.default = EditModal;
