// slice-machine-ui/build/lib/models/common/widgets/Image/Mock/components.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImagesList = exports.ImagesListCards = void 0;
const react_1 = require("react");
const theme_ui_1 = require("theme-ui");
const Flex_1 = require("components/Flex");
const Preview_1 = __importDefault(require("components/Card/Preview"));
const dataset = __importStar(require("./dataset"));
const ImageInput = ({ src, onChange, selected }) => (<input style={{
        padding: "0",
        marginBottom: "8px",
        maxWidth: "100%",
        border: selected
            ? "5px solid rgba(81, 99, 186, 1)"
            : "5px solid transparent",
    }} onClick={onChange} type="image" src={src}/>);
const RenderCol = ({ elements, cols, onSelect }) => (<Flex_1.Col cols={cols}>
    {elements.map(([key, value]) => (<Preview_1.default key={key} title={key} imageUrl={`${value[0].raw}&q=20&w=120`} onClick={() => onSelect(key)}/>))}
  </Flex_1.Col>);
const ImagesListCards = ({ onSelect }) => {
    const entries = Object.entries(dataset);
    const Cols = [
        entries.slice(0, entries.length / 3),
        entries.slice(entries.length / 3, (entries.length / 3) * 2),
        entries.slice((entries.length / 3) * 2),
    ];
    return (<Flex_1.Flex mt={3} px={0} sx={{ display: "flex" }}>
      <RenderCol onSelect={onSelect} cols={3} elements={Cols[0]}/>
      <RenderCol onSelect={onSelect} cols={3} elements={Cols[1]}/>
      <RenderCol onSelect={onSelect} cols={3} elements={Cols[2]}/>
    </Flex_1.Flex>);
};
exports.ImagesListCards = ImagesListCards;
const ImagesList = ({ listName, images, value, onChange }) => {
    const { theme } = theme_ui_1.useThemeUI();
    return (<react_1.Fragment>
      <div style={{
            ...theme.styles.fixedHeader,
            display: "flex",
            background: theme.colors.background,
            padding: "0 8px",
            borderBottom: `1px solid ${theme.colors.borders}`,
        }}>
        <h3>{listName}</h3>
      </div>
      <div style={{
            display: "flex",
            flexWrap: "wrap",
            justifyContent: "space-between",
            padding: "64px 8px 20px",
        }}>
        <div style={{
            flex: `0 ${100 / 2 - 1}%`,
            marginBottom: "4px",
        }}>
          {images.slice(0, images.length / 2).map((image) => (<ImageInput key={image.raw} src={`${image.raw}&q=80&w=400`} onChange={onChange} selected={image.raw.indexOf(value) === 0}/>))}
        </div>
        <div style={{
            flex: `0 ${100 / 2 - 1}%`,
            marginBottom: "4px",
        }}>
          {images.slice(images.length / 2).map((image) => (<ImageInput key={image.raw} src={`${image.raw}&q=80&w=400`} onChange={onChange} selected={image.raw.indexOf(value) === 0}/>))}
        </div>
      </div>
    </react_1.Fragment>);
};
exports.ImagesList = ImagesList;
