// slice-machine-ui/build/lib/models/common/widgets/Image/Mock/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleMockConfig = exports.handleMockContent = exports.initialValues = void 0;
const dataset_1 = require("./dataset");
exports.initialValues = null;
const createImageArray = ({ src: intialSrc }, constraint = {}, thumbnails = []) => {
    const { width = 900, height = 500 } = constraint;
    const src = intialSrc ||
        dataset_1.Computer[Math.floor(Math.random() * dataset_1.Computer.length)].raw.split("?")[0];
    return {
        dimensions: { width, height },
        alt: "Placeholder image",
        copyright: null,
        url: `${src}?w=${width}&h=${height}&fit=crop`,
        ...thumbnails.reduce((acc, e, i) => ({
            ...acc,
            [e.name]: createImageArray({ src }, { width: e.width, height: e.height }),
        }), []),
    };
};
const handleMockContent = (mockContent, { constraint, thumbnails }) => {
    if (Array.isArray(mockContent)) {
        return mockContent;
    }
    const args = [
        {
            src: mockContent,
        },
        constraint,
        thumbnails,
    ];
    return createImageArray(...args);
};
exports.handleMockContent = handleMockContent;
const handleMockConfig = (_, model) => exports.handleMockContent(null, model);
exports.handleMockConfig = handleMockConfig;
