// @slicemachine/core/build/models/Framework.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SupportedFrameworks = exports.FrameworksC = exports.Frameworks = void 0;
const t = __importStar(require("io-ts"));
var Frameworks;
(function (Frameworks) {
    Frameworks["none"] = "none";
    Frameworks["nuxt"] = "nuxt";
    Frameworks["previousNuxt"] = "previousNuxt";
    Frameworks["next"] = "next";
    Frameworks["gatsby"] = "gatsby";
    Frameworks["vue"] = "vue";
    Frameworks["react"] = "react";
    Frameworks["svelte"] = "svelte";
    Frameworks["vanillajs"] = "vanillajs";
    Frameworks["previousNext"] = "previousNext";
})(Frameworks = exports.Frameworks || (exports.Frameworks = {}));
exports.FrameworksC = t.keyof({
    [Frameworks.none]: null,
    [Frameworks.nuxt]: null,
    [Frameworks.previousNuxt]: null,
    [Frameworks.next]: null,
    [Frameworks.gatsby]: null,
    [Frameworks.vue]: null,
    [Frameworks.react]: null,
    [Frameworks.svelte]: null,
    [Frameworks.vanillajs]: null,
    [Frameworks.previousNext]: null,
});
exports.SupportedFrameworks = [
    Frameworks.none,
    Frameworks.nuxt,
    Frameworks.previousNuxt,
    Frameworks.next,
    Frameworks.vue,
    Frameworks.react,
    Frameworks.svelte,
    Frameworks.previousNext,
];
