// @slicemachine/core/build/models/Group.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Groups = exports.GroupSM = exports.GroupConfig = void 0;
const t = __importStar(require("io-ts"));
const widgets_1 = require("@prismicio/types-internal/lib/customtypes/widgets");
const validators_1 = require("@prismicio/types-internal/lib/validators");
const Fields_1 = require("../models/Fields");
const Either_1 = require("fp-ts/lib/Either");
exports.GroupConfig = t.exact(t.partial({
    label: validators_1.StringOrNull,
    repeat: t.boolean,
    fields: Fields_1.FieldsSM,
}));
exports.GroupSM = t.exact(t.intersection([
    t.type({
        type: t.literal(widgets_1.WidgetTypes.Group),
    }),
    t.partial({
        fieldset: validators_1.StringOrNull,
        icon: t.string,
        description: t.string,
        config: exports.GroupConfig,
    }),
]));
exports.Groups = {
    fromSM(group) {
        const fields = (() => {
            var _a;
            if (!((_a = group.config) === null || _a === void 0 ? void 0 : _a.fields))
                return;
            return group.config.fields.reduce((acc, { key, value }) => ({ ...acc, [key]: value }), {});
        })();
        return (0, Either_1.getOrElseW)(() => {
            throw new Error("Error while parsing an SM group.");
        })(widgets_1.Group.decode({
            ...group,
            config: {
                ...group.config,
                ...(fields ? { fields } : {}),
            },
        }));
    },
    toSM(group) {
        const fields = (() => {
            var _a, _b;
            if (!((_a = group.config) === null || _a === void 0 ? void 0 : _a.fields))
                return;
            return Object.entries(((_b = group.config) === null || _b === void 0 ? void 0 : _b.fields) || {}).map(([key, value]) => ({
                key,
                value,
            }));
        })();
        return (0, Either_1.getOrElseW)(() => {
            throw new Error("Error while parsing an SM group.");
        })(exports.GroupSM.decode({
            ...group,
            config: {
                ...group.config,
                ...(fields ? { fields } : {}),
            },
        }));
    },
};
