// @slicemachine/core/build/models/Manifest.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManifestHelper = exports.Manifest = exports.FrameworkCodec = void 0;
const t = __importStar(require("io-ts"));
const Framework_1 = require("./Framework");
const apiEndpoint = new t.Type("apiEndpoint", (input) => typeof input === "string", (input, context) => {
    if (typeof input !== "string")
        return t.failure(input, context, "apiEndpoint should be a string");
    try {
        new URL(input);
    }
    catch {
        return t.failure(input, context, "could not parse apiEndpoint: invalid url.");
    }
    const url = new URL(input);
    if (/\/api\/v2\/?$/.test(url.pathname) === false) {
        return t.failure(input, context, 'apiEndpoint should end with "/api/v2".');
    }
    const regx = new RegExp("^https?://[a-z0-9][a-z0-9-]{2,}[a-z0-9](.cdn)?.(prismic.io|wroom.io|wroom.test)/api/v2/?$", "gi");
    const result = regx.test(input);
    return result
        ? t.success(input)
        : t.failure(input, context, `apiEndpoint should match ${regx.source}`);
}, t.identity);
exports.FrameworkCodec = new t.Type("framework", (input) => Framework_1.FrameworksC.is(input), (input, context) => {
    const frameworks = Object.keys(Framework_1.Frameworks);
    if (typeof input !== "string" || !Framework_1.FrameworksC.is(input)) {
        return t.failure(input, context, `framework should be one of ${frameworks.join(", ")}. Set framework to one of these values or remove it and slice-machine will guess the framework.`);
    }
    return t.success(input);
}, t.identity);
exports.Manifest = t.intersection([
    t.type({
        apiEndpoint,
    }),
    t.partial({
        framework: exports.FrameworkCodec,
        storybook: t.string,
        localSliceSimulatorURL: t.string,
        libraries: t.array(t.string),
        chromaticAppId: t.string,
        _latest: t.string,
        tracking: t.boolean,
    }),
]);
exports.ManifestHelper = {
    localLibraries(manifest) {
        return (manifest.libraries || [])
            .filter((l) => l.startsWith("~/") || l.startsWith("@/"))
            .map((lib) => {
            const prefix = lib.indexOf("~") != -1 ? `~` : `@`;
            return {
                prefix,
                path: lib.replace(`${prefix}/`, ""),
                fullPath: lib,
            };
        });
    },
};
