// @slicemachine/core/build/models/Slices.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SliceZone = exports.SlicesSM = void 0;
const t = __importStar(require("io-ts"));
const Either_1 = require("fp-ts/lib/Either");
const slices_1 = require("@prismicio/types-internal/lib/customtypes/widgets/slices");
const Slices_1 = require("@prismicio/types-internal/lib/customtypes/widgets/slices/Slices");
const widgets_1 = require("@prismicio/types-internal/lib/customtypes/widgets");
exports.SlicesSM = t.type({
    key: t.string,
    value: t.array(t.type({
        key: t.string,
        value: t.union([slices_1.LegacySlice, slices_1.CompositeSlice, slices_1.SharedSliceRef]),
    })),
});
exports.SliceZone = {
    fromSM(slices) {
        return (0, Either_1.getOrElseW)(() => {
            throw new Error("Error while parsing an SM slicezone");
        })(Slices_1.DynamicSlices.decode({
            type: widgets_1.WidgetTypes.Slices,
            fieldset: "Slice Zone",
            config: {
                choices: slices.value.reduce((acc, { key, value }) => ({ ...acc, [key]: value }), {}),
            },
        }));
    },
    toSM(key, slices) {
        var _a;
        return (0, Either_1.getOrElseW)(() => {
            throw new Error("Error while parsing a prismic slicezone");
        })(exports.SlicesSM.decode({
            key,
            value: Object.entries(((_a = slices.config) === null || _a === void 0 ? void 0 : _a.choices) || []).map(([key, value]) => ({ key, value })),
        }));
    },
};
