// @slicemachine/core/build/models/Slice.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Slices = exports.Variations = exports.SliceSM = exports.SliceMock = exports.VariationMock = exports.VariationSM = exports.WidgetsArea = void 0;
const t = __importStar(require("io-ts"));
const slices_1 = require("@prismicio/types-internal/lib/customtypes/widgets/slices");
const Either_1 = require("fp-ts/lib/Either");
const Fields_1 = require("./Fields");
var WidgetsArea;
(function (WidgetsArea) {
    WidgetsArea["Primary"] = "primary";
    WidgetsArea["Items"] = "items";
})(WidgetsArea = exports.WidgetsArea || (exports.WidgetsArea = {}));
exports.VariationSM = t.intersection([
    t.type({
        id: t.string,
        name: t.string,
        description: t.string,
        docURL: t.string,
        version: t.string,
    }),
    t.partial({
        imageUrl: t.string,
        primary: Fields_1.FieldsSM,
        items: Fields_1.FieldsSM,
        display: t.string,
    }),
]);
exports.VariationMock = t.type({
    variation: t.string,
    name: t.string,
    slice_type: t.string,
    items: t.array(t.unknown),
    primary: t.record(t.string, t.unknown),
});
exports.SliceMock = t.array(exports.VariationMock);
exports.SliceSM = t.intersection([
    t.type({
        id: t.string,
        type: t.literal(slices_1.SlicesTypes.SharedSlice),
        name: t.string,
        variations: t.array(exports.VariationSM),
    }),
    t.partial({
        description: t.string,
    }),
]);
exports.Variations = {
    fromSM(variation) {
        var _a, _b;
        return (0, Either_1.getOrElseW)(() => {
            throw new Error("Error while parsing an SM slice variation.");
        })(slices_1.Variation.decode({
            ...variation,
            primary: (_a = variation.primary) === null || _a === void 0 ? void 0 : _a.reduce((acc, { key, value }) => ({ ...acc, [key]: value }), {}),
            items: (_b = variation.items) === null || _b === void 0 ? void 0 : _b.reduce((acc, { key, value }) => ({ ...acc, [key]: value }), {}),
        }));
    },
    toSM(variation) {
        return (0, Either_1.getOrElseW)(() => {
            throw new Error("Error while parsing a prismic slice variation.");
        })(exports.VariationSM.decode({
            ...variation,
            primary: Object.entries(variation.primary || {}).map(([key, value]) => ({
                key,
                value,
            })),
            items: Object.entries(variation.items || {}).map(([key, value]) => ({
                key,
                value,
            })),
        }));
    },
};
exports.Slices = {
    fromSM(slice) {
        return (0, Either_1.getOrElseW)(() => {
            throw new Error("Error while prismic an SM prismic slice");
        })(slices_1.SharedSlice.decode({
            ...slice,
            variations: slice.variations.map(exports.Variations.fromSM),
        }));
    },
    toSM(slice) {
        return (0, Either_1.getOrElseW)(() => {
            throw new Error("Error while prismic a prismic slice");
        })(exports.SliceSM.decode({
            ...slice,
            variations: slice.variations.map(exports.Variations.toSM),
        }));
    },
};
