// @slicemachine/core/build/models/Variation.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VariationMock = exports.VariationAsArray = exports.VariationAsObject = exports.Variation = exports.AsObject = exports.AsArray = exports.WidgetsArea = void 0;
const t = __importStar(require("io-ts"));
const fields_1 = require("./CustomType/fields");
var WidgetsArea;
(function (WidgetsArea) {
    WidgetsArea["Primary"] = "primary";
    WidgetsArea["Items"] = "items";
})(WidgetsArea = exports.WidgetsArea || (exports.WidgetsArea = {}));
exports.AsArray = t.array(t.type({ key: t.string, value: fields_1.Field }));
exports.AsObject = t.record(t.string, fields_1.Field);
const Variation = (formatReader) => t.intersection([
    t.type({
        id: t.string,
        name: t.string,
        description: t.string,
        docURL: t.string,
        version: t.string,
    }),
    t.partial({
        imageUrl: t.string,
        primary: formatReader,
        items: formatReader,
        display: t.string,
    }),
]);
exports.Variation = Variation;
exports.VariationAsObject = (0, exports.Variation)(exports.AsObject);
exports.VariationAsArray = (0, exports.Variation)(exports.AsArray);
exports.VariationMock = t.type({
    variation: t.string,
    name: t.string,
    slice_type: t.string,
    items: t.array(t.unknown),
    primary: t.record(t.string, t.unknown),
});
