// slice-machine-ui/build/server/src/api/services/getEnv.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const parse_domain_1 = require("parse-domain");
const getPrismicData_1 = __importDefault(require("./getPrismicData"));
const fs_1 = require("@lib/mock/misc/fs");
const manifest_1 = __importStar(require("@lib/env/manifest"));
const http_1 = __importDefault(require("@lib/models/common/http"));
const core_1 = require("@slicemachine/core");
const node_utils_1 = require("@slicemachine/core/build/node-utils");
const preferWroomBase_1 = __importDefault(require("@lib/utils/preferWroomBase"));
const versions_1 = require("@lib/env/versions");
function validate(config) {
    const errors = {};
    if (config.framework &&
        !node_utils_1.Framework.isFrameworkSupported(core_1.Models.Frameworks[config.framework])) {
        const options = Object.values(core_1.Models.SupportedFrameworks);
        errors.framework = {
            message: `The framework set in sm.json is invalid`,
            example: "react",
            run: `Set framework to one of the following: ${options.join(", ")}`,
        };
    }
    return errors;
}
function extractRepo(parsedRepo) {
    switch (parsedRepo.type) {
        case parse_domain_1.ParseResultType.Listed:
            if (parsedRepo.labels.length) {
                return parsedRepo.labels[0];
            }
            if (parsedRepo.subDomains.length) {
                return parsedRepo.subDomains[0];
            }
        default:
            return;
    }
}
async function getEnv(maybeCustomCwd) {
    const cwd = maybeCustomCwd || process.env.CWD || process.cwd();
    if (!cwd) {
        const message = "[api/env]: Unrecoverable error. Could not find cwd. Exiting..";
        console.error(message);
        throw new Error(message);
    }
    const manifestInfo = manifest_1.default(cwd);
    if (manifestInfo.state !== manifest_1.ManifestState.Valid || !manifestInfo.content) {
        console.error(manifestInfo.message);
        throw new Error(manifestInfo.message);
    }
    const base = preferWroomBase_1.default(manifestInfo.content.apiEndpoint);
    const prismicData = getPrismicData_1.default();
    if (!prismicData.isOk()) {
        const message = "[api/env]: Unrecoverable error. ~/.prismic file unreadable";
        console.error(message);
        throw new Error(message);
    }
    const smChangelog = await versions_1.getPackageChangelog(appRoot);
    const maybeErrors = validate(manifestInfo.content);
    const parsedRepo = parse_domain_1.parseDomain(parse_domain_1.fromUrl(manifestInfo.content.apiEndpoint));
    const repo = extractRepo(parsedRepo);
    const mockConfig = fs_1.getConfig(cwd);
    const client = http_1.default(cwd, base, repo, prismicData.value.auth);
    return {
        errors: maybeErrors,
        env: {
            cwd,
            repo,
            manifest: manifestInfo.content,
            prismicData: prismicData.value,
            changelog: smChangelog,
            mockConfig,
            framework: node_utils_1.Framework.defineFramework({
                cwd,
                manifest: manifestInfo.content,
            }),
            isUserLoggedIn: !!prismicData.value.auth && !!repo,
            baseUrl: `http://localhost:${process.env.PORT || "9999"}`,
            client,
        },
    };
}
exports.default = getEnv;
