// slice-machine-ui/build/server/src/api/services/semver.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.compare = void 0;
const core_1 = require("@slicemachine/core");
const files_1 = __importDefault(require("../../../../lib/utils/files"));
const paths_1 = require("../../../../lib/models/paths");
const versions_1 = require("./versions");
const versions_2 = require("./versions");
const DefaultComparison = {
    currentVersion: "",
    updateAvailable: false,
    latestNonBreakingVersion: null,
    packageManager: "npm",
    versions: [],
};
async function compare(cwd) {
    var _a;
    const pkg = core_1.FileSystem.retrieveJsonPackage(cwd);
    if (!pkg.exists || !((_a = pkg.content) === null || _a === void 0 ? void 0 : _a.name))
        return DefaultComparison;
    const currentVersion = pkg.content.version;
    const versions = await versions_2.findPackageVersions(pkg.content.name);
    const updateAvailable = versions_1.isUpdateAvailable(currentVersion, versions);
    const latestNonBreakingVersion = versions_1.findLatestNonBreakingUpdate(currentVersion, versions);
    const packageManager = files_1.default.exists(paths_1.YarnLock(cwd)) ? 'yarn' : 'npm';
    return {
        currentVersion,
        updateAvailable,
        latestNonBreakingVersion,
        packageManager,
        versions
    };
}
exports.compare = compare;
