// slice-machine-ui/build/server/src/api/services/versions.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.findPackageVersions = exports.isUpdateAvailable = exports.findLatestNonBreakingUpdate = exports.getPackageChangelog = void 0;
const npm_registry_fetch_1 = __importDefault(require("npm-registry-fetch"));
const semver_1 = __importDefault(require("semver"));
const axios_1 = __importDefault(require("axios"));
const core_1 = require("@slicemachine/core");
async function getPackageChangelog(dependencyCwd) {
    var _a;
    const pkg = core_1.FileSystem.retrieveJsonPackage(dependencyCwd);
    if (!pkg.exists || !((_a = pkg.content) === null || _a === void 0 ? void 0 : _a.name))
        return {
            currentVersion: "",
            updateAvailable: false,
            latestNonBreakingVersion: null,
            versions: [],
        };
    const currentVersion = pkg.content.version;
    const versions = await findPackageVersions(pkg.content.name);
    const updateAvailable = isUpdateAvailable(currentVersion, versions);
    const latestNonBreakingVersion = findLatestNonBreakingUpdate(currentVersion, versions);
    return {
        currentVersion,
        updateAvailable,
        latestNonBreakingVersion,
        versions,
    };
}
exports.getPackageChangelog = getPackageChangelog;
function findLatestNonBreakingUpdate(current, versions) {
    const minorUpdate = findHighestUpdateByKind("minor", current, versions);
    if (minorUpdate)
        return minorUpdate;
    const patchUpdate = findHighestUpdateByKind("patch", current, versions);
    if (patchUpdate)
        return patchUpdate;
    return null;
}
exports.findLatestNonBreakingUpdate = findLatestNonBreakingUpdate;
function isUpdateAvailable(current, versions) {
    return Boolean(findHighestUpdateByKind("patch", current, versions) ||
        findHighestUpdateByKind("minor", current, versions) ||
        findHighestUpdateByKind("major", current, versions));
}
exports.isUpdateAvailable = isUpdateAvailable;
async function findPackageVersions(packageName) {
    const versions = await fetchVersionsFromNpm(packageName);
    const stableVersions = versions.filter((version) => /^\d+\.\d+\.\d+$/.test(version) && semver_1.default.lte("0.1.0", version));
    const stableVersionsOrdered = stableVersions.sort().reverse();
    const releaseNotesMap = await axios_1.default
        .get("https://api.github.com/repos/prismicio/slice-machine/releases")
        .then((response) => {
        const releaseNotes = response.data;
        return releaseNotes.reduce((map, releaseNote) => {
            map[releaseNote.name] = releaseNote;
            return map;
        }, {});
    })
        .catch(() => {
        console.log(`Couldn't retrieve Github release notes for package ${packageName}`);
        return {};
    });
    return stableVersionsOrdered.map((stableVersion) => {
        return {
            versionNumber: stableVersion,
            releaseNote: releaseNotesMap[stableVersion]
                ? releaseNotesMap[stableVersion].body
                : null,
        };
    });
}
exports.findPackageVersions = findPackageVersions;
async function fetchVersionsFromNpm(packageName) {
    return npm_registry_fetch_1.default
        .json(packageName)
        .then((json) => {
        const versions = (json.versions || {});
        return Object.keys(versions);
    })
        .catch(() => []);
}
const findHighestUpdateByKind = (kind, current, versions) => {
    const minorVersion = current.replace(/^(\d+\.\d+).*/, "$1");
    const majorVersion = current.replace(/^(\d+).*/, "$1");
    const result = versions.reduce((acc, { versionNumber }) => {
        if (!/^\d+\.\d+\.\d+$/.test(versionNumber))
            return acc;
        if (semver_1.default.gt(acc, versionNumber))
            return acc;
        if (kind === "patch" && versionNumber.startsWith(minorVersion))
            return versionNumber;
        if (kind === "minor" &&
            versionNumber.startsWith(majorVersion) &&
            !versionNumber.startsWith(minorVersion)) {
            return versionNumber;
        }
        if (kind === "major" && !versionNumber.startsWith(majorVersion)) {
            return versionNumber;
        }
        return acc;
    }, current);
    return result === current ? null : result;
};
