// slice-machine-ui/build/lib/models/common/widgets/Image/components.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConstraintForm = exports.AddThumbnailButton = exports.ThumbnailButton = void 0;
const react_1 = require("react");
const bs_1 = require("react-icons/bs");
const md_1 = require("react-icons/md");
const theme_ui_1 = require("theme-ui");
const SliceMachineIconButton_1 = __importDefault(require("@components/SliceMachineIconButton"));
const FormFields_1 = require("@components/FormFields");
const formik_1 = require("formik");
const ThumbnailButton = ({ onClick, error, onDelete, active, text, sx, }) => (<SliceMachineIconButton_1.default useActive fitButton error={error} onClick={onClick} active={active} sx={{ position: "relative", width: "64px", height: "64px", ...sx }} Icon={() => (<react_1.Fragment>
        <theme_ui_1.Flex sx={{
            flexDirection: "column",
            justifyContent: "center",
            alignItems: "center",
        }}>
          <bs_1.BsImage size={22}/>
          <theme_ui_1.Text mt={1}>{text}</theme_ui_1.Text>
        </theme_ui_1.Flex>
        {onDelete ? (<theme_ui_1.Badge onClick={onDelete} variant="circle-right">
            <md_1.MdCancel />
          </theme_ui_1.Badge>) : null}
      </react_1.Fragment>)}/>);
exports.ThumbnailButton = ThumbnailButton;
const AddThumbnailButton = ({ onClick }) => (<SliceMachineIconButton_1.default fitButton onClick={onClick} sx={{
        width: "64px",
        height: "64px",
        border: ({ colors }) => `1px solid ${colors.borders}`,
    }} Icon={() => (<theme_ui_1.Flex sx={{
            flexDirection: "column",
            justifyContent: "center",
            alignItems: "center",
        }}>
        <bs_1.BsFillPlusSquareFill size={34} color="#5263BA"/>
      </theme_ui_1.Flex>)}/>);
exports.AddThumbnailButton = AddThumbnailButton;
const ConstraintForm = ({ prefix, required, display, error = {} }) => {
    if (!display) {
        return null;
    }
    const requiredChar = required ? "*" : "";
    const [field, meta, helpers] = formik_1.useField(prefix);
    const createSetField = (key, fn = (v) => v) => (e) => {
        helpers.setTouched(true);
        const cast = fn(e.target.value);
        const value = e.target.type === "number" ? (isNaN(cast) ? null : cast) : cast;
        helpers.setValue({ ...field.value, [key]: value });
    };
    return (<react_1.Fragment>
      <FormFields_1.FormFieldInput fieldName={`${prefix}.name`} meta={{
            ...meta,
            error: meta.error && meta.error.name,
        }} formField={{ label: `Name${requiredChar}`, placeholder: "main" }} field={prefix === "config.constraint"
            ? { value: "main", readOnly: true }
            :
                { value: field.value.name, onChange: createSetField("name") }} variant={prefix === "config.constraint" ? "disabled" : "primary"} sx={{ mb: 3 }}/>
      <FormFields_1.FormFieldInput fieldName={`${prefix}.width`} meta={{
            ...meta,
            error: meta.error && meta.error.width,
        }} formField={{ label: `Width (px)${requiredChar}`, placeholder: " " }} field={{
            type: "number",
            value: field.value.width,
            onChange: createSetField("width", parseInt),
        }} sx={{ mb: 3 }}/>
      <FormFields_1.FormFieldInput fieldName={`${prefix}.height`} meta={{
            ...meta,
            error: meta.error && meta.error.height,
        }} formField={{ label: `Height (px)${requiredChar}`, placeholder: " " }} field={{
            type: "number",
            value: field.value.height,
            onChange: createSetField("height", parseInt),
        }}/>
    </react_1.Fragment>);
};
exports.ConstraintForm = ConstraintForm;
