// slice-machine-ui/build/lib/models/common/widgets/Image/Form.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormFields = void 0;
const yup = __importStar(require("yup"));
const react_1 = require("react");
const defaults_1 = require("@lib/forms/defaults");
const forms_1 = require("@lib/forms");
const Field_1 = __importDefault(require("@lib/builders/common/EditModal/Field"));
const formik_1 = require("formik");
const theme_ui_1 = require("theme-ui");
const Flex_1 = require("components/Flex");
const components_1 = require("./components");
const FormFields = {
    label: defaults_1.DefaultFields.label,
    id: defaults_1.DefaultFields.id,
};
exports.FormFields = FormFields;
const EMPTY_THUMBNAIL = {
    name: "",
    width: "",
    height: "",
};
const thumbText = ({ width, height } = {}, allowAuto) => {
    if (allowAuto && !width && !height) {
        return "auto";
    }
    if (width || height) {
        return `${width ? width : "auto"}x${height ? height : "auto"}`;
    }
    return "...";
};
const Form = (props) => {
    const [thumbI, setThumbI] = react_1.useState(0);
    const { initialValues, values: formValues, errors, fields, touched } = props;
    const { config: { thumbnails, constraint }, } = formValues;
    react_1.useEffect(() => {
        setThumbI(thumbnails.length);
    }, [thumbnails.length]);
    return (<Flex_1.Flex>
      {Object.entries(FormFields).map(([key, field]) => (<Flex_1.Col key={key}>
          <Field_1.default fieldName={forms_1.createFieldNameFromKey(key)} formField={field} fields={fields} initialValues={initialValues}/>
        </Flex_1.Col>))}
      <Flex_1.Col>
        <theme_ui_1.Label htmlFor="thumbnails" variant="label.primary" sx={{
            mt: 1,
            display: "flex",
            alignItems: "center",
        }}>
          Responsive views*
        </theme_ui_1.Label>
        <theme_ui_1.Card p={3}>
          <formik_1.FieldArray name="config.thumbnails" render={({ push, remove }) => (<react_1.Fragment>
                <theme_ui_1.Flex mb={3}>
                  <components_1.ThumbnailButton active={thumbI === 0} text={thumbText(constraint, true)} sx={{ mr: 2 }} onClick={() => setThumbI(0)}/>
                  {thumbnails.map((e, i) => (<components_1.ThumbnailButton key={`thumbnail-button-${+i + 1}`} sx={{ mr: 3 }} active={thumbI === +i + 1} text={thumbText(e)} error={errors.thumbnails &&
                    touched.thumbnails &&
                    touched.thumbnails[i] &&
                    errors.thumbnails &&
                    errors.thumbnails[i]} onDelete={() => remove(i)} onClick={() => setThumbI(+i + 1)}/>))}
                  <components_1.AddThumbnailButton onClick={() => {
                push(EMPTY_THUMBNAIL);
            }}/>
                </theme_ui_1.Flex>
                <components_1.ConstraintForm {...props} display={thumbI === 0} prefix="config.constraint"/>
                {thumbnails.map((_, i) => (<components_1.ConstraintForm {...props} required key={`thumbnail-${+i + 1}`} display={thumbI === +i + 1} prefix={`config.thumbnails[${i}]`}/>))}
              </react_1.Fragment>)}/>
        </theme_ui_1.Card>
      </Flex_1.Col>
    </Flex_1.Flex>);
};
FormFields.constraint = {
    validate: () => yup.object().defined().shape({
        width: yup.number().nullable(),
        height: yup.number().nullable(),
    }),
};
FormFields.thumbnails = {
    validate: () => yup
        .array()
        .defined()
        .of(yup.object().test({
        name: "Thumbnails",
        message: "Must set name and width or height at minimum",
        test: function (value) {
            if (!value.name) {
                return false;
            }
            if ((!value.width && !value.height) ||
                typeof value.width !== "number" ||
                typeof value.height !== "number") {
                return false;
            }
            return true;
        },
    })),
};
exports.default = Form;
