// slice-machine-ui/build/lib/models/common/widgets/Image/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageWidget = void 0;
const yup = __importStar(require("yup"));
const Form_1 = __importStar(require("./Form"));
const bs_1 = require("react-icons/bs");
const forms_1 = require("../../../../forms");
const utils_1 = require("../../../../utils");
const Mock_1 = require("./Mock");
const Form_2 = require("./Mock/Form");
const type_1 = require("./type");
const fields_1 = require("../../CustomType/fields");
const schema = yup.object().shape({
    type: yup
        .string()
        .matches(/^Image$/, {
        excludeEmptyString: true,
    })
        .required(),
    config: forms_1.createValidationSchema(utils_1.removeProp(Form_1.FormFields, "id")),
});
const Meta = {
    icon: bs_1.BsImage,
    title: "Image",
    description: "A responsive image field with constraints",
};
exports.ImageWidget = {
    Meta,
    Form: Form_1.default,
    schema,
    create: (label) => new type_1.ImageField({ label }),
    handleMockConfig: Mock_1.handleMockConfig,
    handleMockContent: Mock_1.handleMockContent,
    MockConfigForm: Form_2.MockConfigForm,
    FormFields: Form_1.FormFields,
    TYPE_NAME: fields_1.FieldType.Image,
};
