// slice-machine-ui/build/lib/models/common/widgets/StructuredText/Mock/components.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HandleMinMax = exports.NumberOfBlocks = exports.PatternCard = void 0;
const theme_ui_1 = require("theme-ui");
const _1 = require(".");
const PatternCard = ({ patternKey, isAllowed, currentKey, onUpdate, }) => {
    return (<theme_ui_1.Label sx={{
            border: "1px solid",
            borderRadius: "3px",
            borderColor: "borders",
            display: "block",
            position: "relative",
            opacity: isAllowed ? "1" : ".4",
            cursor: isAllowed ? "pointer" : "not-allowed",
            bg: "headSection",
            mb: 3,
        }}>
      <theme_ui_1.Flex sx={{ p: 2 }}>
        <theme_ui_1.Radio name="pattern" value={patternKey} disabled={!isAllowed} onChange={() => onUpdate(patternKey, false)} checked={currentKey === patternKey}/>
        {_1.Patterns[patternKey].title}
      </theme_ui_1.Flex>
      <theme_ui_1.Box sx={{ p: 2, pt: 0, mt: 0 }}>
        <theme_ui_1.Text sx={{ as: "p", fontSize: 1 }}>
          {_1.Patterns[patternKey].description}
        </theme_ui_1.Text>
      </theme_ui_1.Box>
    </theme_ui_1.Label>);
};
exports.PatternCard = PatternCard;
const NumberOfBlocks = ({ currentValue, onUpdate }) => {
    const _onUpdate = (e) => {
        const { target: { value }, } = e;
        const val = parseInt(value);
        if (val) {
            onUpdate(val);
        }
    };
    return (<theme_ui_1.Box>
      <theme_ui_1.Label variant="label.primary">Block repetitions</theme_ui_1.Label>
      <theme_ui_1.Input sx={{ maxWidth: "440px", bg: "headSection" }} type="number" onChange={_onUpdate} value={currentValue}/>
    </theme_ui_1.Box>);
};
exports.NumberOfBlocks = NumberOfBlocks;
const PreTab = ({ content }) => (<theme_ui_1.Text as="span" sx={{
        position: "absolute",
        left: "9px",
        height: "calc(100% - 2px)",
        display: "flex",
        alignItems: "center",
        bg: "borders",
        px: 1,
        fontSize: 1,
    }}>
    {content}
  </theme_ui_1.Text>);
const HandleMinMax = ({ title, value, moreInfo, onUpdate }) => {
    const _onUpdate = (key, val) => {
        onUpdate({
            ...value,
            [key]: parseInt(val),
        });
    };
    return (<theme_ui_1.Box my={3}>
      <theme_ui_1.Text as="p" mb={2}>
        {title}
      </theme_ui_1.Text>
      <theme_ui_1.Flex sx={{ maxWidth: "440px", alignItems: "center" }}>
        <theme_ui_1.Label sx={{ display: "flex", alignItems: "center", position: "relative" }}>
          <PreTab content="Min."/>
          <theme_ui_1.Input type="number" onChange={(e) => _onUpdate("min", e.target.value)} value={value.min} sx={{
            bg: "backgroundClear",
            p: 1,
            pl: "48px",
            fontSize: 1,
            ml: 2,
        }}/>
        </theme_ui_1.Label>
        <theme_ui_1.Label sx={{
            ml: 2,
            display: "flex",
            alignItems: "center",
            position: "relative",
        }}>
          <PreTab content="Max."/>
          <theme_ui_1.Input type="number" onChange={(e) => _onUpdate("max", e.target.value)} value={value.max} sx={{
            bg: "backgroundClear",
            p: 1,
            pl: "48px",
            fontSize: 1,
            ml: 2,
        }}/>
        </theme_ui_1.Label>
      </theme_ui_1.Flex>
      <theme_ui_1.Text as="p" sx={{ mt: 2 }}>
        {moreInfo}
      </theme_ui_1.Text>
    </theme_ui_1.Box>);
};
exports.HandleMinMax = HandleMinMax;
