// slice-machine-ui/build/lib/models/common/widgets/StructuredText/Mock/Form.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MockConfigForm = void 0;
const react_1 = require("react");
const theme_ui_1 = require("theme-ui");
const fa_1 = require("react-icons/fa");
const formik_1 = require("formik");
const _1 = require(".");
const components_1 = require("./components");
const Tooltip_1 = __importDefault(require("components/Tooltip"));
const Flex_1 = require("components/Flex");
const consts_1 = require("../../../../../consts");
const dataTip = `To generate mock content, we'll use the selected pattern.<br/>A pattern is an array of RichText options, repeated "block" times.`;
const HandlePatternTypes = ({ options, currentKey, onUpdate, onUpdateBlocks, blocksValue, }) => {
    const { theme } = theme_ui_1.useThemeUI();
    const PatternsWithStatus = Object.entries(_1.Patterns).map(([key, pattern]) => ({
        patternKey: key,
        isAllowed: _1.Patterns[key].test(options),
        pattern,
    }));
    return (<theme_ui_1.Box>
      <Tooltip_1.default id="richtext-mock"/>
      <Flex_1.Flex mt={3}>
        <Flex_1.Col>
          <theme_ui_1.Label variant="label.primary">
            Mock Pattern
            <fa_1.FaRegQuestionCircle data-for="richtext-mock" data-multiline="true" data-tip={dataTip} color={theme.colors.icons} style={{
            position: "relative",
            top: "3px",
            height: "18px",
            marginLeft: "8px",
        }}/>
          </theme_ui_1.Label>
          {PatternsWithStatus.map(({ patternKey, ...rest }) => (<components_1.PatternCard key={patternKey} patternKey={patternKey} onUpdate={onUpdate} currentKey={currentKey} {...rest}/>))}
        </Flex_1.Col>
        <Flex_1.Col>
          <components_1.NumberOfBlocks currentValue={blocksValue} onUpdate={onUpdateBlocks}/>
        </Flex_1.Col>
      </Flex_1.Flex>
    </theme_ui_1.Box>);
};
const Form = () => {
    var _a;
    const [patternTypeCheck, setPatternTypeCheck] = react_1.useState(null);
    const { values, setFieldValue } = formik_1.useFormikContext();
    const options = (values.config.single || values.config.multi || "").split(",");
    const configValues = ((_a = values[consts_1.MockConfigKey]) === null || _a === void 0 ? void 0 : _a.config) || {};
    react_1.useEffect(() => {
        const { mockConfig: { config: { patternType }, }, } = values;
        if (patternType &&
            patternType !== patternTypeCheck &&
            !_1.Patterns[patternType].test(options)) {
            onUpdate({
                key: "patternType",
                updateType: "config",
                value: findValidPattern(options),
            });
            setPatternTypeCheck(patternType);
        }
    });
    const onUpdate = ({ updateType, key, value }) => {
        var _a;
        setFieldValue(consts_1.MockConfigKey, {
            [updateType]: {
                ...(((_a = values[consts_1.MockConfigKey]) === null || _a === void 0 ? void 0 : _a[updateType]) || {}),
                [key]: value,
            },
        });
    };
    const onSetPattern = (value, isPattern) => {
        onUpdate({
            key: isPattern ? "pattern" : "patternType",
            updateType: "config",
            value,
        });
    };
    const onSetBlocks = (value) => {
        onUpdate({ key: "blocks", value, updateType: "config" });
    };
    return (<theme_ui_1.Box>
      <HandlePatternTypes options={options} currentKey={configValues.patternType || _1.DEFAULT_PATTERN_KEY} onUpdate={onSetPattern} onUpdateBlocks={onSetBlocks} blocksValue={configValues.blocks || 1} currentValue={_1.Patterns[configValues.patternType || _1.DEFAULT_PATTERN_KEY].value(options)}/>
    </theme_ui_1.Box>);
};
const findValidPattern = (config) => {
    const patternEntry = Object.entries(_1.Patterns).find(([, pat]) => pat.test(config));
    if (patternEntry) {
        return patternEntry[0];
    }
    return _1.DEFAULT_PATTERN_KEY;
};
Form.onSave = (mockValue, values) => {
    var _a;
    if (!((_a = mockValue === null || mockValue === void 0 ? void 0 : mockValue.config) === null || _a === void 0 ? void 0 : _a.patternType)) {
        return mockValue;
    }
    const { patternType } = mockValue.config;
    const patternObj = _1.Patterns[patternType];
    if (!patternObj) {
        return _1.initialValues;
    }
    const options = (values.config.single || values.config.multi).split(",");
    const isValidPatternType = patternObj.test(options);
    if (isValidPatternType) {
        return mockValue;
    }
    return {
        ...mockValue,
        config: {
            ...mockValue.config,
            patternType: findValidPattern(options),
        },
    };
};
Form.initialValues = _1.initialValues;
exports.MockConfigForm = Form;
