// slice-machine-ui/build/lib/models/common/widgets/StructuredText/Mock/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.initialValues = exports.handleMockContent = exports.handleMockConfig = exports.Patterns = exports.LoremDefaultConfig = exports.DEFAULT_PATTERN_KEY = void 0;
const faker_1 = __importDefault(require("@faker-js/faker"));
const lorem_ipsum_1 = require("lorem-ipsum");
const Mock_1 = require("../../Image/Mock");
const isHeading = (type) => type.indexOf("heading") === 0;
exports.DEFAULT_PATTERN_KEY = "PARAGRAPH";
exports.LoremDefaultConfig = {
    sentencesPerParagraph: {
        min: 1,
        max: 3,
    },
    wordsPerSentence: {
        min: 4,
        max: 16,
    },
};
const optionalType = (options, type) => options.find((e) => e === type) && Math.random() > 0.5 ? [type] : [];
exports.Patterns = {
    PARAGRAPH: {
        title: "Simple Paragraph",
        test: (options) => options.some((e) => e === "paragraph"),
        value: () => ["paragraph"],
        description: "A single paragraph with a variant number of words.",
    },
    HEADING: {
        title: "Section Title",
        test: (options) => options.some(isHeading),
        value: (options) => [options.find(isHeading) || "heading1"],
        description: "A single heading (h1 to h6) with a variant number of words.",
    },
    STORY: {
        title: "Story",
        test: (options) => options.some(isHeading) && options.some((e) => e === "paragraph"),
        value: (options) => [
            options.find(isHeading) || "heading1",
            ...optionalType(options, "image"),
            "paragraph",
        ],
        description: "Content with headings, texts and optional images",
    },
};
const findMatchingPattern = (options) => {
    const PatternEntry = Object.entries(exports.Patterns).find(([key, patt]) => patt.test(options));
    if (PatternEntry && PatternEntry.length) {
        return PatternEntry[1];
    }
    return exports.Patterns[exports.DEFAULT_PATTERN_KEY];
};
const isTextType = (type) => type === "paragraph" || isHeading(type);
const handleText = (contentType, loremConfig) => {
    const lorem = new lorem_ipsum_1.LoremIpsum(loremConfig);
    if (isHeading(contentType)) {
        const fake = faker_1.default.company.bs();
        return [fake[0].toUpperCase(), ...fake.slice(1)].join("");
    }
    return lorem.generateParagraphs(1);
};
const createMockFromConfig = (blocksLen, pattern, loremConfig) => {
    const content = Array(blocksLen)
        .fill()
        .map(() => {
        return pattern.map((contentType) => {
            if (isTextType(contentType)) {
                return {
                    type: contentType,
                    text: handleText(contentType, loremConfig),
                    spans: [],
                };
            }
            if (contentType === "image") {
                const res = Mock_1.handleMockConfig(null, {});
                return {
                    type: contentType,
                    ...res,
                };
            }
        });
    });
    return content.flat();
};
const handleMockConfig = (mockConfigValues, fieldConfig) => {
    const options = (fieldConfig.multi || fieldConfig.single).split(",");
    const patternType = mockConfigValues ? mockConfigValues.patternType : null;
    const patternObj = exports.Patterns[patternType] || findMatchingPattern(options);
    const pattern = patternObj.value(options);
    return createMockFromConfig(mockConfigValues.blocks, pattern, exports.LoremDefaultConfig);
};
exports.handleMockConfig = handleMockConfig;
const handleMockContent = (mockContent, fieldConfig) => {
    if (Array.isArray(mockContent) &&
        mockContent.length &&
        typeof mockContent[0] === "object") {
        return mockContent;
    }
    const options = (fieldConfig.single || fieldConfig.multi).split(",");
    return [
        {
            type: options.find(isHeading) || "paragraph",
            spans: [],
            text: "...",
            ...(typeof mockContent === "object"
                ? mockContent
                :
                    { text: mockContent }),
        },
    ];
};
exports.handleMockContent = handleMockContent;
exports.initialValues = {
    config: {
        patternType: exports.DEFAULT_PATTERN_KEY,
        blocks: 1,
    },
};
