// @slicemachine/core/build/src/utils/cliLogs.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.writeInfo = exports.writeCheck = exports.writeWarning = exports.writeError = exports.spinner = exports.white = exports.yellow = exports.cyan = exports.dim = exports.bold = exports.underline = exports.warning = exports.error = exports.purple = void 0;
const chalk_1 = __importDefault(require("chalk"));
const ora_1 = __importDefault(require("ora"));
exports.purple = chalk_1.default.hex("8c76fc");
exports.error = chalk_1.default.hex("FF6868");
exports.warning = chalk_1.default.hex("F3AD38");
exports.underline = chalk_1.default.underline;
exports.bold = chalk_1.default.bold;
exports.dim = chalk_1.default.dim;
exports.cyan = chalk_1.default.cyan;
exports.yellow = chalk_1.default.yellow;
exports.white = chalk_1.default.white;
exports.spinner = ora_1.default;
const writeError = (msg, prefix = "Error!") => console.error(`${(0, exports.error)(prefix)} ${msg}`);
exports.writeError = writeError;
const writeWarning = (msg, prefix = "Warning!") => console.warn(`${(0, exports.warning)(prefix)} ${msg}`);
exports.writeWarning = writeWarning;
const writeCheck = (msg) => {
    (0, ora_1.default)().succeed(msg);
};
exports.writeCheck = writeCheck;
const writeInfo = (msg) => {
    console.log(`${(0, exports.yellow)("ℹ")} ${(0, exports.dim)(msg)}`);
};
exports.writeInfo = writeInfo;
