// @slicemachine/core/build/src/utils/const.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ACCEPTED_IMAGE_TYPES = exports.USER_SERVICE_STAGING_BASE = exports.USER_SERVICE_BASE = exports.DEFAULT_BASE = exports.DEFAULT_SERVER_PORT = exports.MANIFEST_FILE_NAME = exports.SLICE_SIMULATOR_VUE = exports.SLICE_SIMULATOR_REACT = exports.PRISMIC_VUE = exports.VUE_SLICEZONE = exports.NUXT_SM = exports.NUXT_PRISMIC = exports.NEXT_SLICEZONE = exports.PRISMIC_HELPERS = exports.PRISMIC_CLIENT = exports.PRISMIC_DOM_PACKAGE_NAME = exports.PRISMIC_REACT_PACKAGE_NAME = exports.PREVIOUS_REACT_PACKAGE_NAME = exports.SM_PACKAGE_NAME = exports.INIT_COMMAND = exports.SCRIPT_VALUE = exports.SCRIPT_NAME = void 0;
exports.SCRIPT_NAME = "slicemachine";
exports.SCRIPT_VALUE = "start-slicemachine";
exports.INIT_COMMAND = "npx sm init";
exports.SM_PACKAGE_NAME = "slice-machine-ui";
exports.PREVIOUS_REACT_PACKAGE_NAME = "prismic-reactjs";
exports.PRISMIC_REACT_PACKAGE_NAME = "@prismicio/react";
exports.PRISMIC_DOM_PACKAGE_NAME = "prismic-dom";
exports.PRISMIC_CLIENT = "@prismicio/client";
exports.PRISMIC_HELPERS = "@prismicio/helpers";
exports.NEXT_SLICEZONE = "next-slicezone";
exports.NUXT_PRISMIC = "@nuxtjs/prismic";
exports.NUXT_SM = "nuxt-sm";
exports.VUE_SLICEZONE = "vue-slicezone";
exports.PRISMIC_VUE = "@prismicio/vue";
exports.SLICE_SIMULATOR_REACT = "@prismicio/slice-simulator-react";
exports.SLICE_SIMULATOR_VUE = "@prismicio/slice-simulator-vue";
exports.MANIFEST_FILE_NAME = "sm.json";
exports.DEFAULT_SERVER_PORT = 5555;
exports.DEFAULT_BASE = "https://prismic.io";
exports.USER_SERVICE_BASE = "https://user.internal-prismic.io";
exports.USER_SERVICE_STAGING_BASE = "https://user.wroom.io";
exports.ACCEPTED_IMAGE_TYPES = ["png", "jpg", "jpeg"];
