// @slicemachine/core/build/src/utils/framework.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defineFramework = exports.detectFramework = exports.fancyName = exports.isFrameworkSupported = exports.UnsupportedFrameWorks = exports.Frameworks = void 0;
const filesystem_1 = require("../filesystem");
const Framework_1 = require("../models/Framework");
var Framework_2 = require("../models/Framework");
Object.defineProperty(exports, "Frameworks", { enumerable: true, get: function () { return Framework_2.Frameworks; } });
exports.UnsupportedFrameWorks = Object.values(Framework_1.Frameworks).filter((framework) => !Framework_1.SupportedFrameworks.includes(framework));
function isFrameworkSupported(framework) {
    return Framework_1.SupportedFrameworks.includes(framework);
}
exports.isFrameworkSupported = isFrameworkSupported;
function capitaliseFirstLetter(str) {
    return str.charAt(0).toUpperCase() + str.slice(1);
}
function fancyName(str) {
    switch (str) {
        case Framework_1.Frameworks.next:
            return "Next.js";
        default:
            return capitaliseFirstLetter(str);
    }
}
exports.fancyName = fancyName;
function detectFramework(pkg, supportedFrameworks = Framework_1.SupportedFrameworks) {
    const { dependencies, devDependencies, peerDependencies } = pkg;
    const deps = { ...peerDependencies, ...devDependencies, ...dependencies };
    const frameworkEntry = Object.values(supportedFrameworks).find((f) => deps[f] && deps[f].length);
    return frameworkEntry || Framework_1.Frameworks.vanillajs;
}
exports.detectFramework = detectFramework;
function defineFramework({ cwd, supportedFrameworks = Framework_1.SupportedFrameworks, manifest, }) {
    if ((manifest === null || manifest === void 0 ? void 0 : manifest.framework) && isFrameworkSupported(manifest.framework))
        return manifest.framework;
    const pkg = (0, filesystem_1.retrieveJsonPackage)(cwd);
    if (!pkg.exists || !pkg.content) {
        const message = "[api/env]: Unrecoverable error. Could not find package.json. Exiting..";
        console.error(message);
        throw new Error(message);
    }
    return detectFramework(pkg.content, supportedFrameworks);
}
exports.defineFramework = defineFramework;
