// @slicemachine/core/build/src/utils/lib.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getInfoFromPath = exports.getFormattedLibIdentifier = exports.findIndexFile = void 0;
const path_1 = __importDefault(require("path"));
const files_1 = __importDefault(require("./files"));
var Prefix;
(function (Prefix) {
    Prefix["A"] = "@/";
    Prefix["B"] = "~/";
    Prefix["C"] = "/";
})(Prefix || (Prefix = {}));
const Identifiers = {
    "@/": 2,
    "~/": 2,
    "/": 1,
};
const findIndexFile = (libPath) => {
    try {
        const dir = files_1.default.readDirectory(libPath);
        const maybeF = dir.find((f) => files_1.default.isFile(path_1.default.join(libPath, f)) && f.startsWith("index."));
        return maybeF ? path_1.default.join(libPath, maybeF) : null;
    }
    catch (e) {
        return null;
    }
};
exports.findIndexFile = findIndexFile;
const getFormattedLibIdentifier = (libPath) => {
    const maybeIdentifier = Object.keys(Identifiers).find((e) => libPath.indexOf(e) === 0);
    const isLocal = !!maybeIdentifier;
    return {
        isLocal,
        identifier: maybeIdentifier,
        from: isLocal
            ? libPath.slice(Identifiers[maybeIdentifier])
            : libPath,
    };
};
exports.getFormattedLibIdentifier = getFormattedLibIdentifier;
function getInfoFromPath(startPath, libPath) {
    const { isLocal, from } = (0, exports.getFormattedLibIdentifier)(libPath);
    const pathToLib = path_1.default.join(startPath || process.cwd(), isLocal ? "" : "node_modules", isLocal ? libPath.substring(1, libPath.length) : libPath);
    const pathExists = files_1.default.exists(pathToLib);
    const pathToSlices = path_1.default.join(pathToLib, isLocal ? "." : "slices");
    return {
        isLocal,
        from,
        pathExists,
        pathToLib,
        pathToSlices,
    };
}
exports.getInfoFromPath = getInfoFromPath;
