// @slicemachine/core/build/src/utils/roles.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.canUpdateCustomTypes = exports.RolesValidator = exports.RolePerLocaleProfile = exports.Roles = void 0;
const t = __importStar(require("io-ts"));
var Roles;
(function (Roles) {
    Roles["WRITER"] = "Writer";
    Roles["OWNER"] = "Owner";
    Roles["PUBLISHER"] = "Publisher";
    Roles["ADMIN"] = "Administrator";
    Roles["SUPERUSER"] = "SuperUser";
    Roles["MANAGER"] = "Manager";
    Roles["READONLY"] = "Readonly";
})(Roles = exports.Roles || (exports.Roles = {}));
const RolesV = t.union([
    t.literal(Roles.WRITER),
    t.literal(Roles.OWNER),
    t.literal(Roles.PUBLISHER),
    t.literal(Roles.ADMIN),
    t.literal(Roles.SUPERUSER),
    t.literal(Roles.MANAGER),
    t.literal(Roles.READONLY),
]);
exports.RolePerLocaleProfile = t.record(t.string, RolesV);
exports.RolesValidator = t.union([RolesV, exports.RolePerLocaleProfile]);
function canUpdateCustomTypes(role) {
    if (role === Roles.OWNER)
        return true;
    if (role === Roles.ADMIN)
        return true;
    if (role === Roles.SUPERUSER)
        return true;
    return false;
}
exports.canUpdateCustomTypes = canUpdateCustomTypes;
