// slice-machine-ui/build/lib/builders/common/Zone/Card/components/NewField.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("react");
const formik_1 = require("formik");
const theme_ui_1 = require("theme-ui");
const defaults_1 = require("@lib/forms/defaults");
const forms_1 = require("@lib/forms");
const Widgets = __importStar(require("@lib/models/common/widgets/withGroup"));
const fields_1 = require("@lib/models/common/CustomType/fields");
const ErrorTooltip_1 = __importDefault(require("./ErrorTooltip"));
const RefInput = (args) => {
    return <theme_ui_1.Input {...args} ref={args.innerRef}/>;
};
const NewField = ({ widgetTypeName, fields, onSave, onCancelNewField }) => {
    const fieldRef = react_1.useRef(null);
    const { theme } = theme_ui_1.useThemeUI();
    const widget = Widgets[widgetTypeName];
    if (!widget) {
        console.error(`Widget of type "${widgetTypeName}" not found. This is a problem on our side!`);
        return <div>Unexpected error. Contact us for more info.</div>;
    }
    const { Meta: { icon: WidgetIcon }, FormFields: widgetFormFields, } = widget;
    const FormFields = {
        id: widgetFormFields.id,
    };
    const initialValues = {
        ...forms_1.createInitialValues(FormFields),
        ...(widgetTypeName === fields_1.FieldType.UID ? { id: "uid" } : {}),
        widgetTypeName,
    };
    const validationSchema = forms_1.createValidationSchema(FormFields);
    react_1.useEffect(() => {
        if (fieldRef.current) {
            fieldRef.current.focus();
        }
    }, [fieldRef]);
    return (<formik_1.Formik validateOnChange onSubmit={onSave} validationSchema={validationSchema} initialValues={initialValues}>
      {({ errors }) => (<formik_1.Form>
          <theme_ui_1.Flex as="li" sx={{
                p: 3,
                px: 3,
                mx: 0,
                ml: "34px",
                alignItems: "center",
                variant: "styles.listItem",
                border: (t) => { var _a; return `1px solid ${(_a = t.colors) === null || _a === void 0 ? void 0 : _a.borders}`; },
            }}>
            <theme_ui_1.Flex sx={{
                alignItems: "center",
            }}>
              <WidgetIcon style={{
                color: theme.colors.primary,
                marginRight: "8px",
                borderRadius: "4px",
                padding: "4px",
                border: "2px solid",
                borderColor: theme.colors.primary,
            }} size={28}/>
              <theme_ui_1.Flex sx={{
                display: "flex",
                alignItems: "center",
            }}>
                <theme_ui_1.Text as="p" sx={{ mr: 3, minWidth: "56px" }}>
                  field id
                </theme_ui_1.Text>
                <formik_1.Field name="id" placeholder="myField" type="text" validate={(value) => defaults_1.validateId({
                value,
                fields,
                initialId: null,
            })} as={RefInput} innerRef={fieldRef} sx={{
                border: ({ colors }) => errors.id
                    ? `1px solid tomato`
                    :
                        `1px solid ${colors.primary}`,
                "&:focus": {
                    border: errors.id
                        ? `1px solid tomato`
                        : "1px solid yellow",
                },
            }}/>
                <ErrorTooltip_1.default errors={errors}/>
              </theme_ui_1.Flex>
            </theme_ui_1.Flex>
            <theme_ui_1.Box>
              <theme_ui_1.Button onClick={onCancelNewField} variant="secondary" type="button">
                Cancel
              </theme_ui_1.Button>
              <theme_ui_1.Button sx={{ ml: 2 }} type="submit">
                Add
              </theme_ui_1.Button>
            </theme_ui_1.Box>
          </theme_ui_1.Flex>
        </formik_1.Form>)}
    </formik_1.Formik>);
};
exports.default = NewField;
