// slice-machine-ui/build/lib/forms/fields.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Input = exports.Select = exports.CheckBox = void 0;
const types_1 = require("./types");
const _1 = require(".");
const CheckBox = (label, required = true, defaultValue = true) => ({
    type: types_1.FormTypes.CHECKBOX,
    validate: {
        required: _1.createValidationArgs(required, defaultRequired),
    },
    required,
    label,
    defaultValue,
    yupType: "boolean",
});
exports.CheckBox = CheckBox;
const Select = (label, options, required = true, multi = true, defaultValue = true) => ({
    type: types_1.FormTypes.SELECT,
    required,
    label,
    options,
    multi,
    defaultValue,
    yupType: "array",
});
exports.Select = Select;
const defaultMin = [3, "String is too short. Min: 3"];
const defaultMax = [35, "String is too long. Max: 35"];
const defaultRequired = ["Field is required"];
const Input = (label, conditions = {
    min: defaultMin,
    max: defaultMax,
    required: defaultRequired,
}, fieldLevelValidation, defaultValue) => {
    const { min, max, required, matches } = conditions || {};
    return {
        type: types_1.FormTypes.INPUT,
        label,
        validate: {
            required: _1.createValidationArgs(required, defaultRequired),
            min: _1.createValidationArgs(min, defaultMin),
            max: _1.createValidationArgs(max, defaultMax),
            matches,
        },
        defaultValue,
        fieldLevelValidation,
        yupType: "string",
    };
};
exports.Input = Input;
