// @slicemachine/core/build/src/libraries/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.libraries = exports.handleLibraryPath = void 0;
const fs_1 = __importDefault(require("fs"));
const path_1 = __importDefault(require("path"));
const files_1 = __importDefault(require("../utils/files"));
const lib_1 = require("../utils/lib");
const component_1 = require("./component");
const Library_1 = require("../models/Library");
function handleLibraryPath(cwd, libPath) {
    const { from, isLocal, pathExists, pathToSlices, pathToLib } = (0, lib_1.getInfoFromPath)(cwd, libPath);
    if (!pathExists) {
        return;
    }
    const pathsToComponents = files_1.default.readDirectory(pathToSlices.split(path_1.default.sep).join(path_1.default.posix.sep))
        .map((curr) => path_1.default.join(pathToSlices, curr))
        .filter((e) => {
        const f = e.split(path_1.default.sep).pop();
        return fs_1.default.lstatSync(e).isDirectory() && !(f === null || f === void 0 ? void 0 : f.startsWith("."));
    });
    const pathToSlice = `${isLocal ? "./" : ""}${from}${pathToSlices
        .split(from)
        .slice(1)
        .join("")}`;
    const allComponents = pathsToComponents.reduce((acc, curr) => {
        const componentInfo = (0, component_1.getComponentInfo)(curr, [cwd, path_1.default.join(cwd, ".slicemachine/assets")], from);
        if (!componentInfo) {
            return acc;
        }
        const { model } = componentInfo;
        return [
            ...acc,
            {
                from,
                href: from.replace(/\//g, "--"),
                pathToSlice,
                infos: componentInfo,
                model,
                migrated: false,
            },
        ];
    }, []);
    const meta = Library_1.LibraryMeta.build(pathToLib);
    return {
        path: pathToLib,
        isLocal,
        name: from,
        components: allComponents,
        meta,
    };
}
exports.handleLibraryPath = handleLibraryPath;
function libraries(cwd, libraries) {
    return (libraries || [])
        .map((lib) => handleLibraryPath(cwd, lib))
        .filter(Boolean);
}
exports.libraries = libraries;
__exportStar(require("./screenshot"), exports);
