// @slicemachine/core/build/src/libraries/screenshot.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.resolvePathsToScreenshot = exports.generatePathsToScreenshot = exports.createPathToScreenshot = exports.Extensions = void 0;
const path_1 = __importDefault(require("path"));
const files_1 = __importDefault(require("../utils/files"));
var Extensions;
(function (Extensions) {
    Extensions["jpeg"] = "jpeg";
    Extensions["jpg"] = "jpg";
    Extensions["png"] = "png";
})(Extensions = exports.Extensions || (exports.Extensions = {}));
function createPathToScreenshot({ path: filePath, from, sliceName, variationId, extension, }) {
    return path_1.default.join(filePath, from, sliceName, variationId, `preview.${extension}`);
}
exports.createPathToScreenshot = createPathToScreenshot;
function generatePathsToScreenshot({ base, from, sliceName, variationId, }) {
    return Object.values(Extensions).map((imageType) => {
        return path_1.default.join(base, from, sliceName, variationId, `preview.${imageType}`);
    });
}
exports.generatePathsToScreenshot = generatePathsToScreenshot;
function resolvePathsToScreenshot({ paths, from, sliceName, variationId, }) {
    const possiblePaths = paths
        .map((base) => generatePathsToScreenshot({
        base,
        from,
        sliceName,
        variationId,
    }))
        .flat();
    const screenshot = files_1.default.readFirstOf(possiblePaths)((v) => v);
    if (!screenshot)
        return screenshot;
    return {
        path: screenshot.path,
    };
}
exports.resolvePathsToScreenshot = resolvePathsToScreenshot;
