// slice-machine-ui/build/lib/models/common/widgets/Group/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroupWidget = void 0;
const yup = __importStar(require("yup"));
const defaults_1 = require("@lib/forms/defaults");
const md_1 = require("react-icons/md");
const fields_1 = require("../../CustomType/fields");
const ListItem_1 = __importDefault(require("./ListItem"));
const type_1 = require("./type");
const Meta = {
    icon: md_1.MdPlaylistAdd,
    title: "Group",
    description: "A Group of Prismic widgets",
};
const schema = yup.object().shape({
    type: yup
        .string()
        .matches(/^Group$/, { excludeEmptyString: true })
        .required(),
    config: yup.object().shape({
        fields: yup.array(),
        label: yup.string(),
        placeholder: yup.string(),
    }),
});
exports.GroupWidget = {
    Meta,
    FormFields: defaults_1.DefaultFields,
    schema,
    create: (label) => new type_1.GroupField({ label }),
    CustomListItem: ListItem_1.default,
    TYPE_NAME: fields_1.FieldType.Group,
};
