// slice-machine-ui/build/lib/models/common/widgets/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkToMedia = exports.UID = exports.Timestamp = exports.Text = exports.StructuredText = exports.Select = exports.Number = exports.Link = exports.Image = exports.GeoPoint = exports.Embed = exports.Date = exports.ContentRelationship = exports.Color = exports.Boolean = void 0;
const Boolean_1 = require("./Boolean");
Object.defineProperty(exports, "Boolean", { enumerable: true, get: function () { return Boolean_1.BooleanWidget; } });
const Color_1 = require("./Color");
Object.defineProperty(exports, "Color", { enumerable: true, get: function () { return Color_1.ColorWidget; } });
const ContentRelationship_1 = require("./ContentRelationship");
Object.defineProperty(exports, "ContentRelationship", { enumerable: true, get: function () { return ContentRelationship_1.ContentRelationshipWidget; } });
const Date_1 = require("./Date");
Object.defineProperty(exports, "Date", { enumerable: true, get: function () { return Date_1.DateWidget; } });
const Embed_1 = require("./Embed");
Object.defineProperty(exports, "Embed", { enumerable: true, get: function () { return Embed_1.EmbedWidget; } });
const GeoPoint_1 = require("./GeoPoint");
Object.defineProperty(exports, "GeoPoint", { enumerable: true, get: function () { return GeoPoint_1.GeoPointWidget; } });
const Image_1 = require("./Image");
Object.defineProperty(exports, "Image", { enumerable: true, get: function () { return Image_1.ImageWidget; } });
const Link_1 = require("./Link");
Object.defineProperty(exports, "Link", { enumerable: true, get: function () { return Link_1.LinkWidget; } });
const LinkToMedia_1 = require("./LinkToMedia");
Object.defineProperty(exports, "LinkToMedia", { enumerable: true, get: function () { return LinkToMedia_1.LinkToMediaWidget; } });
const Number_1 = require("./Number");
Object.defineProperty(exports, "Number", { enumerable: true, get: function () { return Number_1.NumberWidget; } });
const Select_1 = require("./Select");
Object.defineProperty(exports, "Select", { enumerable: true, get: function () { return Select_1.SelectWidget; } });
const StructuredText_1 = require("./StructuredText");
Object.defineProperty(exports, "StructuredText", { enumerable: true, get: function () { return StructuredText_1.StructuredTextWidget; } });
const Text_1 = require("./Text");
Object.defineProperty(exports, "Text", { enumerable: true, get: function () { return Text_1.TextWidget; } });
const Timestamp_1 = require("./Timestamp");
Object.defineProperty(exports, "Timestamp", { enumerable: true, get: function () { return Timestamp_1.TimestampWidget; } });
const UID_1 = require("./UID");
Object.defineProperty(exports, "UID", { enumerable: true, get: function () { return UID_1.UIDWidget; } });
