// slice-machine-ui/build/lib/models/common/widgets/withGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SliceZone = exports.SharedSlice = exports.Group = exports.UID = exports.Timestamp = exports.Text = exports.StructuredText = exports.Select = exports.Number = exports.LinkToMedia = exports.Link = exports.Image = exports.GeoPoint = exports.Embed = exports.Date = exports.ContentRelationship = exports.Color = exports.Boolean = void 0;
var _1 = require(".");
Object.defineProperty(exports, "Boolean", { enumerable: true, get: function () { return _1.Boolean; } });
var _2 = require(".");
Object.defineProperty(exports, "Color", { enumerable: true, get: function () { return _2.Color; } });
var _3 = require(".");
Object.defineProperty(exports, "ContentRelationship", { enumerable: true, get: function () { return _3.ContentRelationship; } });
var _4 = require(".");
Object.defineProperty(exports, "Date", { enumerable: true, get: function () { return _4.Date; } });
var _5 = require(".");
Object.defineProperty(exports, "Embed", { enumerable: true, get: function () { return _5.Embed; } });
var _6 = require(".");
Object.defineProperty(exports, "GeoPoint", { enumerable: true, get: function () { return _6.GeoPoint; } });
var _7 = require(".");
Object.defineProperty(exports, "Image", { enumerable: true, get: function () { return _7.Image; } });
var _8 = require(".");
Object.defineProperty(exports, "Link", { enumerable: true, get: function () { return _8.Link; } });
var _9 = require(".");
Object.defineProperty(exports, "LinkToMedia", { enumerable: true, get: function () { return _9.LinkToMedia; } });
var _10 = require(".");
Object.defineProperty(exports, "Number", { enumerable: true, get: function () { return _10.Number; } });
var _11 = require(".");
Object.defineProperty(exports, "Select", { enumerable: true, get: function () { return _11.Select; } });
var _12 = require(".");
Object.defineProperty(exports, "StructuredText", { enumerable: true, get: function () { return _12.StructuredText; } });
var _13 = require(".");
Object.defineProperty(exports, "Text", { enumerable: true, get: function () { return _13.Text; } });
var _14 = require(".");
Object.defineProperty(exports, "Timestamp", { enumerable: true, get: function () { return _14.Timestamp; } });
var _15 = require(".");
Object.defineProperty(exports, "UID", { enumerable: true, get: function () { return _15.UID; } });
var Group_1 = require("./Group");
Object.defineProperty(exports, "Group", { enumerable: true, get: function () { return Group_1.GroupWidget; } });
exports.SharedSlice = null;
exports.SliceZone = null;
