// @slicemachine/core/build/node-utils/files.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const fs_1 = __importDefault(require("fs"));
const path_1 = __importDefault(require("path"));
const ERROR_CODES = { ENOENT: "ENOENT" };
const _format = "utf8";
function write(pathToFile, value, options = { recursive: true }) {
    if (options.recursive) {
        const directoryPath = path_1.default.dirname(pathToFile);
        mkdir(directoryPath, { recursive: true });
    }
    if (typeof value === "string")
        fs_1.default.writeFileSync(pathToFile, value, _format);
    else
        fs_1.default.writeFileSync(pathToFile, JSON.stringify(value, null, 2), _format);
}
function readBuffer(pathToFile) {
    return fs_1.default.readFileSync(pathToFile);
}
function readString(pathToFile) {
    return fs_1.default.readFileSync(pathToFile, { encoding: _format });
}
function readEntity(pathToFile, validate) {
    const entity = safeReadJson(pathToFile);
    if (entity) {
        return validate(entity);
    }
    return new Error(`Could not parse file "${path_1.default.basename(pathToFile)}"`);
}
function safeReadEntity(pathToFile, validate) {
    try {
        const result = readEntity(pathToFile, validate);
        if (result instanceof Error)
            return null;
        return result;
    }
    catch (e) {
        return null;
    }
}
function readJson(pathToFile) {
    return JSON.parse(readString(pathToFile));
}
function safeReadJson(pathToFile) {
    try {
        return JSON.parse(readString(pathToFile));
    }
    catch (e) {
        return null;
    }
}
function readFirstOf(filePaths) {
    return (converter) => {
        return filePaths.reduce((acc, filePath) => {
            if (acc)
                return acc;
            else {
                const pathWithOpts = typeof filePath === "string" ? { path: filePath } : filePath;
                if (exists(pathWithOpts.path)) {
                    const optsOrDefault = pathWithOpts.options || {};
                    const test = {
                        path: pathWithOpts.path,
                        ...optsOrDefault,
                        value: converter(readString(pathWithOpts.path)),
                    };
                    return test;
                }
                else
                    return acc;
            }
        }, undefined);
    };
}
function isDirectory(source) {
    try {
        return fs_1.default.lstatSync(source).isDirectory();
    }
    catch (e) {
        return false;
    }
}
function isFile(source) {
    return fs_1.default.lstatSync(source).isFile();
}
function readDirectory(source) {
    return fs_1.default.readdirSync(source, { encoding: _format });
}
function mkdir(target, options) {
    return fs_1.default.mkdirSync(target, options);
}
function exists(pathToFile) {
    try {
        return Boolean(fs_1.default.lstatSync(pathToFile));
    }
    catch (e) {
        if (e.code === ERROR_CODES.ENOENT)
            return false;
        throw e;
    }
}
function append(filePath, data) {
    fs_1.default.appendFileSync(filePath, data, { encoding: _format });
}
function copy(src, dest, options = { recursive: false }) {
    if (options.recursive) {
        const directoryPath = path_1.default.dirname(dest);
        mkdir(directoryPath, { recursive: true });
    }
    fs_1.default.copyFileSync(src, dest);
}
function remove(src) {
    fs_1.default.unlinkSync(src);
}
function removeAll(srcs) {
    srcs.forEach((src) => remove(src));
}
function flushDirectories(directory, recursive = true) {
    try {
        const paths = fs_1.default.readdirSync(directory);
        paths.forEach((p) => {
            const maybedir = path_1.default.join(directory, p);
            if (isDirectory(maybedir)) {
                fs_1.default.rmdirSync(maybedir, { recursive });
            }
        });
    }
    catch (e) { }
}
exports.default = {
    write,
    readBuffer,
    readString,
    readEntity,
    safeReadEntity,
    readJson,
    safeReadJson,
    readFirstOf,
    isDirectory,
    isFile,
    readDirectory,
    mkdir,
    exists,
    append,
    copy,
    remove,
    removeAll,
    flushDirectories,
};
