// @slicemachine/core/build/node-utils/manifest.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateManifestSMVersion = exports.patchManifest = exports.maybeRepoNameFromSMFile = exports.retrieveManifest = exports.createManifest = void 0;
const files_1 = __importDefault(require("./files"));
const paths_1 = require("./paths");
const Manifest_1 = require("../models/Manifest");
const io_ts_reporters_1 = require("io-ts-reporters");
const function_1 = require("fp-ts/function");
const Either_1 = require("fp-ts/Either");
function createManifest(cwd, manifest) {
    const manifestPath = (0, paths_1.SMConfigPath)(cwd);
    files_1.default.write(manifestPath, manifest, { recursive: false });
}
exports.createManifest = createManifest;
function retrieveManifest(cwd, validate = false) {
    const manifestPath = (0, paths_1.SMConfigPath)(cwd);
    if (!files_1.default.exists(manifestPath)) {
        return {
            exists: false,
            content: null,
        };
    }
    const content = files_1.default.safeReadJson(manifestPath);
    if (validate === false) {
        return {
            exists: true,
            content,
        };
    }
    if (!content)
        throw new Error("Could not parse sm.json");
    return (0, function_1.pipe)(Manifest_1.Manifest.decode(content), (0, Either_1.fold)((errors) => {
        return { exists: true, content: null, errors };
    }, (manifest) => ({ exists: true, content: manifest })));
}
exports.retrieveManifest = retrieveManifest;
function maybeRepoNameFromSMFile(cwd, base) {
    var _a;
    try {
        const baseUrl = new URL(base);
        const maybeSMFile = retrieveManifest(cwd);
        if (maybeSMFile.exists === false)
            return null;
        if (!((_a = maybeSMFile.content) === null || _a === void 0 ? void 0 : _a.apiEndpoint))
            return null;
        const repoUrl = new URL(maybeSMFile.content.apiEndpoint);
        const correctBase = repoUrl.hostname.includes(baseUrl.hostname);
        if (correctBase === false)
            return null;
        return repoUrl.hostname.split(".")[0];
    }
    catch {
        return null;
    }
}
exports.maybeRepoNameFromSMFile = maybeRepoNameFromSMFile;
function patchManifest(cwd, data) {
    const manifest = retrieveManifest(cwd);
    if (!manifest.exists || !manifest.content)
        return false;
    const updatedManifest = {
        ...manifest.content,
        ...data,
    };
    files_1.default.write((0, paths_1.SMConfigPath)(cwd), updatedManifest);
    return true;
}
exports.patchManifest = patchManifest;
function updateManifestSMVersion(cwd, version) {
    const maybeManifest = retrieveManifest(cwd);
    if (maybeManifest.errors) {
        const messages = (0, io_ts_reporters_1.formatValidationErrors)(maybeManifest.errors);
        messages.forEach((message) => {
            console.log("[core/sm.json] " + message);
        });
    }
    const content = maybeManifest.content;
    if (content === null || content === void 0 ? void 0 : content._latest)
        return false;
    return patchManifest(cwd, { _latest: version });
}
exports.updateManifestSMVersion = updateManifestSMVersion;
