// @slicemachine/core/build/node-utils/pkg.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackageJsonHelper = exports.PackageJsonC = exports.addJsonPackageSmScript = exports.patchJsonPackage = exports.retrieveJsonPackage = void 0;
const t = __importStar(require("io-ts"));
const Either_1 = require("fp-ts/Either");
const paths_1 = require("./paths");
const files_1 = __importDefault(require("./files"));
const consts_1 = require("../consts");
function retrieveJsonPackage(cwd) {
    const pkgPath = (0, paths_1.JsonPackagePath)(cwd);
    if (!files_1.default.exists(pkgPath)) {
        return {
            exists: false,
            content: null,
        };
    }
    const content = files_1.default.safeReadJson(pkgPath);
    return {
        exists: true,
        content,
    };
}
exports.retrieveJsonPackage = retrieveJsonPackage;
function patchJsonPackage(cwd, data) {
    const pkg = retrieveJsonPackage(cwd);
    if (!pkg.exists || !pkg.content)
        return false;
    const updatedPkg = {
        ...pkg.content,
        ...data,
    };
    files_1.default.write((0, paths_1.JsonPackagePath)(cwd), updatedPkg, { recursive: false });
    return true;
}
exports.patchJsonPackage = patchJsonPackage;
function addJsonPackageSmScript(cwd) {
    const pkg = retrieveJsonPackage(cwd);
    if (!pkg.exists || !pkg.content)
        return false;
    const { scripts = {} } = pkg.content;
    if (scripts[consts_1.SCRIPT_NAME])
        return false;
    return patchJsonPackage(cwd, {
        scripts: { ...scripts, [consts_1.SCRIPT_NAME]: consts_1.SCRIPT_VALUE },
    });
}
exports.addJsonPackageSmScript = addJsonPackageSmScript;
exports.PackageJsonC = t.exact(t.partial({
    name: t.union([t.string, t.void]),
    dependencies: t.record(t.string, t.string),
}));
exports.PackageJsonHelper = {
    fromPath(pkgPath) {
        return files_1.default.readEntity(pkgPath, (payload) => {
            return (0, Either_1.getOrElseW)(() => new Error(`Unable to decode package.json`))(exports.PackageJsonC.decode(payload));
        });
    },
};
