// slice-machine-ui/build/server/src/api/slices/create/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const path_1 = __importDefault(require("path"));
const util_1 = require("util");
const copy_template_dir_1 = __importDefault(require("copy-template-dir"));
const getEnv_1 = __importDefault(require("../../services/getEnv"));
const str_1 = require("@lib/utils/str");
const files_1 = __importDefault(require("@lib/utils/files"));
const consts_1 = require("@lib/consts");
const IO = __importStar(require("../../io"));
const save_1 = __importDefault(require("../save"));
const paths_1 = require("@lib/models/paths");
const consts_2 = require("@lib/consts");
const copy = util_1.promisify(copy_template_dir_1.default);
const copyTemplate = async (env, templatePath, from, sliceName) => {
    try {
        await copy(templatePath, path_1.default.join(env.cwd, from, sliceName), {
            componentName: sliceName,
            componentId: str_1.snakelize(sliceName),
            variationId: consts_1.DEFAULT_VARIATION_ID,
        });
    }
    catch (e) {
        const message = `[create] Could not copy template. Full error: ${e}`;
        console.error(message);
        return {
            err: new Error(message),
            status: 500,
            reason: message,
        };
    }
};
const fromTemplate = async (env, from, sliceName) => {
    const templatePath = path_1.default.join(appRoot, "templates", "slice", env.framework);
    if (!files_1.default.isDirectory(templatePath)) {
        const message = `[create] Framework "${env.framework}" is not supported. (${templatePath}).`;
        console.error(message);
        return {
            err: new Error(message),
            status: 500,
            reason: message,
        };
    }
    return copyTemplate(env, templatePath, from, sliceName);
};
async function handler({ sliceName, from, }) {
    if (consts_2.RESERVED_SLICE_NAME.includes(sliceName)) {
        const msg = `The slice name '${sliceName}' is reserved for slice machine use`;
        return { err: new Error(msg), status: 400, reason: msg };
    }
    const { env } = await getEnv_1.default();
    const pathToModel = paths_1.paths(env.cwd, "").library(from).slice(sliceName).model();
    const templatePath = paths_1.SliceTemplateConfig(env.cwd);
    if (files_1.default.exists(templatePath) && files_1.default.isDirectory(templatePath)) {
        await copyTemplate(env, templatePath, from, sliceName);
    }
    else {
        const maybeError = await fromTemplate(env, from, sliceName);
        if (maybeError) {
            return maybeError;
        }
    }
    if (files_1.default.exists(pathToModel)) {
        const smModel = IO.Slice.readSlice(pathToModel);
        const res = await save_1.default({
            body: { sliceName, from, model: smModel, mockConfig: {} },
        });
        return { ...res, variationId: consts_1.DEFAULT_VARIATION_ID };
    }
    const msg = `[create] Could not find file model.json. Exiting...`;
    return { err: new Error(msg), status: 500, reason: msg };
}
exports.default = handler;
