// slice-machine-ui/build/lib/models/common/widgets/LinkToMedia/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkToMediaWidget = void 0;
const yup = __importStar(require("yup"));
const md_1 = require("react-icons/md");
const Mock_1 = require("./Mock");
const Form_1 = require("./Mock/Form");
const fields_1 = require("../../CustomType/fields");
const Link_1 = require("@lib/models/common/widgets/Link");
const Form_2 = __importStar(require("@lib/models/common/widgets/Link/Form"));
const type_1 = require("@lib/models/common/widgets/LinkToMedia/type");
const Meta = {
    icon: md_1.MdAttachment,
    title: "Link to media",
    description: "A link to files, document and media",
};
const linkToMediaConfigSchema = Link_1.linkConfigSchema.shape({
    select: yup
        .string()
        .required()
        .matches(/^media$/, { excludeEmptyString: true }),
});
const schema = yup.object().shape({
    type: yup
        .string()
        .matches(/^Link$/, { excludeEmptyString: true })
        .required(),
    config: linkToMediaConfigSchema.optional(),
});
exports.LinkToMediaWidget = {
    handleMockConfig: Mock_1.handleMockConfig,
    handleMockContent: Mock_1.handleMockContent,
    MockConfigForm: Form_1.MockConfigForm,
    Meta,
    FormFields: Form_2.FormFields,
    schema,
    Form: Form_2.default,
    create: (label) => new type_1.LinkToMediaField({ label }),
    TYPE_NAME: fields_1.FieldType.Link,
    CUSTOM_NAME: "LinkToMedia",
};
