// slice-machine-ui/build/lib/models/common/widgets/Boolean/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BooleanWidget = void 0;
const yup = __importStar(require("yup"));
const bs_1 = require("react-icons/bs");
const Mock_1 = require("./Mock");
const Form_1 = require("./Mock/Form");
const type_1 = require("./type");
const utils_1 = require("../../../../utils");
const forms_1 = require("../../../../forms");
const defaults_1 = require("../../../../forms/defaults");
const fields_1 = require("../../../../forms/fields");
const fields_2 = require("../../CustomType/fields");
const Meta = {
    icon: bs_1.BsToggleOn,
    title: "Boolean",
    description: "An input that is either true or false",
};
const FormFields = {
    label: defaults_1.DefaultFields.label,
    id: defaults_1.DefaultFields.id,
    placeholder_false: fields_1.Input("False Placeholder", { required: false }, null, "false"),
    placeholder_true: fields_1.Input("True Placeholder", { required: false }, null, "true"),
    default_value: fields_1.CheckBox("Default to true"),
};
const schema = yup.object().shape({
    type: yup
        .string()
        .matches(/^Boolean$/, { excludeEmptyString: true })
        .required(),
    config: forms_1.createValidationSchema(utils_1.removeProp(FormFields, "id")),
});
exports.BooleanWidget = {
    TYPE_NAME: fields_2.FieldType.Boolean,
    handleMockContent: Mock_1.handleMockContent,
    handleMockConfig: Mock_1.handleMockConfig,
    MockConfigForm: Form_1.MockConfigForm,
    create: (label) => new type_1.BooleanField({ label }),
    Meta,
    schema,
    FormFields,
};
