// slice-machine-ui/build/lib/models/common/http/DefaultClient.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const path_1 = __importDefault(require("path"));
const Either_1 = require("fp-ts/Either");
const upload_1 = __importDefault(require("./upload"));
const files_1 = __importDefault(require("../../../utils/files"));
const UserProfile_1 = require("@slicemachine/core/build/models/UserProfile");
const axios_1 = __importDefault(require("axios"));
const SharedSlicesApi = {
    STAGE: "https://customtypes.wroom.io/",
    PROD: "https://customtypes.prismic.io/",
};
const AuthApi = {
    STAGE: "https://auth.wroom.io/",
    PROD: "https://auth.prismic.io/",
};
const UserService = {
    STAGE: "https://user.wroom.io/",
    PROD: "https://user.internal-prismic.io/",
};
const AclProviderApi = {
    STAGE: "https://2iamcvnxf4.execute-api.us-east-1.amazonaws.com/stage/",
    PROD: "https://0yyeb2g040.execute-api.us-east-1.amazonaws.com/prod/",
};
const SlicesPrefix = "slices/";
const ValidatePrefix = "validate/";
const RefreshTokenPrefix = "refreshtoken/";
const CustomTypesPrefix = "customtypes/";
function createApiUrl(base, { STAGE, PROD }) {
    if (base && base.includes("wroom.io")) {
        return STAGE;
    }
    return PROD;
}
function createFetcher(apiUrl, repo, auth) {
    return function runFetch(prefix, body, action = "", method = "get") {
        const headers = {
            repository: repo,
            Authorization: `Bearer ${auth}`,
            "User-Agent": "slice-machine",
        };
        return fetch(new URL(action, `${apiUrl}${prefix}`).toString(), {
            headers,
            method,
            ...(method === "post"
                ? {
                    body: "object" === typeof body ? JSON.stringify(body) : body,
                }
                : null),
        });
    };
}
const initFetcher = (base, ApiUrls, devConfigArgs, repo, auth) => {
    const apiUrl = createApiUrl(base, ApiUrls);
    const args = devConfigArgs ? devConfigArgs : [apiUrl, repo, auth];
    return createFetcher(args[0], args[1], args[2]);
};
class DefaultClient {
    constructor(cwd, base, repo, auth) {
        this.cwd = cwd;
        this.base = base;
        this.repo = repo;
        this.auth = auth;
        this.images = {
            createAcl: async () => {
                return this.aclFetcher("", undefined, "create", "get");
            },
            deleteFolder: async (body) => {
                return this.aclFetcher("", body, "delete-folder", "post");
            },
            post: async (params) => {
                return upload_1.default(params);
            },
        };
        const devConfig = (() => {
            if (!cwd) {
                return {};
            }
            try {
                return files_1.default.readJson(path_1.default.join(cwd, "sm.dev.json"));
            }
            catch (e) {
                return {};
            }
        })();
        this.base = base;
        this.auth = auth;
        this.apiFetcher = initFetcher(base, SharedSlicesApi, devConfig.sharedSlicesApi, repo, auth);
        this.aclFetcher = initFetcher(base, AclProviderApi, devConfig.aclProviderApi, repo, auth);
    }
    static validate(base, auth) {
        return fetch(`${createApiUrl(base, AuthApi)}${ValidatePrefix}?token=${auth}`, {
            method: "GET",
        });
    }
    static refreshToken(base, auth) {
        return fetch(`${createApiUrl(base, AuthApi)}${RefreshTokenPrefix}?token=${auth}`, {
            method: "GET",
        });
    }
    static async profile(base, auth) {
        const result = await axios_1.default.get(`${createApiUrl(base, UserService)}profile`, { headers: { Authorization: `Bearer ${auth}` } });
        return Either_1.getOrElseW(() => {
            throw new Error(`Unable to parse profile: ${JSON.stringify(result.data)}`);
        })(UserProfile_1.UserProfile.decode(result.data));
    }
    async getSlice() {
        return this.apiFetcher(SlicesPrefix);
    }
    async getCustomTypes() {
        return this.apiFetcher(CustomTypesPrefix);
    }
    async insertCustomType(body) {
        return this.apiFetcher(CustomTypesPrefix, body, "insert", "post");
    }
    async updateCustomType(body) {
        return this.apiFetcher(CustomTypesPrefix, body, "update", "post");
    }
    async insertSlice(body) {
        return this.apiFetcher(SlicesPrefix, body, "insert", "post");
    }
    async updateSlice(body) {
        return this.apiFetcher(SlicesPrefix, body, "update", "post");
    }
}
exports.default = DefaultClient;
