// slice-machine-ui/build/lib/models/common/http/FakeClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class FakeClient {
    constructor() {
        this.images = {
            createAcl: async () => {
                return {
                    status: 200,
                    json() {
                        return {};
                    },
                };
            },
            deleteFolder: async () => {
                return { status: 200 };
            },
            post: async () => {
                return { status: 200 };
            },
        };
        this.mutate = (jsonResponse = []) => ({
            status: 403,
            statusText: "Unauthorized",
            fake: true,
            err: "You are not connected to Prismic",
            json() {
                return jsonResponse;
            },
        });
    }
    async getSlice() {
        return {
            status: 200,
            statusText: "ok",
            fake: true,
            json() {
                return [];
            },
        };
    }
    async getCustomTypes() {
        return {
            status: 200,
            statusText: "ok",
            fake: true,
            json() {
                return [];
            },
        };
    }
    async insertCustomType() {
        return this.mutate({});
    }
    async updateCustomType() {
        return this.mutate({});
    }
    async insertSlice() {
        return this.mutate();
    }
    async updateSlice() {
        return this.mutate();
    }
}
exports.default = FakeClient;
